From 72f81cd0693c1abfdb5d86bf207cfcd3201646d7 Mon Sep 17 00:00:00 2001
Message-Id: <72f81cd0693c1abfdb5d86bf207cfcd3201646d7.1528231893.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Tue, 5 Jun 2018 16:51:28 -0400
Subject: [PATCH 1/1] Compile issues

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 SPECS/kernel-rt.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/kernel-rt.spec b/SPECS/kernel-rt.spec
index ab6daf3..3c0e4b4 100644
--- a/SPECS/kernel-rt.spec
+++ b/SPECS/kernel-rt.spec
@@ -420,6 +420,7 @@ Patch1029: dpt_i2o-fix-build-warning.patch
 # DRBD was choking on write same
 Patch1030: turn-off-write-same-in-smartqpi-driver.patch
 Patch1031: restrict-iSCSI-kthreads-to-CPUs-in-cpu_kthread_mask.patch
+Patch1032: fix-compilation-issues.patch
 
 BuildRoot: %{_tmppath}/kernel-%{KVERREL}-root
 
@@ -784,6 +785,7 @@ ApplyPatch aic94xx-Skip-reading-user-settings-if-flash-is-not-f.patch
 ApplyPatch dpt_i2o-fix-build-warning.patch
 ApplyPatch turn-off-write-same-in-smartqpi-driver.patch
 ApplyPatch restrict-iSCSI-kthreads-to-CPUs-in-cpu_kthread_mask.patch
+ApplyPatch fix-compilation-issues.patch
 
 # move off upstream version mechanism
 if [ -e localversion-rt ]; then
-- 
1.8.3.1

