From 8d084dcc717666af0283e4ddeefaba7e331d879a Mon Sep 17 00:00:00 2001
Message-Id: <8d084dcc717666af0283e4ddeefaba7e331d879a.1524081065.git.Jim.Somerville@windriver.com>
From: Scott Little <scott.little@windriver.com>
Date: Tue, 9 May 2017 13:00:12 -0400
Subject: [PATCH 01/39] tis debrand and add centos certificate

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 SPECS/kernel-rt.spec | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/SPECS/kernel-rt.spec b/SPECS/kernel-rt.spec
index b855a01..5f4a216 100644
--- a/SPECS/kernel-rt.spec
+++ b/SPECS/kernel-rt.spec
@@ -340,6 +340,12 @@ Source25: merge.pl
 Source27: sanity_check.py
 Source29: extrakeys.pub
 
+Source37: centos.cer
+%if %{?released_kernel}
+%define pesign_name redhatsecureboot301
+%else
+%define pesign_name redhatsecureboot003
+%endif
 
 ### Configuration files
 Source50: kernel-%{version}-x86_64-rt.config
@@ -359,6 +365,9 @@ Patch001: avoid-debuginfo-hash-collision-scripts.patch
 
 # Empty final patch file to facilitate testing of kernel patches
 Patch999999: linux-kernel-test.patch
+Patch1000: debrand-single-cpu.patch
+Patch1001: debrand-rh_taint.patch
+Patch1002: debrand-rh-i686-cpu.patch
 
 BuildRoot: %{_tmppath}/kernel-%{KVERREL}-root
 
@@ -799,7 +808,7 @@ BuildKernel() {
     fi
 # EFI SecureBoot signing, x86_64-only
 %ifarch x86_64
-    %pesign -s -i $KernelImage -o $KernelImage.signed -a %{SOURCE13} -c %{SOURCE14} -n %{pesign_name}
+    %pesign -s -i $KernelImage -o $KernelImage.signed -a %{SOURCE37} -c %{SOURCE37} -n %{pesign_name}
     mv $KernelImage.signed $KernelImage
 %endif
     $CopyKernel $KernelImage $RPM_BUILD_ROOT/%{image_install_path}/$InstallName-$KernelVer
-- 
1.8.3.1

