From a64cf29a67df13212e173c4aa74fe9de538ea97d Mon Sep 17 00:00:00 2001
From: Angie Wang <Angie.Wang@windriver.com>
Date: Mon, 28 May 2018 20:18:19 -0400
Subject: [PATCH 1/1] meta add NotIn validation

---
 SPECS/python-voluptuous.spec | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/SPECS/python-voluptuous.spec b/SPECS/python-voluptuous.spec
index 2b1217c..9f82511 100644
--- a/SPECS/python-voluptuous.spec
+++ b/SPECS/python-voluptuous.spec
@@ -6,7 +6,7 @@
 
 Name: python-%{upname}
 Version: 0.8.9
-Release: 1%{?dist}
+Release: 1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary: A Python data validation library
 License: BSD
 Group: Development/Languages
@@ -15,6 +15,10 @@ URL: http://github.com/alecthomas/voluptuous
 # Use the tarball from github instead
 #Source0: http://pypi.python.org/packages/source/v/voluptuous/%{upname}-%{version}.tar.gz
 Source0: http://github.com/alecthomas/voluptuous/archive/%{version}.tar.gz#/%{upname}-%{version}.tar.gz
+
+# WRS
+Patch0001: 0001-add-NotIn-validation.patch
+
 BuildRequires: python2-devel python-setuptools
 
 BuildArch: noarch
@@ -38,6 +42,7 @@ primarily intended for validating data coming into Python as JSON, YAML, etc.
 
 %prep
 %setup -q -n %{upname}-%{version}
+%patch0001 -p1
 rm -rf %{upname}.egg-info
 
 %if 0%{?with_python3}
-- 
1.8.3.1

