From 1c898dfc32b11e94ad90ab76fcff2ba2b65dfa6d Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Tue, 31 Jan 2017 16:51:23 -0500
Subject: [PATCH] calculate rather than hardcode shim-unsigned version

---
 SPECS/shim-signed.spec | 13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/SPECS/shim-signed.spec b/SPECS/shim-signed.spec
index f271a05..1b00621 100644
--- a/SPECS/shim-signed.spec
+++ b/SPECS/shim-signed.spec
@@ -4,10 +4,6 @@ Release:        2%{?_tis_dist}.%{tis_patch_ver}
 Summary:        First-stage UEFI bootloader
 Provides:	shim = %{version}-%{release}
 
-# note that tis_patch_ver cannot be used in the unsigned_release definition,
-# as the variable represents the patch level of shim-signed, and we have to
-# specifiy the patch of shim-unsigned
-%define unsigned_release 1.el7%{_tis_dist}.1
 
 License:        BSD
 URL:            http://www.codon.org.uk/~mjg59/shim/
@@ -32,14 +28,13 @@ Source5:	BOOT.CSV
 %global efiarchlc aa64
 %global shimsrc %{SOURCE2}
 %endif
-%define unsigned_dir %{_datadir}/shim/%{efiarchlc}-%{version}-%{unsigned_release}/
+
 
 BuildRequires: git
 BuildRequires: openssl-devel openssl
 BuildRequires: pesign >= 0.106-5%{dist}
 BuildRequires: efivar-devel
-# BuildRequires: shim-unsigned = %{version}-%{unsigned_release}
-BuildRequires: shim-unsigned = %{version}-%{unsigned_release}
+BuildRequires: shim-unsigned 
 
 # for mokutil's configure
 BuildRequires: autoconf automake
@@ -119,6 +114,10 @@ git config --unset user.name
 # if we already have a presigned EFI image, then do not do signing -- just
 # use the presigned one.
 
+# %define unsigned_release 1.el7%{_tis_dist}.1
+%global unsigned_release %(rpm -q --queryformat '%%{RELEASE}' shim-unsigned | sort --version-sort | tail -1)
+%define unsigned_dir %{_datadir}/shim/%{efiarchlc}-%{version}-%{unsigned_release}/
+
 if [ -e %{unsigned_dir}shim-presigned.efi ]; then
 	cp %{unsigned_dir}shim-presigned.efi shim.efi
 	cp %{unsigned_dir}shim-presigned.efi shim-%{efidir}.efi
-- 
1.8.3.1

