From 16a82ea84332a117c4524caaa4209b912e18e888 Mon Sep 17 00:00:00 2001
From: Bin Qian <bin.qian@windriver.com>
Date: Wed, 6 Dec 2017 08:47:12 -0500
Subject: [PATCH 1/1] TiS tboot

---
 SPECS/tboot.spec | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/SPECS/tboot.spec b/SPECS/tboot.spec
index c2d5eb7..f04dd17 100644
--- a/SPECS/tboot.spec
+++ b/SPECS/tboot.spec
@@ -12,9 +12,10 @@ Source0:        http://downloads.sourceforge.net/%{name}/%{name}-%{version}.tar.
 BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 
 Patch01:	0001-MANPATH-should-not-be-used-as-install-dir.patch
+Patch999:       1000-tboot-for-tis.patch
 
 BuildRequires:  trousers-devel
-BuildRequires:  openssl-devel
+BuildRequires:  openssl-devel git
 ExclusiveArch:  x86_64
 
 %description
@@ -24,7 +25,13 @@ and verified launch of an OS kernel/VMM.
 
 %prep
 %setup -q
-%patch01 -p1 -b .0001
+
+git init
+git config user.email "example@example.com"
+git config user.name "RHEL example"
+git add .
+git commit -a -q -m "baseline."
+git am %{patches}
 
 %build
 CFLAGS="$RPM_OPT_FLAGS"; export CFLAGS
-- 
2.7.4

