From 49c76cf9f9025a94b64fff0bb0746c74f536e620 Mon Sep 17 00:00:00 2001
From: Joseph Richard <Joseph.Richard@windriver.com>
Date: Fri, 10 Nov 2017 11:31:49 -0500
Subject: [PATCH 3/8] control provider network testing state

---
 manifests/init.pp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/manifests/init.pp b/manifests/init.pp
index 6f611ef..6a223df 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -443,6 +443,7 @@ class neutron (
   $host_driver                          = 'neutron.drivers.host.NoopHostDriver',
   $fm_driver                            = 'neutron.drivers.fm.NoopFmDriver',
   $vlan_transparent                     = false,
+  $pnet_audit_enabled                   = true,
   # DEPRECATED PARAMETERS
   $rabbit_password                      = $::os_service_default,
   $rabbit_host                          = $::os_service_default,
@@ -531,6 +532,7 @@ instead.")
     'DEFAULT/host_driver':                  value => $host_driver;
     'DEFAULT/fm_driver':                    value => $fm_driver;
     'DEFAULT/vlan_transparent':             value => $vlan_transparent;
+    'pnet_connectivity/pnet_audit_enabled': value => $pnet_audit_enabled;
   }
 
   oslo::messaging::default { 'neutron_config':
-- 
1.8.3.1

