From bcdf10df2ba5f68bb61ff2b86f0a6cb813271e7d Mon Sep 17 00:00:00 2001
From: Joseph Richard <Joseph.Richard@windriver.com>
Date: Fri, 10 Nov 2017 11:31:50 -0500
Subject: [PATCH 4/8] set wsgi pool size to 10 from 100

---
 manifests/init.pp | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/manifests/init.pp b/manifests/init.pp
index 6a223df..78b994b 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -329,6 +329,10 @@
 #      transport://user:pass@host1:port[,hostN:portN]/virtual_host
 #   Defaults to $::os_service_default.
 #
+# [*wsgi_default_pool_size*]
+#   (optional) Size of the pool of greenthreads used by wsgi (integer value)
+#   Defaults to $::os_service_default.
+#
 #
 # WRS PARAMETERS
 #
@@ -439,6 +443,7 @@ class neutron (
   $notification_driver                  = $::os_service_default,
   $notification_topics                  = $::os_service_default,
   $notification_transport_url           = $::os_service_default,
+  $wsgi_default_pool_size               = 100,
   # WRS
   $host_driver                          = 'neutron.drivers.host.NoopHostDriver',
   $fm_driver                            = 'neutron.drivers.fm.NoopFmDriver',
@@ -549,6 +554,13 @@ instead.")
     transport_url => $notification_transport_url,
   }
 
+  oslo::service { 'neutron_config':
+    wsgi_default_pool_size => $wsgi_default_pool_size,
+    cert_file => $cert_file,
+    key_file => $key_file,
+    ca_file => $ca_file,
+  }
+
   if ! is_service_default ($service_plugins) and ($service_plugins) {
     if is_array($service_plugins) {
       neutron_config { 'DEFAULT/service_plugins': value => join($service_plugins, ',') }
@@ -602,9 +614,6 @@ instead.")
   # SSL Options
   neutron_config {
     'DEFAULT/use_ssl': value => $use_ssl;
-    'ssl/cert_file':   value => $cert_file;
-    'ssl/key_file':    value => $key_file;
-    'ssl/ca_file':     value => $ca_file;
   }
 
 }
-- 
1.8.3.1

