From c74cce7e2b7aa268f8f9b2238e662879701fe9d4 Mon Sep 17 00:00:00 2001
From: Joseph Richard <Joseph.Richard@windriver.com>
Date: Fri, 10 Nov 2017 11:31:50 -0500
Subject: [PATCH 6/8] Add parameters to classes

---
 manifests/init.pp   |  2 ++
 manifests/server.pp | 13 +++++++++++++
 2 files changed, 15 insertions(+)

diff --git a/manifests/init.pp b/manifests/init.pp
index 78b994b..4e76091 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -449,6 +449,7 @@ class neutron (
   $fm_driver                            = 'neutron.drivers.fm.NoopFmDriver',
   $vlan_transparent                     = false,
   $pnet_audit_enabled                   = true,
+  $log_format                           = '[%(name)s] %(message)s',
   # DEPRECATED PARAMETERS
   $rabbit_password                      = $::os_service_default,
   $rabbit_host                          = $::os_service_default,
@@ -537,6 +538,7 @@ instead.")
     'DEFAULT/host_driver':                  value => $host_driver;
     'DEFAULT/fm_driver':                    value => $fm_driver;
     'DEFAULT/vlan_transparent':             value => $vlan_transparent;
+    'DEFAULT/log_format':                   value => $log_format;
     'pnet_connectivity/pnet_audit_enabled': value => $pnet_audit_enabled;
   }
 
diff --git a/manifests/server.pp b/manifests/server.pp
index 8a1a3f7..b2cc64c 100644
--- a/manifests/server.pp
+++ b/manifests/server.pp
@@ -258,6 +258,7 @@ class neutron::server (
   $enable_proxy_headers_parsing     = $::os_service_default,
   # WRS
   $router_status_managed            = $neutron::params::router_status_managed,
+  $vhost_user_enabled               = true,
   # DEPRECATED PARAMETERS
   $log_dir                          = undef,
   $log_file                         = undef,
@@ -378,6 +379,18 @@ class neutron::server (
     }
   }
 
+  # WRS
+  if (str2bool($::is_virtual)) {
+    $vhost_user_enabled_real = false
+  } else {
+    $vhost_user_enabled_real = $vhost_user_enabled
+  }
+
+  neutron_config {
+    # Set to false on virtualbox, true otherwise
+    'vhost/vhost_user_enabled': value => $vhost_user_enabled_real;
+  }
+
   if ($auth_strategy == 'keystone') {
 
     include ::neutron::keystone::authtoken
-- 
1.8.3.1

