From 535eddc2aae95719cf40becca35058f7670ebb4e Mon Sep 17 00:00:00 2001
From: Joseph Richard <Joseph.Richard@windriver.com>
Date: Fri, 10 Nov 2017 11:31:51 -0500
Subject: [PATCH 7/8] update neutron bgp to use class variables

---
 manifests/bgp.pp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/manifests/bgp.pp b/manifests/bgp.pp
index d61656c..07029f3 100644
--- a/manifests/bgp.pp
+++ b/manifests/bgp.pp
@@ -23,8 +23,8 @@ class neutron::bgp (
   Neutron_bgp_dragent_config<||> ~> Service['neutron-bgp-dragent']
 
   neutron_bgp_dragent_config {
-    'bgp/bgp_router_id': value => $neutron_bgp_router_id;
-    'bgp/bgp_speaker_driver': value => $neutron_bgp_speaker_driver;
+    'bgp/bgp_router_id': value => $bgp_router_id;
+    'bgp/bgp_speaker_driver': value => $bgp_speaker_driver;
   }
 
   service { 'neutron-bgp-dragent':
-- 
1.8.3.1

