From a140bd7cdc5bdac336619544141aa64daf54590a Mon Sep 17 00:00:00 2001
From: Daniel Badea <daniel.badea@windriver.com>
Date: Tue, 31 Oct 2017 16:00:04 +0200
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/puppet-cinder.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-cinder.spec b/SPECS/puppet-cinder.spec
index 6af9d49..f9089b6 100644
--- a/SPECS/puppet-cinder.spec
+++ b/SPECS/puppet-cinder.spec
@@ -1,7 +1,7 @@
 %{!?upstream_version: %global upstream_version %{version}%{?milestone}}
 Name:           puppet-cinder
 Version:        11.3.0
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Puppet module for OpenStack Cinder
 License:        ASL 2.0
 
-- 
2.7.4

