diff --git a/scripts/drbd.ocf b/scripts/drbd.ocf
index 2105209..0aa9702 100644
--- a/scripts/drbd.ocf
+++ b/scripts/drbd.ocf
@@ -217,13 +217,17 @@ do_drbdadm() {
 }
 
 set_master_score() {
-	# Use quiet mode (-Q) to quench logging. Actual score updates
-	# will get logged by attrd anyway
-	do_cmd ${HA_SBIN_DIR}/crm_master -Q -l reboot -v $1
+	if [ -x ${HA_SBIN_DIR}/crm_master ]; then
+		# Use quiet mode (-Q) to quench logging. Actual score updates
+		# will get logged by attrd anyway
+		do_cmd ${HA_SBIN_DIR}/crm_master -Q -l reboot -v $1
+	fi
 }
 
 remove_master_score() {
-	do_cmd ${HA_SBIN_DIR}/crm_master -l reboot -D
+	if [ -x ${HA_SBIN_DIR}/crm_master ]; then
+		do_cmd ${HA_SBIN_DIR}/crm_master -l reboot -D
+	fi
 }
 
 _sh_status_process() {
