From ea19e3020367cfaf6da20dd690433ee72a24120c Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 2 May 2016 15:17:54 -0400
Subject: [PATCH 1/1] Avoid kernel/userspace version check

---
 user/drbdadm_usage_cnt.c | 32 +-------------------------------
 1 file changed, 1 insertion(+), 31 deletions(-)

diff --git a/user/drbdadm_usage_cnt.c b/user/drbdadm_usage_cnt.c
index ff6d5c8..c6cb4ad 100644
--- a/user/drbdadm_usage_cnt.c
+++ b/user/drbdadm_usage_cnt.c
@@ -244,37 +244,7 @@ static int vcs_ver_cmp(struct vcs_rel *rev1, struct vcs_rel *rev2)
 
 void warn_on_version_mismatch(void)
 {
-	char *msg;
-	int cmp;
-
-	/* get the kernel module version from /proc/drbd */
-	vcs_get_current();
-
-	/* get the userland version from REL_VERSION */
-	vcs_get_userland();
-
-	cmp = vcs_ver_cmp(&userland_version, &current_vcs_rel);
-	/* no message if equal */
-	if (cmp == 0)
-		return;
-	if (cmp > 0xffff || cmp < -0xffff)	 /* major version differs! */
-		msg = "mixing different major numbers will not work!";
-	else if (cmp < 0)		/* userland is older. always warn. */
-		msg = "you should upgrade your drbd tools!";
-	else if (cmp & 0xff00)		/* userland is newer minor version */
-		msg = "please don't mix different DRBD series.";
-	else		/* userland is newer, but only differ in sublevel. */
-		msg = "preferably kernel and userland versions should match.";
-
-	fprintf(stderr, "DRBD module version: %u.%u.%u\n"
-			"   userland version: %u.%u.%u\n%s\n",
-			current_vcs_rel.version.major,
-			current_vcs_rel.version.minor,
-			current_vcs_rel.version.sublvl,
-			userland_version.version.major,
-			userland_version.version.minor,
-			userland_version.version.sublvl,
-			msg);
+	return;
 }
 
 void add_lib_drbd_to_path(void)
-- 
1.8.3.1

