From 100b44d99b0bcbac92abd2122becbfd88d155e09 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Wed, 22 Nov 2017 20:45:28 -0500
Subject: [PATCH] Increase short cmd timeout to 15 secs

---
 user/drbdadm_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/user/drbdadm_main.c b/user/drbdadm_main.c
index b89e91a..19c5a44 100644
--- a/user/drbdadm_main.c
+++ b/user/drbdadm_main.c
@@ -1467,7 +1467,7 @@ void m__system(char **argv, int flags, const char *res_name, pid_t *kid, int *fd
 		alarm_raised = 0;
 		switch (flags & SLEEPS_MASK) {
 		case SLEEPS_SHORT:
-			timeout = 5;
+			timeout = 15;
 			break;
 		case SLEEPS_LONG:
 			timeout = COMM_TIMEOUT + 1;
-- 
1.8.3.1

