From 36f161731c1041805d4d3de98a0f280917692757 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 1 May 2017 14:39:38 -0400
Subject: [PATCH] Add TIS Patches

---
 SPECS/puppet-stdlib.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/puppet-stdlib.spec b/SPECS/puppet-stdlib.spec
index 769c0ef..7304088 100644
--- a/SPECS/puppet-stdlib.spec
+++ b/SPECS/puppet-stdlib.spec
@@ -15,6 +15,8 @@ URL:            https://github.com/puppetlabs/puppetlabs-stdlib
 
 Source0:        https://github.com/puppetlabs/%{upstream_name}/archive/%{commit}.tar.gz#/%{upstream_name}-%{shortcommit}.tar.gz
 
+Patch0001:      0001-Filter-password-in-logs.patch
+
 BuildArch:      noarch
 
 Requires:       puppet >= 2.7.0
@@ -24,6 +26,7 @@ Standard library of resources for Puppet modules.
 
 %prep
 %setup -q -n %{upstream_name}-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

