From d5890a17f5b07a9d17665c2b4138bb244ab6c680 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:58:59 -0400
Subject: [PATCH 2/2] WRS: 0001-Update-package-versioning-for-TIS-format.patch

---
 SPECS/centos-release.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/centos-release.spec b/SPECS/centos-release.spec
index be95b1b..91dad61 100644
--- a/SPECS/centos-release.spec
+++ b/SPECS/centos-release.spec
@@ -13,7 +13,7 @@
 
 Name:           centos-release
 Version:        %{base_release_version}
-Release:        %{centos_rel}%{?dist}
+Release:        %{centos_rel}.el7.centos%{?_tis_dist}.%{tis_patch_ver}
 Summary:        %{product_family} release file
 Group:          System Environment/Base
 License:        GPLv2
-- 
1.9.1

