From 9c4f92869a737451e5827686ce916169d485d3be Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:58:59 -0400
Subject: [PATCH] WRS: centos-release-include-TiS-changes.patch

---
 SPECS/centos-release.spec | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/SPECS/centos-release.spec b/SPECS/centos-release.spec
index 67f1550..02e198e 100644
--- a/SPECS/centos-release.spec
+++ b/SPECS/centos-release.spec
@@ -27,6 +27,8 @@ Provides:       system-release(releasever) = %{base_release_version}
 Source0:        centos-release-%{base_release_version}-%{centos_rel}.tar.gz
 Source1:        85-display-manager.preset
 Source2:        90-default.preset
+Source3:        issue
+Source4:        issue.net
 
 %description
 %{product_family} release files
@@ -120,10 +122,16 @@ mkdir -p %{buildroot}%{_prefix}/lib/systemd/system-preset/
 install -m 0644 %{SOURCE1} %{buildroot}%{_prefix}/lib/systemd/system-preset/
 install -m 0644 %{SOURCE2} %{buildroot}%{_prefix}/lib/systemd/system-preset/
 
+# Overwrite default issue files with cgcs related files.
+install -m 0644 %{SOURCE3} %{buildroot}/etc/issue
+install -m 0644 %{SOURCE4} %{buildroot}/etc/issue.net
+sed -i -e "s/xxxPLATFORM_RELEASExxx/%{platform_release}/g" \
+    %{buildroot}/etc/issue \
+    %{buildroot}/etc/issue.net
+
 %posttrans
 /usr/bin/uname -m | grep -q 'x86_64'  && echo 'centos' >/etc/yum/vars/contentdir || echo 'altarch' > /etc/yum/vars/contentdir
 
-
 %clean
 rm -rf %{buildroot}
 
-- 
2.7.4

