From 328f19996b93b5be5cd856e600111d0dc87c8616 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:58:59 -0400
Subject: [PATCH 2/2] WRS: 0001-Update-package-versioning-for-TIS-format.patch

---
 SPECS/centos-release.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/centos-release.spec b/SPECS/centos-release.spec
index ce64be4..8c05c28 100644
--- a/SPECS/centos-release.spec
+++ b/SPECS/centos-release.spec
@@ -14,7 +14,7 @@
 
 Name:           centos-release
 Version:        %{base_release_version}
-Release:        %{centos_rel}.1%{?dist}
+Release:        %{centos_rel}.1.el7.centos%{?_tis_dist}.%{tis_patch_ver}
 Summary:        %{product_family} release file
 Group:          System Environment/Base
 License:        GPLv2
-- 
2.7.4

