From e18f905977fa94fa20a2d9a9cc565dc8d7fe8dac Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:58:59 -0400
Subject: [PATCH 1/2] WRS: centos-release-include-TiS-changes.patch

---
 SPECS/centos-release.spec | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/SPECS/centos-release.spec b/SPECS/centos-release.spec
index 67f1550..ce64be4 100644
--- a/SPECS/centos-release.spec
+++ b/SPECS/centos-release.spec
@@ -27,6 +27,8 @@ Provides:       system-release(releasever) = %{base_release_version}
 Source0:        centos-release-%{base_release_version}-%{centos_rel}.tar.gz
 Source1:        85-display-manager.preset
 Source2:        90-default.preset
+Source3:        issue
+Source4:        issue.net
 
 %description
 %{product_family} release files
@@ -123,6 +125,12 @@ install -m 0644 %{SOURCE2} %{buildroot}%{_prefix}/lib/systemd/system-preset/
 %posttrans
 /usr/bin/uname -m | grep -q 'x86_64'  && echo 'centos' >/etc/yum/vars/contentdir || echo 'altarch' > /etc/yum/vars/contentdir
 
+# Overwrite default issue files with cgcs related files.
+install -m 0644 %{SOURCE3} %{buildroot}/etc/issue
+install -m 0644 %{SOURCE4} %{buildroot}/etc/issue.net
+sed -i -e "s/xxxPLATFORM_RELEASExxx/%{platform_release}/g" \
+    %{buildroot}/etc/issue \
+    %{buildroot}/etc/issue.net
 
 %clean
 rm -rf %{buildroot}
-- 
2.7.4

