From ebc0ab6d19290438c444058b08e56794fdb5bfb4 Mon Sep 17 00:00:00 2001
Message-Id: <ebc0ab6d19290438c444058b08e56794fdb5bfb4.1587851141.git.Jim.Somerville@windriver.com>
From: eric zhang <eric.zhang@windriver.com>
Date: Fri, 5 Jan 2018 21:05:50 -0500
Subject: [PATCH 1/2] Update package versioning for TIS format

Signed-off-by: eric zhang <eric.zhang@windriver.com>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 SPECS/libibverbs.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/libibverbs.spec b/SPECS/libibverbs.spec
index 673516e..1640a8b 100644
--- a/SPECS/libibverbs.spec
+++ b/SPECS/libibverbs.spec
@@ -12,7 +12,7 @@
 
 Name: libibverbs
 Version: 41mlnx1
-Release: OFED.5.0.0.0.9.50218
+Release: OFED.5.0.0.0.9.50218%{?_tis_dist}.%{tis_patch_ver}
 Summary: A library for direct userspace use of RDMA (InfiniBand/iWARP) hardware
 
 Group: System Environment/Libraries
-- 
1.8.3.1

