From bfa58d25ad24537cc798ad207f7e5458cf53877f Mon Sep 17 00:00:00 2001
From: eric zhang <eric.zhang@windriver.com>
Date: Fri, 5 Jan 2018 21:05:50 -0500
Subject: [PATCH 2/3] upversion with release num

Signed-off-by: eric zhang <eric.zhang@windriver.com>
Signed-off-by: Shuicheng Lin <shuicheng.lin@intel.com>
---
 SPECS/libibverbs.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/libibverbs.spec b/SPECS/libibverbs.spec
index 519e421..4faf18c 100644
--- a/SPECS/libibverbs.spec
+++ b/SPECS/libibverbs.spec
@@ -12,7 +12,7 @@
 
 Name: libibverbs
 Version: 41mlnx1
-Release: OFED.4.7.0.0.2.47329
+Release: OFED.4.7.0.0.2.47329%{?_tis_dist}.%{tis_patch_ver}
 Summary: A library for direct userspace use of RDMA (InfiniBand/iWARP) hardware
 
 Group: System Environment/Libraries
-- 
2.7.4

