From e78c78816e4e07f7c1ba8bd415195efb0cd27563 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Fri, 18 Nov 2016 13:53:47 -0500
Subject: [PATCH 1/1] libibverbs: fix build failures due to unwanted sgid bit

---
 SPECS/libibverbs.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/libibverbs.spec b/SPECS/libibverbs.spec
index b8913c6..1e7b1eb 100644
--- a/SPECS/libibverbs.spec
+++ b/SPECS/libibverbs.spec
@@ -106,6 +106,7 @@ rm -rf $RPM_BUILD_ROOT
 %if %{_with_valgrind}
 %{_libdir}/mlnx_ofed/valgrind/lib*.so
 %endif
+%dir %attr(0755,root,root) %{_includedir}/infiniband
 %{_includedir}/*
 %{_mandir}/man3/*
 %{_mandir}/man7/*
-- 
1.8.3.1

