From f6393c8e666fc5e7e58a6f7dbef8e521dc9fc205 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Fri, 18 Nov 2016 13:53:47 -0500
Subject: [PATCH 3/3] libibverbs: fix build failures due to unwanted sgid bit

---
 SPECS/libibverbs.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/libibverbs.spec b/SPECS/libibverbs.spec
index 4faf18c..83940de 100644
--- a/SPECS/libibverbs.spec
+++ b/SPECS/libibverbs.spec
@@ -111,6 +111,7 @@ rm -rf $RPM_BUILD_ROOT
 %if %{_with_valgrind}
 %{_libdir}/mlnx_ofed/valgrind/lib*.so
 %endif
+%dir %attr(0755,root,root) %{_includedir}/infiniband
 %{_includedir}/*
 %{_mandir}/man3/*
 %{_mandir}/man7/*
-- 
2.7.4

