From 287fb29ea7d1bb31af50701bd7ac0e12c1524197 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Wed, 1 Nov 2017 14:44:13 -0500
Subject: [PATCH 2/2] Add-TIS-patch

---
 SPECS/puppet-rabbitmq.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-rabbitmq.spec b/SPECS/puppet-rabbitmq.spec
index dc2c81b..c96099b 100644
--- a/SPECS/puppet-rabbitmq.spec
+++ b/SPECS/puppet-rabbitmq.spec
@@ -21,6 +21,7 @@ License:        ASL 2.0
 URL:            https://github.com/puppetlabs/puppetlabs-rabbitmq
 
 Source0:        https://github.com/puppetlabs/%{upstream_name}/archive/%{commit}.tar.gz#/%{upstream_name}-%{shortcommit}.tar.gz
+Patch0001:      0001-Roll-up-TIS-patches.patch
 
 BuildArch:      noarch
 
@@ -33,6 +34,7 @@ Installs, configures, and manages RabbitMQ.
 
 %prep
 %setup -q -n %{tarsources}-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

