From 35277432f8da1569fdb472b4459b687695807ece Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Fri, 2 Feb 2018 12:32:19 -0600
Subject: [PATCH] metapatch for fact removal

---
 SPECS/puppet-rabbitmq.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-rabbitmq.spec b/SPECS/puppet-rabbitmq.spec
index 17a12bc..7f3a412 100644
--- a/SPECS/puppet-rabbitmq.spec
+++ b/SPECS/puppet-rabbitmq.spec
@@ -25,6 +25,7 @@ Patch0001:      0001-Roll-up-TIS-patches.patch
 Patch0002:      0002-Changed-cipher-specification-to-openssl-format.patch
 Patch0003:      0003-Eliminate-Puppet-4-deprecation-warnings.patch
 Patch0004:      0004-Partially-revert-upstream-commit-f7c3a4a637d59f3065d.patch
+Patch0005:      0005-Remove-the-rabbitmq_nodename-fact.patch
 
 BuildArch:      noarch
 
@@ -41,6 +42,7 @@ Installs, configures, and manages RabbitMQ.
 %patch0002 -p1
 %patch0003 -p1
 %patch0004 -p1
+%patch0005 -p1
 
 
 find . -type f -name ".*" -exec rm {} +
-- 
1.8.3.1

