From b628fac8eec011503e5f86f17d9e68b7a2cc1e56 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 7 Mar 2017 13:17:56 -0500
Subject: [PATCH] Configure journald to forward to syslog

---
 src/journal/journald.conf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/journal/journald.conf b/src/journal/journald.conf
index 3907dfb..ef86ffe 100644
--- a/src/journal/journald.conf
+++ b/src/journal/journald.conf
@@ -12,7 +12,7 @@
 # See journald.conf(5) for details.
 
 [Journal]
-#Storage=auto
+Storage=none
 #Compress=yes
 #Seal=yes
 #SplitMode=uid
@@ -27,7 +27,7 @@
 #RuntimeMaxFileSize=
 #MaxRetentionSec=
 #MaxFileSec=1month
-#ForwardToSyslog=yes
+ForwardToSyslog=yes
 #ForwardToKMsg=no
 #ForwardToConsole=no
 #ForwardToWall=yes
-- 
1.8.3.1

