From f0d91192bc8778899d01e7723ababcf7a3808c91 Mon Sep 17 00:00:00 2001
From: Kam Nasim <kam.nasim@windriver.com>
Date: Fri, 6 Oct 2017 14:10:25 -0400
Subject: [PATCH] meta patch for enabling audispd and LOG_AUTH facility

---
 SPECS/audit.spec | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/SPECS/audit.spec b/SPECS/audit.spec
index e144332..a188b89 100644
--- a/SPECS/audit.spec
+++ b/SPECS/audit.spec
@@ -16,6 +16,10 @@ Patch2: audit-2.7.5-no-backlog-wait-time.patch
 Patch3: audit-2.7.7-queue_error_action.patch
 # BZ 1460110 - aureport does not report all anomalies
 Patch4: audit-2.7.7-aureport.patch
+
+# WRS Patches
+Patch1000: 0001-enable-audispd-and-auth-facility.patch
+
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 BuildRequires: openldap-devel
 BuildRequires: swig
@@ -95,6 +99,9 @@ behavior.
 %patch3 -p1
 %patch4 -p1
 
+# WRS patches
+%patch1000 -p1
+
 %build
 %configure --sbindir=/sbin --libdir=/%{_lib} --with-python=yes --with-libwrap --enable-gssapi-krb5=yes --with-libcap-ng=yes --with-arm --with-aarch64 \
 --without-golang --enable-zos-remote --enable-systemd
-- 
1.9.1

