From dc0677a11cede908f266ce4695320759b4694d18 Mon Sep 17 00:00:00 2001
From: Kam Nasim <kam.nasim@windriver.com>
Date: Fri, 6 Oct 2017 14:10:25 -0400
Subject: [PATCH 2/2] meta patch for enabling audispd and LOG_AUTH facility

---
 SPECS/audit.spec | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/SPECS/audit.spec b/SPECS/audit.spec
index 2bbd782..9e70f5d 100644
--- a/SPECS/audit.spec
+++ b/SPECS/audit.spec
@@ -20,6 +20,10 @@ Patch4: audit-2.8.2-ipv6-bind.patch
 Patch5: audit-2.8.2-fix-reset-lost-return.patch
 # This patch makes date a numeric field so auparse_search works
 Patch6: audit-2.8.2-auparse-numeric_field.patch
+
+# WRS Patches
+Patch1000: 0001-enable-audispd-and-auth-facility.patch
+
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 BuildRequires: openldap-devel
 BuildRequires: swig
@@ -100,6 +104,9 @@ like relay events to remote machines.
 %patch5 -p1
 %patch6 -p1
 
+# WRS patches
+%patch1000 -p1
+
 %build
 %configure --sbindir=/sbin --libdir=/%{_lib} --with-python=yes \
 	--with-libwrap --enable-gssapi-krb5=yes \
-- 
2.7.4

