From 296b76674e19cfed71b77735f38c158b41a991b1 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 14 Aug 2018 02:57:08 +0800
Subject: [PATCH] Update package versioning for TIS format

Signed-off-by: zhipengl <zhipengs.liu@intel.com>
---
 SPECS/parted.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/parted.spec b/SPECS/parted.spec
index 9316ebe..a79b888 100644
--- a/SPECS/parted.spec
+++ b/SPECS/parted.spec
@@ -4,7 +4,7 @@
 Summary: The GNU disk partition manipulation program
 Name:    parted
 Version: 3.1
-Release: 29%{?dist}
+Release: 29.el7%{?_tis_dist}.%{tis_patch_ver}
 License: GPLv3+
 Group:   Applications/System
 URL:     http://www.gnu.org/software/parted
-- 
2.7.4

