From 4859fab94b298a4ca449216c3035b8b024c661a3 Mon Sep 17 00:00:00 2001
From: Kristine Bujold <kristine.bujold@windriver.com>
Date: Wed, 21 Dec 2016 15:28:53 -0500
Subject: [PATCH 1/2] WRS: syscalls.patch

---
 SPECS/parted.spec | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/SPECS/parted.spec b/SPECS/parted.spec
index 749687f..644c2df 100644
--- a/SPECS/parted.spec
+++ b/SPECS/parted.spec
@@ -12,6 +12,7 @@ URL:     http://www.gnu.org/software/parted
 Source0: ftp://ftp.gnu.org/gnu/%{name}/%{name}-%{version}.tar.xz
 Source1: ftp://ftp.gnu.org/gnu/%{name}/%{name}-%{version}.tar.xz.sig
 Source2: pubkey.jim.meyering
+Source3: resizepart.sh
 
 Patch0: parted-3.0-libparted-copy-pmbr_boot-when-duplicating-GPT-disk.patch
 Patch1: parted-3.1-libparted-check-PMBR-before-GPT-partition-table-8052.patch
@@ -59,6 +60,9 @@ Patch40: 0040-partprobe-Open-the-device-once-for-probing-1339705.patch
 Patch41: 0041-tests-Stop-timing-t9040-1172675.patch
 Patch42: 0042-tests-Set-optimal-blocks-to-64-for-scsi_debug-device.patch
 
+# WRS
+Patch43: syscalls.patch
+
 Buildroot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 BuildRequires: e2fsprogs-devel
 BuildRequires: readline-devel
@@ -121,7 +125,7 @@ git commit -a -m "run iconv"
 autoreconf
 autoconf
 CFLAGS="$RPM_OPT_FLAGS -Wno-unused-but-set-variable"; export CFLAGS
-%configure --enable-selinux --disable-static
+%configure --enable-selinux --disable-static --disable-device-mapper
 # Don't use rpath!
 %{__sed} -i 's|^hardcode_libdir_flag_spec=.*|hardcode_libdir_flag_spec=""|g' libtool
 %{__sed} -i 's|^runpath_var=LD_RUN_PATH|runpath_var=DIE_RPATH_DIE|g' libtool
@@ -150,6 +154,9 @@ popd
 
 %find_lang %{name}
 
+# WRS
+install -d %{buildroot}%{_sbindir}
+install -m 755 %{SOURCE3} %{buildroot}%{_sbindir}/resizepart.sh
 
 %check
 export LD_LIBRARY_PATH=$(pwd)/libparted/.libs
@@ -185,6 +192,9 @@ fi
 %{_libdir}/libparted-fs-resize.so*
 %{_infodir}/parted.info.gz
 
+# WRS
+%{_sbindir}/resizepart.sh
+
 %files devel
 %defattr(-,root,root,-)
 %{_includedir}/parted
-- 
1.8.3.1

