From 292ba3e769183377dfccc012e979ddbc53fc0356 Mon Sep 17 00:00:00 2001
From: Andy Ning <andy.ning@windriver.com>
Date: Tue, 12 Jun 2018 09:19:40 -0400
Subject: [PATCH 1/1] add username and password authentication support to api

---
 SPECS/puppet-glance.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-glance.spec b/SPECS/puppet-glance.spec
index 1e49ae0..5bb536f 100644
--- a/SPECS/puppet-glance.spec
+++ b/SPECS/puppet-glance.spec
@@ -10,6 +10,7 @@ URL:            https://launchpad.net/puppet-glance
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 Patch0001:      0001-Roll-up-TIS-patches.patch
 Patch0002:      0002-Add-region-cache-support.patch
+Patch0003:      0003-add-username-password-authentication-support-to-api.patch
 
 BuildArch:      noarch
 
@@ -27,6 +28,7 @@ Puppet module for OpenStack Glance
 %setup -q -n openstack-glance-%{upstream_version}
 %patch0001 -p1
 %patch0002 -p1
+%patch0003 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

