From 0b4a6feba98f0e0bfc3db007463cd43a43797d45 Mon Sep 17 00:00:00 2001
From: Dahir Osman <dahir.osman@windriver.com>
Date: Tue, 14 Jun 2016 14:55:39 -0400
Subject: [PATCH] Add build dependency

To fix package complaining that libnl is not available
---
 SPECS/libibverbs.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/libibverbs.spec b/SPECS/libibverbs.spec
index 4fc96d1..864c31c 100644
--- a/SPECS/libibverbs.spec
+++ b/SPECS/libibverbs.spec
@@ -22,6 +22,7 @@ Source: http://openfabrics.org/downloads/verbs/libibverbs-%{version}.tar.gz
 BuildRoot: %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-XXXXXX)
 Requires(post): /sbin/ldconfig
 Requires(postun): /sbin/ldconfig
+BuildRequires: libnl-devel
 %if %{_with_valgrind}
 BuildRequires: valgrind-devel
 %endif
-- 
1.8.3.1

