From ebe97c306c3251fc4ba58110ad4f0fa709d04969 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:22:44 -0400
Subject: [PATCH] WRS: 0001-Update-package-versioning-for-TIS-format.patch

---
 SPECS/logrotate.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/logrotate.spec b/SPECS/logrotate.spec
index c22af07..0dbde7d 100644
--- a/SPECS/logrotate.spec
+++ b/SPECS/logrotate.spec
@@ -1,7 +1,7 @@
 Summary: Rotates, compresses, removes and mails system log files
 Name: logrotate
 Version: 3.8.6
-Release: 15%{?dist}
+Release: 15.el7%{?_tis_dist}.%{tis_patch_ver}
 License: GPL+
 Group: System Environment/Base
 URL: https://github.com/logrotate/logrotate
-- 
2.7.4

