From dfd8e1c325e43cc8a5eb5ba939e355f89515a1d3 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Fri, 19 Jan 2018 23:04:46 -0500
Subject: Add upstream patches

---
 SPECS/logrotate.spec | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/SPECS/logrotate.spec b/SPECS/logrotate.spec
index 0eac8bf..99a22ac 100644
--- a/SPECS/logrotate.spec
+++ b/SPECS/logrotate.spec
@@ -48,6 +48,9 @@ Patch17: logrotate-3.8.6-config-mode-err.patch
 # fix #1483800 - update references to project page
 Patch18: logrotate-3.8.6-upstream-url.patch
 
+Patch10001: 0001-createOutputFile-rename-already-existing-file.patch
+Patch10002: 0002-createOutputFile-eliminate-stat-open-TOCTOU-race.patch
+
 Requires: coreutils >= 5.92 popt
 BuildRequires: libselinux-devel popt-devel libacl-devel acl
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
@@ -85,6 +88,9 @@ log files on your system.
 %patch17 -p1
 %patch18 -p1
 
+%patch10001 -p1
+%patch10002 -p1
+
 %build
 make %{?_smp_mflags} RPM_OPT_FLAGS="$RPM_OPT_FLAGS" WITH_SELINUX=yes WITH_ACL=yes
 
-- 
2.7.4

