From 487be212ccf9debb2c15dcc360ee4c429779b6d0 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 26 Sep 2016 17:50:11 -0400
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/python-django.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/python-django.spec b/SPECS/python-django.spec
index d0e54bb..5cf8579 100644
--- a/SPECS/python-django.spec
+++ b/SPECS/python-django.spec
@@ -29,7 +29,7 @@
 Name:           python-django
 
 Version:        1.8.14
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        A high-level Python Web framework
 
 Group:          Development/Languages
-- 
1.8.3.1

