From 8145058f9ee3042cbc2214751333b25a9eafedf0 Mon Sep 17 00:00:00 2001
From: Wei Zhou <wei.zhou@windriver.com>
Date: Wed, 11 Oct 2017 22:23:10 -0500
Subject: [PATCH 1/1] meta size number format

Signed-off-by: SidneyAn <ran1.an@intel.com>
---
 SPECS/python-django.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/python-django.spec b/SPECS/python-django.spec
index 79557de..5bef0ec 100644
--- a/SPECS/python-django.spec
+++ b/SPECS/python-django.spec
@@ -32,6 +32,7 @@ Source0:        https://files.pythonhosted.org/packages/source/D/Django/Django-%
 
 # WRS
 Patch0:         session-filebase-backend-fix.patch
+Patch1:         size-number-format.patch
 
 BuildArch:      noarch
 
-- 
1.8.3.1

