From 6cee1f687c1148254fcddc34a7f7de334a26dba1 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Fri, 26 Aug 2016 15:10:15 -0400
Subject: [PATCH 1/1] WRS: spec-include-TiS-patches.patch

---
 SPECS/python-django.spec | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/SPECS/python-django.spec b/SPECS/python-django.spec
index 269991d..d0e54bb 100644
--- a/SPECS/python-django.spec
+++ b/SPECS/python-django.spec
@@ -39,6 +39,9 @@ Source0:        https://files.pythonhosted.org/packages/source/D/Django/Django-%
 
 Patch0:         python-django-1.8.3-shell-completion.patch
 
+# WRS
+Patch1:         session-filebase-backend-fix.patch
+
 BuildArch:      noarch
 BuildRequires:  python2-devel
 
@@ -149,6 +152,9 @@ rm -rf Django.egg-info
 
 %patch0
 
+# WRS
+%patch1 -p1
+
 # empty files
 for f in \
     django/contrib/staticfiles/models.py \
-- 
1.9.1

