From 6cee1f687c1148254fcddc34a7f7de334a26dba1 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Fri, 26 Aug 2016 15:10:15 -0400
Subject: [PATCH 1/1] WRS: spec-include-TiS-patches.patch

Signed-off-by: SidneyAn <ran1.an@intel.com>
---
 SPECS/python-django.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/SPECS/python-django.spec b/SPECS/python-django.spec
index 269991d..d0e54bb 100644
--- a/SPECS/python-django.spec
+++ b/SPECS/python-django.spec
@@ -30,6 +30,9 @@ License:        BSD
 URL:            http://www.djangoproject.com/
 Source0:        https://files.pythonhosted.org/packages/source/D/Django/Django-%{version}.tar.gz
 
+# WRS
+Patch0:         session-filebase-backend-fix.patch
+
 BuildArch:      noarch
 
 
@@ -146,7 +149,7 @@ principle.
 %endif
 
 %prep
-%autosetup -n %{pkgname}-%{version}
+%autosetup -n %{pkgname}-%{version} -p1
 
 
 %build
-- 
1.8.3.1

