From b194e15e5ac26ff762b1c0eb4b3bbca3b618854c Mon Sep 17 00:00:00 2001
Message-Id: <b194e15e5ac26ff762b1c0eb4b3bbca3b618854c.1587850440.git.Jim.Somerville@windriver.com>
From: eric zhang <eric.zhang@windriver.com>
Date: Wed, 14 Mar 2018 14:56:21 -0400
Subject: [PATCH 1/1] Add build dependencies and package versioning for TiS
 format

Signed-off-by: eric zhang <eric.zhang@windriver.com>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 SPECS/rdma-core.spec | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/SPECS/rdma-core.spec b/SPECS/rdma-core.spec
index d040b59..e95d9fc 100644
--- a/SPECS/rdma-core.spec
+++ b/SPECS/rdma-core.spec
@@ -7,6 +7,12 @@
 %endif
 %endif
 
+%if "%{_tis_build_type}" == "rt"
+%define bt_ext -rt
+%else
+%undefine bt_ext
+%endif
+
 %{!?cmake: %global cmake cmake}
 %{!?make_jobs: %global make_jobs make VERBOSE=1 %{?_smp_mflags}}
 %{!?cmake_install: %global cmake_install DESTDIR=%{buildroot} make install}
@@ -35,7 +41,7 @@
 
 Name: rdma-core
 Version: 50mlnx1
-Release: 1%{?dist}.50218
+Release: 1.50218%{?_tis_dist}.%{tis_patch_ver}
 Summary: RDMA core userspace libraries and daemons
 Group: System Environment/Libraries
 
@@ -103,7 +109,9 @@ BuildRequires: python-docutils
 %if 0%{?fedora} >= 21 || 0%{?rhel} >= 8
 BuildRequires: perl-generators
 %endif
-
+BuildRequires: cmake
+BuildRequires: libnl3-devel
+BuildRequires: mlnx-ofa_kernel%{?bt_ext}-devel
 # Red Hat/Fedora previously shipped redhat/ as a stand-alone
 # package called 'rdma', which we're supplanting here.
 Provides: rdma = %{version}-%{release}
-- 
1.8.3.1

