--- a/src/OVAL/probes/probe/icache.c
+++ b/src/OVAL/probes/probe/icache.c
@@ -497,6 +497,7 @@
  */
 static int probe_cobj_memcheck(size_t item_cnt)
 {
+#if !(defined(__FreeBSD__) || defined(__FreeBSD_kernel__))
 	if (item_cnt > PROBE_RESULT_MEMCHECK_CTRESHOLD) {
 		struct proc_memusage mu_proc;
 		struct sys_memusage  mu_sys;
@@ -524,6 +525,7 @@
 			return (1);
 		}
 	}
+#endif
 
 	return (0);
 }
