From 231334d30e9ad3f32dc915f973c71ac18d9c8191 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:11:59 -0400
Subject: [PATCH 05/10] WRS: Fix-VG-activity-bug-in-heartbeat-LVM-script.patch

---
 SPECS/resource-agents.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/resource-agents.spec b/SPECS/resource-agents.spec
index 28b6e50..832d588 100644
--- a/SPECS/resource-agents.spec
+++ b/SPECS/resource-agents.spec
@@ -247,6 +247,7 @@ Patch1111: pgsql.patch
 
 Patch1113: create-var-run-resource-agents.patch
 Patch1114: notify-rmon-of-shutdown-before-shutting-down.patch
+Patch1115: Fix-VG-activity-bug-in-heartbeat-LVM-script.patch
 
 Obsoletes:	heartbeat-resources <= %{version}
 Provides:	heartbeat-resources = %{version}
@@ -551,6 +552,7 @@ exit 1
 
 %patch1113 -p1
 %patch1114 -p1
+%patch1115 -p1
 
 %build
 if [ ! -f configure ]; then
-- 
1.9.1

