From 75bbb081b6cd87f7da59d597e1737f7b4ed395c7 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:11:59 -0400
Subject: [PATCH 05/10] WRS: Fix-VG-activity-bug-in-heartbeat-LVM-script.patch

---
 SPECS/resource-agents.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/resource-agents.spec b/SPECS/resource-agents.spec
index 3d8ece2..9fa7f86 100644
--- a/SPECS/resource-agents.spec
+++ b/SPECS/resource-agents.spec
@@ -274,6 +274,7 @@ Patch1111: pgsql.patch
 
 Patch1113: create-var-run-resource-agents.patch
 Patch1114: notify-rmon-of-shutdown-before-shutting-down.patch
+Patch1115: Fix-VG-activity-bug-in-heartbeat-LVM-script.patch
 
 Obsoletes:	heartbeat-resources <= %{version}
 Provides:	heartbeat-resources = %{version}
@@ -605,6 +606,7 @@ exit 1
 
 %patch1113 -p1
 %patch1114 -p1
+%patch1115 -p1
 
 %build
 if [ ! -f configure ]; then
-- 
2.7.4

