From 72bfa2ad793df8fc68b3b6abe87bd7bfbaf99fb1 Mon Sep 17 00:00:00 2001
From: Bin Qian <bin.qian@windriver.com>
Date: Wed, 29 Aug 2018 11:22:32 -0400
Subject: [PATCH] ipaddr2 avoid failing when svc i/f down

---
 SPECS/resource-agents.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/resource-agents.spec b/SPECS/resource-agents.spec
index 520d9c3..959f4db 100644
--- a/SPECS/resource-agents.spec
+++ b/SPECS/resource-agents.spec
@@ -283,6 +283,7 @@ Patch1117: lvm_cleanup_refs_on_stop.patch
 Patch1118: ipaddr2_if_down.patch
 Patch1119: ipaddr2_ignore_lo_if_state.patch
 Patch1120: Re-enable-background-execution-of-arp-commands.patch
+Patch1121: ipaddr2-avoid-failing-svc-if-down.patch
 
 Obsoletes:	heartbeat-resources <= %{version}
 Provides:	heartbeat-resources = %{version}
@@ -620,6 +621,7 @@ exit 1
 %patch1118 -p1
 %patch1119 -p1
 %patch1120 -p1
+%patch1121 -p1
 
 %build
 if [ ! -f configure ]; then
-- 
2.7.4

