From 00b88829aad297c6732617a706501b466bb9be7a Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Mon, 28 May 2018 14:12:45 -0500
Subject: [PATCH] metapatch for arp_bg

---
 SPECS/resource-agents.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/resource-agents.spec b/SPECS/resource-agents.spec
index c70d20b..520d9c3 100644
--- a/SPECS/resource-agents.spec
+++ b/SPECS/resource-agents.spec
@@ -282,6 +282,7 @@ Patch1116: ocf-shellfuncs_change_logtag.patch
 Patch1117: lvm_cleanup_refs_on_stop.patch
 Patch1118: ipaddr2_if_down.patch
 Patch1119: ipaddr2_ignore_lo_if_state.patch
+Patch1120: Re-enable-background-execution-of-arp-commands.patch
 
 Obsoletes:	heartbeat-resources <= %{version}
 Provides:	heartbeat-resources = %{version}
@@ -618,6 +619,7 @@ exit 1
 %patch1117 -p1
 %patch1118 -p1
 %patch1119 -p1
+%patch1120 -p1
 
 %build
 if [ ! -f configure ]; then
-- 
2.7.4

