From 8c725e6e7a4931066488a74227f902269850a263 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:11:58 -0400
Subject: [PATCH 02/10] WRS: spec-avoid-dir-collisions.patch

---
 SPECS/resource-agents.spec | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/SPECS/resource-agents.spec b/SPECS/resource-agents.spec
index 6c1ec89..7d53666 100644
--- a/SPECS/resource-agents.spec
+++ b/SPECS/resource-agents.spec
@@ -704,14 +704,15 @@ rm -rf %{buildroot}
 %endif
 
 %if %{with linuxha}
-%dir /usr/lib/ocf
-%dir /usr/lib/ocf/resource.d
-%dir /usr/lib/ocf/lib
+#%dir /usr/lib/ocf
+#%dir /usr/lib/ocf/resource.d
+#%dir /usr/lib/ocf/lib
 
-/usr/lib/ocf/lib/heartbeat
+/usr/lib/ocf/lib/heartbeat/*
 
-/usr/lib/ocf/resource.d/heartbeat
-/usr/lib/ocf/resource.d/openstack
+/usr/lib/ocf/resource.d/heartbeat/*
+/usr/lib/ocf/resource.d/heartbeat/.ocf-*
+/usr/lib/ocf/resource.d/openstack/*
 %if %{with rgmanager}
 /usr/lib/ocf/resource.d/redhat
 %endif
@@ -734,7 +735,7 @@ rm -rf %{buildroot}
 
 %{_includedir}/heartbeat
 
-%dir %attr (1755, root, root)	%{_var}/run/resource-agents
+#%dir %attr (1755, root, root)	%{_var}/run/resource-agents
 
 %{_mandir}/man7/*.7*
 
@@ -845,7 +846,7 @@ rm -rf %{buildroot}
 %exclude %{_mandir}/man8/ldirectord.8.gz
 
 # For compatability with pre-existing agents
-%dir %{_sysconfdir}/ha.d
+#%dir %{_sysconfdir}/ha.d
 %{_sysconfdir}/ha.d/shellfuncs
 
 %{_libexecdir}/heartbeat
-- 
2.7.4

