From d8db42cf6aac88403c9f9101230061864a96eec0 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:11:59 -0400
Subject: [PATCH 07/10] WRS: spec-lvm-cleanup-refs-on-stop.patch

---
 SPECS/resource-agents.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/resource-agents.spec b/SPECS/resource-agents.spec
index c325197..21fa049 100644
--- a/SPECS/resource-agents.spec
+++ b/SPECS/resource-agents.spec
@@ -276,6 +276,7 @@ Patch1113: create-var-run-resource-agents.patch
 Patch1114: notify-rmon-of-shutdown-before-shutting-down.patch
 Patch1115: Fix-VG-activity-bug-in-heartbeat-LVM-script.patch
 Patch1116: ocf-shellfuncs_change_logtag.patch
+Patch1117: lvm_cleanup_refs_on_stop.patch
 
 Obsoletes:	heartbeat-resources <= %{version}
 Provides:	heartbeat-resources = %{version}
@@ -609,6 +610,7 @@ exit 1
 %patch1114 -p1
 %patch1115 -p1
 %patch1116 -p1
+%patch1117 -p1
 
 %build
 if [ ! -f configure ]; then
-- 
2.7.4

