From d5796dce7ea3bc1a90a3faed3b7acff36583c5e0 Mon Sep 17 00:00:00 2001
From: Bart Wensley <barton.wensley@windriver.com>
Date: Wed, 21 Nov 2018 12:21:41 -0600
Subject: [PATCH 1/1] spec ipaddr2 use host scope for addresses on loopback

---
 SPECS/resource-agents.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/resource-agents.spec b/SPECS/resource-agents.spec
index 959f4db..a98064a 100644
--- a/SPECS/resource-agents.spec
+++ b/SPECS/resource-agents.spec
@@ -284,6 +284,7 @@ Patch1118: ipaddr2_if_down.patch
 Patch1119: ipaddr2_ignore_lo_if_state.patch
 Patch1120: Re-enable-background-execution-of-arp-commands.patch
 Patch1121: ipaddr2-avoid-failing-svc-if-down.patch
+Patch1122: ipaddr2-use-host-scope-for-addresses-on-loopback.patch
 
 Obsoletes:	heartbeat-resources <= %{version}
 Provides:	heartbeat-resources = %{version}
@@ -622,6 +623,7 @@ exit 1
 %patch1119 -p1
 %patch1120 -p1
 %patch1121 -p1
+%patch1122 -p1
 
 %build
 if [ ! -f configure ]; then
-- 
1.8.3.1

