From 774432f5665f102f7de7b35be0cb753964c4905c Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:11:59 -0400
Subject: [PATCH 03/10] WRS: spec-add-create-var-run-resource-agents.patch

---
 SPECS/resource-agents.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/resource-agents.spec b/SPECS/resource-agents.spec
index 7d53666..0725e97 100644
--- a/SPECS/resource-agents.spec
+++ b/SPECS/resource-agents.spec
@@ -272,6 +272,7 @@ Patch1109:  umount-in-namespace.patch
 Patch1110: lvm_vg_activation.patch
 Patch1111: pgsql.patch
 
+Patch1113: create-var-run-resource-agents.patch
 
 Obsoletes:	heartbeat-resources <= %{version}
 Provides:	heartbeat-resources = %{version}
@@ -601,6 +602,7 @@ exit 1
 %patch1110 -p1
 %patch1111 -p1
 
+%patch1113 -p1
 
 %build
 if [ ! -f configure ]; then
-- 
2.7.4

