From 1f6128eb7bf5287db6def45f278acf771a711a42 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:11:58 -0400
Subject: [PATCH] WRS: spec-include-TiS-patches.patch

---
 SPECS/resource-agents.spec | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/SPECS/resource-agents.spec b/SPECS/resource-agents.spec
index a16660a..6c1ec89 100644
--- a/SPECS/resource-agents.spec
+++ b/SPECS/resource-agents.spec
@@ -263,6 +263,16 @@ Patch202:	bz1536548-sap_redhat_cluster_connector-fix-unknown-gvi-function.patch
 Patch203:	bz1543366-redis-add-support-for-tunneling-replication-traffic.patch
 Patch204:	bz1546083-galera-fix-temp-logfile-rights.patch
 
+# WRS
+Patch1105:  filesystem_rmon.patch
+Patch1106:  new_ocf_return_codes.patch
+Patch1107:  ipaddr2_check_if_state.patch
+Patch1108:  copyright.patch
+Patch1109:  umount-in-namespace.patch
+Patch1110: lvm_vg_activation.patch
+Patch1111: pgsql.patch
+
+
 Obsoletes:	heartbeat-resources <= %{version}
 Provides:	heartbeat-resources = %{version}
 
@@ -582,6 +592,16 @@ exit 1
 %patch203 -p1
 %patch204 -p1
 
+# WRS
+%patch1105 -p1
+%patch1106 -p1
+%patch1107 -p1
+%patch1108 -p1
+%patch1109 -p1
+%patch1110 -p1
+%patch1111 -p1
+
+
 %build
 if [ ! -f configure ]; then
 	./autogen.sh
-- 
2.7.4

