From a51920cbcd728da7d867a78daf1dc78e5e3d7598 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:05 -0400
Subject: WRS: dhclient-disable-NSUPDATE.patch

---
 SPECS/dhcp.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/dhcp.spec b/SPECS/dhcp.spec
index fc17edb..b0a385e 100644
--- a/SPECS/dhcp.spec
+++ b/SPECS/dhcp.spec
@@ -118,6 +118,7 @@ Patch103: dhclient-ipv6-bind-to-interface.patch
 Patch104: dhclient-ipv6-conditionally-set-hostname.patch
 Patch105: dhclient-handle-wrs-install-uuid.patch
 Patch106: dhclient-dhcp6-wrs-install-uuid.patch
+Patch107: dhclient-disable-NSUPDATE.patch        
 
 BuildRequires: autoconf
 BuildRequires: automake
@@ -454,6 +455,7 @@ rm -rf includes/isc-dhcp
 %patch104 -p1
 %patch105 -p1
 %patch106 -p1
+%patch107 -p1
 
 # Update paths in all man pages
 for page in client/dhclient.conf.5 client/dhclient.leases.5 \
@@ -475,6 +477,7 @@ done
 #libtoolize --copy --force
 autoreconf --verbose --force --install
 
+# WR: failover and tracing need to be disabled in order to disable NSUPDATE
 CFLAGS="%{optflags} -fno-strict-aliasing" \
 %configure \
     --with-srv-lease-file=%{_localstatedir}/lib/dhcpd/dhcpd.leases \
@@ -489,6 +492,7 @@ CFLAGS="%{optflags} -fno-strict-aliasing" \
     --with-ldap \
     --with-ldapcrypto \
     --with-libbind=%{_includedir} --with-libbind-libs=%{_libdir} \
+    --disable-tracing --disable-failover \
     --disable-static \
 %if %sdt
     --enable-systemtap \
-- 
2.7.4

