From fd8de766f09dd6cb836b7f9823eaa0ad17eaf2d3 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:05 -0400
Subject: WRS: mark-dhclient.conf-as-config.patch

---
 SPECS/dhcp.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/dhcp.spec b/SPECS/dhcp.spec
index 70c7a6d..fc17edb 100644
--- a/SPECS/dhcp.spec
+++ b/SPECS/dhcp.spec
@@ -722,7 +722,7 @@ done
 %attr(0644,root,root) %{_mandir}/man8/dhclient-script.8.gz
 
 # WRS
-%{dhcpconfdir}/dhclient.conf
+%config(noreplace) %{dhcpconfdir}/dhclient.conf
 %{dhcpconfdir}/dhclient-enter-hooks
 
 %files common
-- 
2.7.4

