From 238c53632f948d293cbcfb4c92fba29096808080 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:05 -0400
Subject: WRS: remove-unecessary-dhcp-exit-hooks-file.patch

---
 SPECS/dhcp.spec | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/SPECS/dhcp.spec b/SPECS/dhcp.spec
index fd66dc4..42409f6 100644
--- a/SPECS/dhcp.spec
+++ b/SPECS/dhcp.spec
@@ -40,10 +40,10 @@ Source9: azure-cloud.sh
 Source10: README.scripts
 
 # WRS
-Source20: dhclient-exit-hooks
 Source21: dhclient.conf
 Source22: dhclient-enter-hooks
 
+
 Patch0:   dhcp-4.2.0-errwarn-message.patch
 Patch1:   dhcp-4.2.4-dhclient-options.patch
 Patch2:   dhcp-4.2.0-release-by-ifup.patch
@@ -603,7 +603,6 @@ EOF
 find ${RPM_BUILD_ROOT}/%{_libdir} -name '*.la' -exec '/bin/rm' '-f' '{}' ';';
 
 # WRS: Overwrite with our config
-%{__install} -p -m 0755 %{SOURCE20} %{buildroot}%{dhcpconfdir}/dhclient-exit-hooks
 %{__install} -p -m 0644 %{SOURCE21} %{buildroot}%{dhcpconfdir}/dhclient.conf
 %{__install} -p -m 0755 %{SOURCE22} %{buildroot}%{dhcpconfdir}/dhclient-enter-hooks
 
@@ -723,7 +722,6 @@ done
 %attr(0644,root,root) %{_mandir}/man8/dhclient-script.8.gz
 
 # WRS
-%{dhcpconfdir}/dhclient-exit-hooks
 %{dhcpconfdir}/dhclient.conf
 %{dhcpconfdir}/dhclient-enter-hooks
 
-- 
2.7.4

