From 79b8e2f34a5746e43e64e375c4fbf236de064c63 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:05 -0400
Subject: WRS: spec-dhcp-enter-hooks.patch

---
 SPECS/dhcp.spec | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/SPECS/dhcp.spec b/SPECS/dhcp.spec
index 904e3ad..fd66dc4 100644
--- a/SPECS/dhcp.spec
+++ b/SPECS/dhcp.spec
@@ -42,7 +42,7 @@ Source10: README.scripts
 # WRS
 Source20: dhclient-exit-hooks
 Source21: dhclient.conf
-
+Source22: dhclient-enter-hooks
 
 Patch0:   dhcp-4.2.0-errwarn-message.patch
 Patch1:   dhcp-4.2.4-dhclient-options.patch
@@ -605,6 +605,7 @@ find ${RPM_BUILD_ROOT}/%{_libdir} -name '*.la' -exec '/bin/rm' '-f' '{}' ';';
 # WRS: Overwrite with our config
 %{__install} -p -m 0755 %{SOURCE20} %{buildroot}%{dhcpconfdir}/dhclient-exit-hooks
 %{__install} -p -m 0644 %{SOURCE21} %{buildroot}%{dhcpconfdir}/dhclient.conf
+%{__install} -p -m 0755 %{SOURCE22} %{buildroot}%{dhcpconfdir}/dhclient-enter-hooks
 
 %pre
 # /usr/share/doc/setup/uidgid
@@ -724,6 +725,7 @@ done
 # WRS
 %{dhcpconfdir}/dhclient-exit-hooks
 %{dhcpconfdir}/dhclient.conf
+%{dhcpconfdir}/dhclient-enter-hooks
 
 %files common
 %doc LICENSE README RELNOTES doc/References.txt
-- 
2.7.4

