From 7e7a9f1bce5884da1e57d5bdc4f5215b7231924e Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:05 -0400
Subject: [PATCH] WRS: spec-include-TiS-patches.patch

---
 SPECS/dhcp.spec | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/SPECS/dhcp.spec b/SPECS/dhcp.spec
index 14da097..904e3ad 100644
--- a/SPECS/dhcp.spec
+++ b/SPECS/dhcp.spec
@@ -111,6 +115,11 @@ Patch70:  dhcp-4.2.5-reference_count_overflow.patch
 Patch71:  dhcp-4.2.5-centos-branding.patch


+# WRS
+Patch101: dhclient-restrict-interfaces-to-command-line.patch
+Patch102: dhclient-ipv6-bind-to-interface.patch
+Patch103: dhclient-ipv6-conditionally-set-hostname.patch
+
 BuildRequires: autoconf
 BuildRequires: automake
 BuildRequires: libtool
@@ -439,6 +451,11 @@ rm -rf includes/isc-dhcp
 %patch70 -p1 -b .reference_overflow
 %patch71 -p1

+# WRS
+%patch101 -p1
+%patch102 -p1
+%patch103 -p1
+
 # Update paths in all man pages
 for page in client/dhclient.conf.5 client/dhclient.leases.5 \
             client/dhclient-script.8 client/dhclient.8 ; do
--
2.7.4

