From 7e7a9f1bce5884da1e57d5bdc4f5215b7231924e Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:05 -0400
Subject: [PATCH]: spec-include-TiS-patches.patch

---
 SPECS/dhcp.spec | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/SPECS/dhcp.spec b/SPECS/dhcp.spec
index 7bbd360..4bfbf5c 100644
--- a/SPECS/dhcp.spec
+++ b/SPECS/dhcp.spec
@@ -80,6 +80,10 @@ Patch43:  dhcp-4.3.6-reference_count_overflow.patch
 Patch44:  dhcp-replay_file_limit.patch
 Patch45:  dhcp-4.2.5-expiry_before_renewal_v2.patch

+Patch101: dhclient-restrict-interfaces-to-command-line.patch
+Patch102: dhclient-ipv6-bind-to-interface.patch
+Patch103: dhclient-ipv6-conditionally-set-hostname.patch
+
 BuildRequires: autoconf
 BuildRequires: automake
 BuildRequires: libtool
@@ -344,6 +348,10 @@ rm bind/bind.tar.gz
 # https://bugzilla.redhat.com/show_bug.cgi?id=1647786
 %patch45 -p1 -b .t2-expirity

+%patch101 -p1
+%patch102 -p1
+%patch103 -p1
+
 # DHCLIENT_DEFAULT_PREFIX_LEN  64 -> 128
 # https://bugzilla.gnome.org/show_bug.cgi?id=656610
 sed -i -e 's|DHCLIENT_DEFAULT_PREFIX_LEN 64|DHCLIENT_DEFAULT_PREFIX_LEN 128|g' includes/site.h
-- 
2.7.4

