From 869e94c0bb51471063fab9fa6bdff54f1db8407e Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Tue, 7 Nov 2017 15:25:12 -0500
Subject: [PATCH] Update-package-versioning-for-TIS-format

---
 SPECS/cloud-init.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/cloud-init.spec b/SPECS/cloud-init.spec
index 89691ae..935c970 100644
--- a/SPECS/cloud-init.spec
+++ b/SPECS/cloud-init.spec
@@ -7,7 +7,7 @@
 
 Name:           cloud-init
 Version:        0.7.9
-Release:        9%{?dist}.2
+Release:        9.el7.centos.2%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Cloud instance init scripts
 
 Group:          System Environment/Base
-- 
1.8.3.1

