From 9a97a6ade7470e525b2013b85782b887a96b7489 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Tue, 7 Nov 2017 15:25:12 -0500
Subject: [PATCH] Update-package-versioning-for-TIS-format

---
 SPECS/cloud-init.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/cloud-init.spec b/SPECS/cloud-init.spec
index 390ed96..ef82dd1 100644
--- a/SPECS/cloud-init.spec
+++ b/SPECS/cloud-init.spec
@@ -7,7 +7,7 @@
 
 Name:           cloud-init
 Version:        0.7.9
-Release:        24%{?dist}.1
+Release:        24.el7.centos.1%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Cloud instance init scripts
 
 Group:          System Environment/Base
-- 
2.7.4

