From 1fa41f1116464169399c683abd9920c7bbf28a5b Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Wed, 8 Nov 2017 11:13:02 -0500
Subject: [PATCH] spec-include-tis-changes

---
 SPECS/cloud-init.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/cloud-init.spec b/SPECS/cloud-init.spec
index d7dc778..1842e08 100644
--- a/SPECS/cloud-init.spec
+++ b/SPECS/cloud-init.spec
@@ -42,6 +42,8 @@ Patch18: ci-cc_mounts-check-if-mount-a-on-no-change-fstab-path.patch
 # For bz#1648375 - [Azure] [RHEL 8] Cloud-init fixes to support fast provisioning for Azure[8.0.1]
 Patch19: ci-Azure-Return-static-fallback-address-as-if-failed-to.patch
 
+Patch1000: cloud-init-interactive-parted.patch
+
 BuildArch:      noarch
 
 BuildRequires:  pkgconfig(systemd)
-- 
2.7.4

