From 1fa41f1116464169399c683abd9920c7bbf28a5b Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Wed, 8 Nov 2017 11:13:02 -0500
Subject: [PATCH] spec-include-tis-changes

---
 SPECS/cloud-init.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/SPECS/cloud-init.spec b/SPECS/cloud-init.spec
index ef82dd1..022e779 100644
--- a/SPECS/cloud-init.spec
+++ b/SPECS/cloud-init.spec
@@ -69,7 +69,10 @@ Patch41: ci-DataSourceAzure.py-use-hostnamectl-to-set-hostname.patch
 # For bz#1578702 - cloud-init-0.7.9-9.el7_4.6 breaks IPv4/IPv6 dual-stack EC2 instances in AWS [rhel-7.5.z]
 Patch42: ci-sysconfig-Don-t-disable-IPV6_AUTOCONF.patch
 
-Patch9999: cloud-init-add-centos-os.patch
+Patch0999: cloud-init-add-centos-os.patch
+
+# WRS patches
+Patch2000: cloud-init-interactive-parted.patch
 
 # Deal with noarch -> arch
 # https://bugzilla.redhat.com/show_bug.cgi?id=1067089
-- 
2.7.4

