From 1fa41f1116464169399c683abd9920c7bbf28a5b Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Wed, 8 Nov 2017 11:13:02 -0500
Subject: [PATCH] spec-include-tis-changes

---
 SPECS/cloud-init.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/cloud-init.spec b/SPECS/cloud-init.spec
index a873d81..1175601 100644
--- a/SPECS/cloud-init.spec
+++ b/SPECS/cloud-init.spec
@@ -38,6 +38,8 @@ Patch15: ci-Fix-string-missmatch-when-mounting-ntfs.patch
 # For bz#1602784 - cloud-init: Sometimes image boots fingerprints is configured, there's a network device present but it's not configured
 Patch16: ci-net-Make-sysconfig-renderer-compatible-with-Network-.patch
 
+Patch1000: cloud-init-interactive-parted.patch
+
 BuildArch:      noarch
 
 BuildRequires:  pkgconfig(systemd)
-- 
2.7.4

