From 0fcc5269a8ee26e7aee8301c13f740dff9c2acee Mon Sep 17 00:00:00 2001
From: Kam Nasim <kam.nasim@windriver.com>
Date: Wed, 25 Oct 2017 17:48:23 -0400
Subject: [PATCH] squash titanium patches

---
 SPECS/puppet-keystone.spec | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/SPECS/puppet-keystone.spec b/SPECS/puppet-keystone.spec
index 956664d..f0a18e8 100644
--- a/SPECS/puppet-keystone.spec
+++ b/SPECS/puppet-keystone.spec
@@ -9,6 +9,9 @@ URL:            https://launchpad.net/puppet-keystone
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 
+# WRS Patches
+Patch0001:      0001-pike-rebase-squash-titanium-patches.patch
+
 BuildArch:      noarch
 
 Requires:       puppet-apache
@@ -24,6 +27,9 @@ Puppet module for OpenStack Keystone
 %prep
 %setup -q -n openstack-keystone-%{upstream_version}
 
+# WRS Patches
+%patch0001 -p1
+
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
 find . \( -name "*.pl" -o -name "*.sh"  \) -exec chmod +x {} +
-- 
1.8.3.1

