commit a1122168b836f9c786c4a6c58c96dde548921a06
Author: Shoaib Nasir <shoaib.nasir@windriver.com>
Date:   Thu Feb 15 15:15:33 2018 -0500

    WRS: 0004-remove-eventlet_and_bindhost-from-keystoneconf.patch

diff --git a/SPECS/puppet-keystone.spec b/SPECS/puppet-keystone.spec
index 42352df..36cf461 100644
--- a/SPECS/puppet-keystone.spec
+++ b/SPECS/puppet-keystone.spec
@@ -12,6 +12,7 @@ Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_versio
 # WRS Patches
 Patch0001:      0001-pike-rebase-squash-titanium-patches.patch
 Patch0002:      0002-remove-the-Keystone-admin-app.patch
+Patch0003:      0003-remove-eventlet_bindhost-from-Keystoneconf.patch
 
 BuildArch:      noarch
 
@@ -31,6 +32,7 @@ Puppet module for OpenStack Keystone
 # WRS Patches
 %patch0001 -p1
 %patch0002 -p1
+%patch0003 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
