From a53d9bfc15e1d24a604fcc461f9a6e1483ed262d Mon Sep 17 00:00:00 2001
From: Andy Ning <andy.ning@windriver.com>
Date: Mon, 4 Nov 2019 20:51:22 -0500
Subject: [PATCH 1/1] update Barbican admin secret's user/project IDs

Signed-off-by: Andy Ning <andy.ning@windriver.com>
---
 SPECS/puppet-keystone.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-keystone.spec b/SPECS/puppet-keystone.spec
index 5b5e255..ae233f0 100644
--- a/SPECS/puppet-keystone.spec
+++ b/SPECS/puppet-keystone.spec
@@ -15,6 +15,7 @@ Patch0002:      0002-remove-the-Keystone-admin-app.patch
 Patch0003:      0003-remove-eventlet_bindhost-from-Keystoneconf.patch
 Patch0004:      0004-escape-special-characters-in-bootstrap.patch
 Patch0005:      0005-Add-support-for-fernet-receipts.patch
+Patch0006:      0006-update-Barbican-admin-secret-s-user-project-IDs-duri.patch

 BuildArch:      noarch

@@ -37,6 +38,7 @@ Puppet module for OpenStack Keystone
 %patch0003 -p1
 %patch0004 -p1
 %patch0005 -p1
+%patch0006 -p1

 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
--
1.8.3.1

