From: Ceph Packaging Team <team+ceph@tracker.debian.org>
Date: Thu, 11 Sep 2025 21:04:53 +0000
Subject: add-option-to-disable-ceph-dencoder

===================================================================
---
 src/tools/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/tools/CMakeLists.txt b/src/tools/CMakeLists.txt
index fc8539f..5a34a20 100644
--- a/src/tools/CMakeLists.txt
+++ b/src/tools/CMakeLists.txt
@@ -126,4 +126,6 @@ if(WITH_RBD)
   endif()
 endif(WITH_RBD)
 
+if(NOT DISABLE_DENCODER)
 add_subdirectory(ceph-dencoder)
+endif(DISABLE_DENCODER)
