Index: cloud-init-0.7.4/cloudinit/config/cc_growpart.py
===================================================================
--- cloud-init-0.7.4.orig/cloudinit/config/cc_growpart.py
+++ cloud-init-0.7.4/cloudinit/config/cc_growpart.py
@@ -97,7 +97,7 @@ class ResizeParted(object):
     def resize(self, diskdev, partnum, partdev):
         before = get_size(partdev)
         try:
-            util.subp(["parted", diskdev, "resizepart", partnum])
+            util.subp(["resizepart.sh", diskdev, partnum])
         except util.ProcessExecutionError as e:
             raise ResizeFailedException(e)
 
