From a6c86c1badb9cebb44ffbb53a8ff5d5529b12fd5 Mon Sep 17 00:00:00 2001
From: Matt Peters <matt.peters@windriver.com>
Date: Fri, 3 Feb 2017 16:06:51 -0500
Subject: [PATCH 1/1] e2fsprogs disable tests

---
 Makefile.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index f327d19..7faca16 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -16,8 +16,8 @@ INSTALL = @INSTALL@
 QUOTA_LIB_SUBDIR= lib/quota
 
 LIB_SUBDIRS=lib/et lib/ss lib/e2p $(UUID_LIB_SUBDIR) lib/ext2fs $(BLKID_LIB_SUBDIR) $(QUOTA_LIB_SUBDIR) intl
-PROG_SUBDIRS=e2fsck $(DEBUGFS_DIR) misc $(RESIZE_DIR) tests/progs po
-SUBDIRS=util $(LIB_SUBDIRS) $(PROG_SUBDIRS) tests
+PROG_SUBDIRS=e2fsck $(DEBUGFS_DIR) misc $(RESIZE_DIR) po
+SUBDIRS=util $(LIB_SUBDIRS) $(PROG_SUBDIRS)
 
 SUBS= util/subst.conf lib/config.h lib/dirpaths.h \
 	lib/ext2fs/ext2_types.h lib/blkid/blkid_types.h lib/uuid/uuid_types.h
-- 
1.8.3.1

