From d52fda6215af4f2d51884a10c04d3c7a44d100dd Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 13 Nov 2017 16:38:15 -0500
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/anaconda.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/anaconda.spec b/SPECS/anaconda.spec
index 69a55d2..2848ba3 100644
--- a/SPECS/anaconda.spec
+++ b/SPECS/anaconda.spec
@@ -7,7 +7,7 @@
 Summary: Graphical system installer
 Name:    anaconda
 Version: 29.19.1.13
-Release: 1%{?dist}
+Release: 1.el8%{?_tis_dist}.%{tis_patch_ver}
 License: GPLv2+ and MIT
 Group:   Applications/System
 URL:     http://fedoraproject.org/wiki/Anaconda
-- 
2.7.4

