From 6bec7d96120f7eef019ab2841265bf4b74ebc64d Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 13 Nov 2017 17:22:49 -0500
Subject: [PATCH] Add TIS patches

---
 SPECS/anaconda.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/anaconda.spec b/SPECS/anaconda.spec
index b345630..41de025 100644
--- a/SPECS/anaconda.spec
+++ b/SPECS/anaconda.spec
@@ -15,6 +15,9 @@ URL:     http://fedoraproject.org/wiki/Anaconda
 Patch100: centos-l10n.patch
 Patch101: 0001-branding-syspurpose-doesn-t-make-sense-for-CentOS.patch
 Patch102: 0002-add-centos-background-color.patch
+
+Patch10001: 0001-TIS-Progress-and-error-handling.patch
+
 # To generate Source0 do:
 # git clone https://github.com/rhinstaller/anaconda
 # git checkout -b archive-branch anaconda-%%{version}-%%{release}
-- 
2.7.4

