From 9ebc2f9343cc214fb1e590221e4791f10a2f87d1 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Fri, 17 Nov 2017 12:08:27 -0500
Subject: [PATCH] revert 7.4 grub2 efi handling

---
 SPECS/anaconda.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/anaconda.spec b/SPECS/anaconda.spec
index 79e1c55..2e5ece9 100644
--- a/SPECS/anaconda.spec
+++ b/SPECS/anaconda.spec
@@ -26,6 +26,7 @@ Patch10: anaconda-centos-armhfp-extloader.patch
 
 # WRS
 Patch10001: 0001-TIS-Progress-and-error-handling.patch
+Patch10002: 0002-revert-7.4-grub2-efi-handling.patch
 
 # Versions of required components (done so we make sure the buildrequires
 # match the requires versions of things).
@@ -255,6 +256,7 @@ runtime on NFS/HTTP/FTP servers or local disks.
 
 # WRS
 %patch10001 -p1
+%patch10002 -p1
 
 %build
 %configure --disable-static \
-- 
1.8.3.1

