From e54422230c27e53436fe94a639a04aaf65f787e1 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Thu, 23 Nov 2017 16:50:10 -0500
Subject: [PATCH] Add TIS patches for host lookup

---
 SPECS/anaconda.spec | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/SPECS/anaconda.spec b/SPECS/anaconda.spec
index 174dbee..8541334 100644
--- a/SPECS/anaconda.spec
+++ b/SPECS/anaconda.spec
@@ -26,6 +26,8 @@ Patch1000:	yumpayload-dont-verify-disabled-repos.patch
 # WRS
 Patch10001: 0001-TIS-Progress-and-error-handling.patch
 Patch10002: 0002-revert-7.4-grub2-efi-handling.patch
+Patch10003: 0003-Set-default-hostname-to-localhost.patch
+Patch10004: 0004-Cache-server-ip-in-etc-hosts.patch
 
 # Versions of required components (done so we make sure the buildrequires
 # match the requires versions of things).
@@ -250,6 +252,8 @@ runtime on NFS/HTTP/FTP servers or local disks.
 # WRS
 %patch10001 -p1
 %patch10002 -p1
+%patch10003 -p1
+%patch10004 -p1
 
 %build
 %configure --disable-static \
@@ -268,6 +272,10 @@ desktop-file-install ---dir=%{buildroot}%{_datadir}/applications %{buildroot}%{_
 mkdir -p %{buildroot}%{_datadir}/anaconda/site-python
 install -m 0644 pyanaconda/sitecustomize.py %{buildroot}%{_datadir}/anaconda/site-python/
 %endif
+
+# Add anaconda-preexec script
+install -m 0755 scripts/anaconda-preexec %{buildroot}%{_sbindir}/anaconda-preexec
+
 # NOTE: If you see "error: Installed (but unpackaged) file(s) found" that include liveinst files,
 #       check the IS_LIVEINST_ARCH in configure.ac to make sure your architecture is properly defined
 
@@ -316,6 +324,7 @@ update-desktop-database &> /dev/null || :
 %{_sysconfdir}/X11/xinit/xinitrc.d/*
 %{_datadir}/applications/*.desktop
 %endif
+%{_sbindir}/anaconda-preexec
 
 %files gui
 %{_libdir}/python*/site-packages/pyanaconda/ui/gui/*
-- 
1.8.3.1

