From cf5d8b554d6fdacf3ad3d18333bd00f8b937ff54 Mon Sep 17 00:00:00 2001
Message-Id: <cf5d8b554d6fdacf3ad3d18333bd00f8b937ff54.1507754332.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Wed, 11 Oct 2017 16:38:36 -0400
Subject: [PATCH 1/1] Changes for CentOS 7.4 support

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 kcompat.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/kcompat.h b/kcompat.h
index 59e32a8..3d4e8f6 100644
--- a/kcompat.h
+++ b/kcompat.h
@@ -31,9 +31,5 @@
 
 #define __GFP_RECLAIM __GFP_WAIT
 
-#define inode_lock(_node) mutex_lock(&_node->i_mutex)
-#define inode_unlock(_node) mutex_unlock(&_node->i_mutex)
-
-
 #endif
 #endif
-- 
1.8.3.1

