From b575f4c50e8726c5f9b3227b37a4517c0bbde85c Mon Sep 17 00:00:00 2001
From: Robert Church <robert.church@windriver.com>
Date: Fri, 2 Jun 2017 02:15:19 +0000
Subject: [PATCH] Add PausedSync states to acceptable cstate to avoid metdata
 creation

---
 manifests/resource/up.pp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/manifests/resource/up.pp b/manifests/resource/up.pp
index 160c8c2..3e2fdac 100644
--- a/manifests/resource/up.pp
+++ b/manifests/resource/up.pp
@@ -14,7 +14,7 @@ define drbd::resource::up (
   exec { "initialize DRBD metadata for ${name}":
     command => "yes yes | drbdadm create-md ${name}",
     onlyif  => "test -e ${disk}",
-    unless  => "drbdadm dump-md ${name} || (drbdadm cstate ${name} | egrep -q '^(Sync|Connected|WFConnection|StandAlone|Verify)') || (drbdadm show-gi ${name} | grep 'meta-data: need apply-al')",
+    unless  => "drbdadm dump-md ${name} || (drbdadm cstate ${name} | egrep -q '^(PausedSync|Sync|Connected|WFConnection|StandAlone|Verify)') || (drbdadm show-gi ${name} | grep 'meta-data: need apply-al')",
     before  => Service['drbd'],
     require => [
       Exec['modprobe drbd'],
-- 
1.8.3.1

