From f964faeb50150b64ed2481f721e544f4c3e2ca55 Mon Sep 17 00:00:00 2001
From: Al Bailey <al.bailey@windriver.com>
Date: Tue, 29 Nov 2016 15:06:32 -0500
Subject: [PATCH 1/1] meta patch0

---
 SPECS/python-psycopg2.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/python-psycopg2.spec b/SPECS/python-psycopg2.spec
index 2489fc5..c11a680 100644
--- a/SPECS/python-psycopg2.spec
+++ b/SPECS/python-psycopg2.spec
@@ -36,6 +36,8 @@ Url:		http://www.psycopg.org/psycopg/
 
 Source0:	http://www.psycopg.org/psycopg/tarballs/PSYCOPG-2-5/psycopg2-%{version}.tar.gz
 
+Patch0:         Manually-patch-170-173-187-and-187-fix.patch
+
 BuildRequires:	postgresql-devel
 BuildRequires:	python-devel
 BuildRequires:	python-debug
@@ -103,6 +105,7 @@ Zope Database Adapter for PostgreSQL, called ZPsycopgDA
 
 %prep
 %setup -q -n psycopg2-%{version}
+%patch0 -p1
 
 %build
 for python in %{python_runtimes} ; do
-- 
1.8.3.1

