From 78314a11e44751d23da70c5b935c73579650a2bd Mon Sep 17 00:00:00 2001
From: Al Bailey <al.bailey@windriver.com>
Date: Tue, 29 Nov 2016 14:29:21 -0500
Subject: [PATCH 1/1] Setup spec for Titanium Cloud

---
 SPECS/python-psycopg2.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/python-psycopg2.spec b/SPECS/python-psycopg2.spec
index 8492914..2489fc5 100644
--- a/SPECS/python-psycopg2.spec
+++ b/SPECS/python-psycopg2.spec
@@ -28,7 +28,7 @@
 Summary:	A PostgreSQL database adapter for Python
 Name:		python-psycopg2
 Version:	2.5.1
-Release:	3%{?dist}
+Release:	3.el7%{?_tis_dist}.%{tis_patch_ver}
 # The exceptions allow linking to OpenSSL and PostgreSQL's libpq
 License:	LGPLv3+ with exceptions
 Group:		Applications/Databases
-- 
1.8.3.1

