From 6f3c7142899b780c1beeb1634c1265d2fef0e88c Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Fri, 3 Nov 2017 12:21:53 -0400
Subject: [PATCH 2/4] Roll up TIS patches

---
 SPECS/puppet-nova.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/puppet-nova.spec b/SPECS/puppet-nova.spec
index 4dd32c5..e209d03 100644
--- a/SPECS/puppet-nova.spec
+++ b/SPECS/puppet-nova.spec
@@ -9,6 +9,8 @@ URL:            https://launchpad.net/puppet-nova
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 
+Patch0001:      0001-Roll-up-TIS-patches.patch
+
 BuildArch:      noarch
 
 Requires:       puppet-cinder
@@ -28,6 +30,8 @@ Puppet module for OpenStack Nova
 %prep
 %setup -q -n openstack-nova-%{upstream_version}
 
+%patch0001 -p1
+
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
 find . \( -name "*.pl" -o -name "*.sh"  \) -exec chmod +x {} +
-- 
1.8.3.1

