From 2909066051cfd5f4813d2d431b0bbea44e8022f0 Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Fri, 3 Nov 2017 12:24:53 -0400
Subject: [PATCH 4/4] Setup simple cell

---
 SPECS/puppet-nova.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/puppet-nova.spec b/SPECS/puppet-nova.spec
index 9bd97b0..f20b9a5 100644
--- a/SPECS/puppet-nova.spec
+++ b/SPECS/puppet-nova.spec
@@ -11,6 +11,8 @@ Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_versio
 
 Patch0001:      0001-Roll-up-TIS-patches.patch
 Patch0002:      0002-upstreamable-fix-syntax-error-for-until_complete_rea.patch
+Patch0003:      0003-Create-nova_cell0-database.patch
+Patch0004:      0004-Filter-out-warnings-in-nova-manage-cell_v2-list_cell.patch
 
 BuildArch:      noarch
 
@@ -33,6 +35,8 @@ Puppet module for OpenStack Nova
 
 %patch0001 -p1
 %patch0002 -p1
+%patch0003 -p1
+%patch0004 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

