commit 0aefcfec6930ed8fe4752c6a3ef2047fb7f0a052
Author: Shoaib Nasir <shoaib.nasir@windriver.com>
Date:   Wed Dec 6 19:15:34 2017 -0500

    Create the nova-ironic.conf patch

diff --git a/SPECS/puppet-nova.spec b/SPECS/puppet-nova.spec
index b7d70b5..2271824 100644
--- a/SPECS/puppet-nova.spec
+++ b/SPECS/puppet-nova.spec
@@ -14,6 +14,7 @@ Patch0002:      0002-upstreamable-fix-syntax-error-for-until_complete_rea.patch
 Patch0003:      0003-Create-nova_cell0-database.patch
 Patch0004:      0004-Filter-out-warnings-in-nova-manage-cell_v2-list_cell.patch
 Patch0005:      0005-fix-pci-and-compute-pci.patch
+Patch0006:      0006-Create-Nova-ironic-conf.patch
 
 BuildArch:      noarch
 
@@ -39,6 +40,7 @@ Puppet module for OpenStack Nova
 %patch0003 -p1
 %patch0004 -p1
 %patch0005 -p1
+%patch0006 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
