From 90d515932132d54d5a3ce7091dcd16f537f24d59 Mon Sep 17 00:00:00 2001
From: Joseph Richard <Joseph.Richard@windriver.com>
Date: Fri, 10 Nov 2017 11:31:49 -0500
Subject: [PATCH 2/8] Remove broken authtoken references

---
 manifests/logging.pp | 16 +++++-----------
 1 file changed, 5 insertions(+), 11 deletions(-)

diff --git a/manifests/logging.pp b/manifests/logging.pp
index acb376d..d06cbd4 100644
--- a/manifests/logging.pp
+++ b/manifests/logging.pp
@@ -121,19 +121,13 @@ class neutron::logging (
 
   include ::neutron::deps
 
-  $debug_real = pick($::neutron::debug,$debug)
-  $use_syslog_real = pick($::neutron::use_syslog,$use_syslog)
-  $use_stderr_real = pick($::neutron::use_stderr,$use_stderr)
-  $log_file_real = pick($::neutron::log_file,$log_file)
-  $log_dir_real = pick($::neutron::log_dir,$log_dir)
-
   oslo::log { 'neutron_config':
-    debug                         => $debug_real,
-    use_stderr                    => $use_stderr_real,
-    use_syslog                    => $use_syslog_real,
+    debug                         => $debug,
+    use_stderr                    => $use_stderr,
+    use_syslog                    => $use_syslog,
     syslog_log_facility           => $syslog_log_facility,
-    log_file                      => $log_file_real,
-    log_dir                       => $log_dir_real,
+    log_file                      => $log_file,
+    log_dir                       => $log_dir,
     log_config_append             => $log_config_append,
     log_date_format               => $log_date_format,
     watch_log_file                => $watch_log_file,
-- 
1.8.3.1

