From efc8886902108994a799ab4bf7c5a8ddec0c0a8b Mon Sep 17 00:00:00 2001
From: Matt Peters <matt.peters@windriver.com>
Date: Sat, 26 May 2018 09:02:35 -0500
Subject: [PATCH] unset tunnel_types when tunneling disabled

---
 manifests/agents/ml2/ovs.pp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/manifests/agents/ml2/ovs.pp b/manifests/agents/ml2/ovs.pp
index a10c7b5..f7c19c2 100644
--- a/manifests/agents/ml2/ovs.pp
+++ b/manifests/agents/ml2/ovs.pp
@@ -310,6 +310,7 @@ class neutron::agents::ml2::ovs (
       'ovs/local_ip':              ensure => absent;
       'ovs/int_peer_patch_port':   ensure => absent;
       'ovs/tun_peer_patch_port':   ensure => absent;
+      'agent/tunnel_types':        ensure => absent;
     }
   }
 
-- 
1.8.3.1

