From 61ff7bbc259e5729f58536bdb6e3a554de6be514 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Tue, 27 Mar 2018 12:38:50 -0500
Subject: [PATCH 1/4] Update package versioning for TIS format

---
 SPECS/python.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/python.spec b/SPECS/python.spec
index 2af363e..c3706be 100644
--- a/SPECS/python.spec
+++ b/SPECS/python.spec
@@ -106,7 +106,7 @@ Summary: An interpreted, interactive, object-oriented programming language
 Name: %{python}
 # Remember to also rebase python-docs when changing this:
 Version: 2.7.5
-Release: 69%{?dist}
+Release: 69.el7_5%{?_tis_dist}.%{tis_patch_ver}
 License: Python
 Group: Development/Languages
 Requires: %{python}-libs%{?_isa} = %{version}-%{release}
-- 
2.7.4

