From 2f23c2c6194d828de7bede593926007802cca610 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Tue, 6 Mar 2018 18:21:48 -0600
Subject: [PATCH] Disable unittest check phase

---
 SPECS/python.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/python.spec b/SPECS/python.spec
index a9fd32f..1030c57 100644
--- a/SPECS/python.spec
+++ b/SPECS/python.spec
@@ -54,7 +54,7 @@
 %global with_gdbm 1
 
 # Turn this to 0 to turn off the "check" phase:
-%global run_selftest_suite 1
+%global run_selftest_suite 0
 
 # Some of the files below /usr/lib/pythonMAJOR.MINOR/test  (e.g. bad_coding.py)
 # are deliberately invalid, leading to SyntaxError exceptions if they get
-- 
1.8.3.1

