From 089860dac0d15120466dc7d9e38041dbaae505eb Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Wed, 28 Mar 2018 08:26:56 -0500
Subject: [PATCH] Make Python Faster

---
 SPECS/python.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/SPECS/python.spec b/SPECS/python.spec
index 94adb05..36289ea 100644
--- a/SPECS/python.spec
+++ b/SPECS/python.spec
@@ -1259,6 +1259,8 @@ Patch305: 00305-CVE-2016-2183.patch
 # above:
 Patch5000: 05000-autotool-intermediates.patch
 
+Patch9000: Rollup-optimization-commits.patch
+
 # ======================================================
 # Additional metadata, and subpackages
 # ======================================================
@@ -1670,6 +1672,7 @@ find -name "*~" |xargs rm -f
 %patch5000 -p0 -b .autotool-intermediates
 %endif
 
+%patch9000 -p1
 
 # ======================================================
 # Configuring and building the code:
@@ -1800,7 +1803,7 @@ BuildPython debug \
 BuildPython optimized \
   python \
   python%{pybasever} \
-  "" \
+   "--enable-optimizations --with-computed-gotos" \
   true
 
 
-- 
2.7.4

