From 83b8eb17d13fd5afdb926698082e47cba029c531 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Wed, 28 Mar 2018 15:23:35 -0500
Subject: [PATCH] Turn off with_systemtap to disable dtrace

---
 SPECS/python.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/python.spec b/SPECS/python.spec
index 2441951..c2edce8 100644
--- a/SPECS/python.spec
+++ b/SPECS/python.spec
@@ -42,7 +42,7 @@
 
 %global with_gdb_hooks 1
 
-%global with_systemtap 1
+%global with_systemtap 0
 
 # some arches don't have valgrind so we need to disable its support on them
 %ifarch %{ix86} x86_64 ppc %{power64} s390x aarch64
-- 
1.8.3.1

