From 5727bbd6fa8bd6f75888e3a53d4c50fe13ca5a33 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Thu, 4 Jan 2018 09:15:21 -0600
Subject: [PATCH 2/2] Apply xinetd patch

---
 SPECS/puppet-ironic.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-ironic.spec b/SPECS/puppet-ironic.spec
index ee83ddc..9199102 100644
--- a/SPECS/puppet-ironic.spec
+++ b/SPECS/puppet-ironic.spec
@@ -9,6 +9,7 @@ URL:            https://launchpad.net/puppet-ironic
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 Patch0001:      0001-Ironic-dbsync-NeutronGlance-dep-resolution.patch
+Patch0002:      0002-Dont-complain-about-xinetd-2.0.0.patch
 
 BuildArch:      noarch
 
@@ -26,6 +27,7 @@ Puppet module for OpenStack Ironic
 %prep
 %setup -q -n openstack-ironic-%{upstream_version}
 %patch0001 -p1
+%patch0002 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

