From be3e4af9cd22d1af7f5df456a6b67188b1cc2c25 Mon Sep 17 00:00:00 2001
From: Stefan Dinescu <stefan.dinescu@windriver.com>
Date: Thu, 16 Nov 2017 18:26:34 +0000
Subject: [PATCH] Add region cache support

---
 manifests/api.pp | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/manifests/api.pp b/manifests/api.pp
index d69059d..e563556 100644
--- a/manifests/api.pp
+++ b/manifests/api.pp
@@ -289,6 +289,13 @@
 #    (optional) Enable graceful shutdown on SIGUSR2
 #   Defaults to false, not set.
 #
+# [*remote_registry_region_name*]
+#    (optional) Remote region name for primary region/cloud
+#   Defaults to undef, not set
+#
+# [*data_api*]
+#    (optional) API for image info retrieval
+#   Defaults to glance.db.sqlalchemy.api
 #
 #  === deprecated parameters:
 #
@@ -363,6 +370,8 @@ class glance::api(
   $scrubber_datadir         = '/var/lib/glance/scrubber',
   $cache_raw_conversion_dir = '/opt/img-conversions/glance',
   $graceful_shutdown        = false,
+  $remote_registry_region_name      = undef,
+  $data_api                 = 'glance.db.sqlalchemy.api',
   # DEPRECATED PARAMETERS
   $known_stores                         = false,
 ) inherits glance {
@@ -408,6 +417,7 @@ class glance::api(
     'DEFAULT/cache_raw_conversion_dir': value=> $cache_raw_conversion_dir;
     'DEFAULT/graceful_shutdown':       value => $graceful_shutdown;
     'glance_store/os_region_name':     value => $os_region_name;
+    'DEFAULT/data_api':                value => $data_api;
   }
 
   # task/taskflow_executor config.
@@ -507,6 +517,16 @@ class glance::api(
     glance_api_config { 'paste_deploy/flavor': ensure => absent }
   }
 
+  if $remote_registry_region_name {
+    glance_api_config {
+      'DEFAULT/remote_registry_region_name':
+        ensure => present,
+        value  => $remote_registry_region_name,
+    }
+  } else {
+    glance_api_config { 'DEFAULT/remote_registry_region_name': ensure => absent }
+  }
+
   # keystone config
   if $auth_strategy == 'keystone' {
     include ::glance::api::authtoken
-- 
2.7.4

