From e87af6de1e474fc26c74eafaef2671b006f9b81b Mon Sep 17 00:00:00 2001
From: Andy Ning <andy.ning@windriver.com>
Date: Tue, 12 Jun 2018 09:05:27 -0400
Subject: [PATCH 1/1] add username and password authentication support to api

---
 manifests/api.pp | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/manifests/api.pp b/manifests/api.pp
index e563556..95230ee 100644
--- a/manifests/api.pp
+++ b/manifests/api.pp
@@ -374,6 +374,14 @@ class glance::api(
   $data_api                 = 'glance.db.sqlalchemy.api',
   # DEPRECATED PARAMETERS
   $known_stores                         = false,
+  # add glance API options for distributed keystone
+  $use_user_token                       = true,
+  $admin_user                           = 'admin',
+  $admin_tenant_name                    = 'admin',
+  $admin_password                       = '!!',
+  $auth_url                             = undef,
+  $api_auth_strategy                    = 'keystone',
+  $auth_region                          = 'RegionOne',
 ) inherits glance {
 
   include ::glance::deps
@@ -546,6 +554,19 @@ class glance::api(
     'DEFAULT/registry_client_key_file':  value => $registry_client_key_file;
   }
 
+  # add glanc API options for distributed keystone
+  if !$use_user_token { 
+    glance_api_config {
+      'DEFAULT/use_user_token':            value => $use_user_token;
+      'DEFAULT/admin_user':                value => $admin_user;
+      'DEFAULT/admin_tenant_name':         value => $admin_tenant_name; 
+      'DEFAULT/admin_password':            value => $admin_password;
+      'DEFAULT/auth_url':                  value => $auth_url;
+      'DEFAULT/auth_strategy':             value => $api_auth_strategy;
+      'DEFAULT/auth_region':               value => $auth_region;
+    }
+  }
+
   if $manage_service {
     if $enabled {
       $service_ensure = 'running'
-- 
1.8.3.1

