From 60166c6f060244ec2d1416d8a05d22fd6353424f Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 20 Mar 2017 11:10:38 -0400
Subject: [PATCH 1/2] WRS: 0001-Update-package-versioning-for-TIS-format.patch

---
 SPECS/puppet-haproxy.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-haproxy.spec b/SPECS/puppet-haproxy.spec
index 7d4b215..afb20d5 100644
--- a/SPECS/puppet-haproxy.spec
+++ b/SPECS/puppet-haproxy.spec
@@ -8,7 +8,7 @@
 
 Name:           puppet-haproxy
 Version:        1.5.0
-Release:        4%{?alphatag}%{?dist}
+Release:        4.6ffcb07git.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Configures HAProxy servers and manages the configuration of backend member servers.
 License:        ASL 2.0
 
-- 
2.7.4

