From 38e3d487daef2cfc9b820091cdb1290a8ffe0d2f Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 20 Mar 2017 11:10:38 -0400
Subject: [PATCH 2/2] WRS: 0002-Add-TIS-patch.patch

---
 SPECS/puppet-haproxy.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-haproxy.spec b/SPECS/puppet-haproxy.spec
index afb20d5..f58bd74 100644
--- a/SPECS/puppet-haproxy.spec
+++ b/SPECS/puppet-haproxy.spec
@@ -15,6 +15,7 @@ License:        ASL 2.0
 URL:            https://github.com/puppetlabs/puppetlabs-haproxy
 
 Source0:        https://github.com/puppetlabs/%{upstream_name}/archive/%{commit}.tar.gz#/%{upstream_name}-%{shortcommit}.tar.gz
+Patch0001:      0001-Roll-up-TIS-patches.patch
 
 BuildArch:      noarch
 
@@ -27,6 +28,7 @@ Configures HAProxy servers and manages the configuration of backend member serve
 
 %prep
 %setup -q -n %{upstream_name}-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.7.4

