From 62ba2a5aa0e9e254f65976a26f9b1b5751cddb47 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Sun, 24 Sep 2017 00:15:51 -0400
Subject: [PATCH] Add global_options patch

---
 SPECS/puppet-haproxy.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-haproxy.spec b/SPECS/puppet-haproxy.spec
index 106ae3e..fb11c75 100644
--- a/SPECS/puppet-haproxy.spec
+++ b/SPECS/puppet-haproxy.spec
@@ -17,6 +17,7 @@ URL:            https://github.com/puppetlabs/puppetlabs-haproxy
 Source0:        https://github.com/puppetlabs/%{upstream_name}/archive/%{commit}.tar.gz#/%{upstream_name}-%{shortcommit}.tar.gz
 Patch0001:      0001-Roll-up-TIS-patches.patch
 Patch0002:      0002-disable-config-validation-prechecks.patch
+Patch0003:      0003-Fix-global_options-log-default-value.patch
 
 BuildArch:      noarch
 
@@ -31,6 +32,7 @@ Configures HAProxy servers and manages the configuration of backend member serve
 %setup -q -n %{upstream_name}-%{upstream_version}
 %patch0001 -p1
 %patch0002 -p1
+%patch0003 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.7.4

