commit 4509fa0b8006dd147bb848b5648e334aedd26111
Author: Shoaib Nasir <shoaib.nasir@windriver.com>
Date:   Thu Nov 30 15:02:19 2017 -0500

    Added TIS Versioning to Pike 'puppet-ironic.spec'

diff --git a/SPECS/puppet-ironic.spec b/SPECS/puppet-ironic.spec
index 0375bd3..ba63c5e 100644
--- a/SPECS/puppet-ironic.spec
+++ b/SPECS/puppet-ironic.spec
@@ -1,7 +1,7 @@
 %{!?upstream_version: %global upstream_version %{version}%{?milestone}}
 Name:           puppet-ironic
 Version:        11.3.0
-Release:        1%{?dist}
+Release:        1%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Puppet module for OpenStack Ironic
 License:        ASL 2.0
 
