commit 399f1078a0a79d24b0c5fad573fd912eceacefae
Author: Shoaib Nasir <shoaib.nasir@windriver.com>
Date:   Fri Dec 1 16:16:52 2017 -0500

    Ironic dbsync with Neutron and Glance dependency resolved

diff --git a/SPECS/puppet-ironic.spec b/SPECS/puppet-ironic.spec
index ba63c5e..ee83ddc 100644
--- a/SPECS/puppet-ironic.spec
+++ b/SPECS/puppet-ironic.spec
@@ -8,6 +8,7 @@ License:        ASL 2.0
 URL:            https://launchpad.net/puppet-ironic
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
+Patch0001:      0001-Ironic-dbsync-NeutronGlance-dep-resolution.patch
 
 BuildArch:      noarch
 
@@ -24,6 +25,7 @@ Puppet module for OpenStack Ironic
 
 %prep
 %setup -q -n openstack-ironic-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
