From 20b00ae04480db94b1bb93ddc2cdbbf5cbafd001 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:29:01 -0400
Subject: [PATCH 2/3] WRS: ntpd-started-no-g-option.patch

---
 SOURCES/ntpd.sysconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SOURCES/ntpd.sysconfig b/SOURCES/ntpd.sysconfig
index 49394c7..c783890 100644
--- a/SOURCES/ntpd.sysconfig
+++ b/SOURCES/ntpd.sysconfig
@@ -1,2 +1,2 @@
 # Command line options for ntpd
-OPTIONS="-g"
+OPTIONS="-p /var/run/ntp.pid"
-- 
1.9.1

