Index: lldpd-0.9.0/src/daemon/lldpd.service.in
===================================================================
--- lldpd-0.9.0.orig/src/daemon/lldpd.service.in
+++ lldpd-0.9.0/src/daemon/lldpd.service.in
@@ -9,6 +9,7 @@ Type=notify
 NotifyAccess=main
 EnvironmentFile=-/etc/default/lldpd
 EnvironmentFile=-/etc/sysconfig/lldpd
+ExecStartPre=/bin/mkdir -p /var/run/lldpd
 ExecStart=@sbindir@/lldpd $DAEMON_ARGS $LLDPD_OPTIONS
 Restart=on-failure
 
