From ad85db9465da885a5f186db7f23655a3735a43c5 Mon Sep 17 00:00:00 2001
From: Teresa Ho <teresa.ho@windriver.com>
Date: Fri, 4 Jan 2019 10:49:27 -0500
Subject: [PATCH 1/1] Added ifup-alias-scope.patch

---
 SPECS/initscripts.spec | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/SPECS/initscripts.spec b/SPECS/initscripts.spec
index 6e9fc13..bff1e12 100644
--- a/SPECS/initscripts.spec
+++ b/SPECS/initscripts.spec
@@ -48,6 +48,7 @@ Patch9: sysconfig-unsafe-usage-of-linkdelay-variable.patch
 Patch10: ipv6-static-route-support.patch
 Patch11: ifup-eth-stop-waiting-if-link-is-up.patch
 Patch12: run-dhclient-as-daemon-for-ipv6.patch
+Patch13: ifup-alias-scope.patch
 
 %description
 The initscripts package contains basic system scripts used
@@ -80,7 +81,8 @@ Currently, this consists of various memory checking code.
 %patch10 -p1
 %patch11 -p1
 %patch12 -p1
- 
+%patch13 -p1
+
 %build
 make
 
-- 
1.8.3.1

