From 6c9a61e5ff22f2834dcc5ab91d32b370ef6a878c Mon Sep 17 00:00:00 2001
From: Joseph Richard <Joseph.Richard@windriver.com>
Date: Fri, 22 Dec 2017 15:01:11 -0500
Subject: [PATCH] spec: run dhclient as daemon for ipv6

---
 SPECS/initscripts.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/initscripts.spec b/SPECS/initscripts.spec
index 3adb83f..1ab8bb2 100644
--- a/SPECS/initscripts.spec
+++ b/SPECS/initscripts.spec
@@ -55,6 +55,7 @@ Patch11: sysconfig-affirmative-check-for-link-carrier.patch
 Patch12: sysconfig-unsafe-usage-of-linkdelay-variable.patch
 Patch13: ipv6-static-route-support.patch
 Patch14: ifup-eth-stop-waiting-if-link-is-up.patch
+Patch15: run-dhclient-as-daemon-for-ipv6.patch
 
 %description
 The initscripts package contains basic system scripts used
@@ -86,6 +87,7 @@ Currently, this consists of various memory checking code.
 %patch12 -p1
 %patch13 -p1
 %patch14 -p1
+%patch15 -p1
 
 
 %build
-- 
1.8.3.1

