From 2a4fe5935b7d9714752119e2a0b296a2d7202cd3 Mon Sep 17 00:00:00 2001
From: Denny Khoerniawan <denny.khoerniawan@windriver.com>
Date: Wed, 15 Nov 2017 14:09:05 -0500
Subject: [PATCH] spec: ifup-eth stop waiting if link is up

---
 SPECS/initscripts.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/initscripts.spec b/SPECS/initscripts.spec
index 82c5be0..3adb83f 100644
--- a/SPECS/initscripts.spec
+++ b/SPECS/initscripts.spec
@@ -54,6 +54,7 @@ Patch10: run-ifdown-on-all-interfaces.patch
 Patch11: sysconfig-affirmative-check-for-link-carrier.patch
 Patch12: sysconfig-unsafe-usage-of-linkdelay-variable.patch
 Patch13: ipv6-static-route-support.patch
+Patch14: ifup-eth-stop-waiting-if-link-is-up.patch
 
 %description
 The initscripts package contains basic system scripts used
@@ -84,6 +85,7 @@ Currently, this consists of various memory checking code.
 %patch11 -p1
 %patch12 -p1
 %patch13 -p1
+%patch14 -p1
 
 
 %build
-- 
1.8.3.1

