From f93a873f57cf8134e371668c7514923e6fda80f3 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:49:27 -0400
Subject: [PATCH 02/13] WRS: stop-creating-shared-dirs.patch

Signed-off-by: zhipengl <zhipengs.liu@intel.com>
---
 SPECS/initscripts.spec | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/SPECS/initscripts.spec b/SPECS/initscripts.spec
index d3c2ce6..967317f 100644
--- a/SPECS/initscripts.spec
+++ b/SPECS/initscripts.spec
@@ -189,10 +189,7 @@ rm -rf $RPM_BUILD_ROOT
 /usr/lib/systemd/rhel-*
 /usr/lib/systemd/system/*
 /etc/inittab
-%dir /etc/rc.d
-%dir /etc/rc.d/rc[0-9].d
 /etc/rc[0-9].d
-%dir /etc/rc.d/init.d
 /etc/rc.d/init.d/*
 %config(noreplace) /etc/sysctl.conf
 /usr/lib/sysctl.d/00-system.conf
@@ -214,7 +211,6 @@ rm -rf $RPM_BUILD_ROOT
 /usr/sbin/ppp-watch
 %{_mandir}/man*/*
 %dir %attr(775,root,root) /var/run/netreport
-%dir /etc/ppp
 %dir /etc/ppp/peers
 /etc/ppp/ip-up
 /etc/ppp/ip-down
@@ -222,8 +218,6 @@ rm -rf $RPM_BUILD_ROOT
 /etc/ppp/ip-down.ipv6to4
 /etc/ppp/ipv6-up
 /etc/ppp/ipv6-down
-%dir /etc/NetworkManager
-%dir /etc/NetworkManager/dispatcher.d
 /etc/NetworkManager/dispatcher.d/00-netreport
 %doc sysconfig.txt sysvinitfiles static-routes-ipv6 ipv6-tunnel.howto ipv6-6to4.howto changes.ipv6 COPYING
 %doc examples
-- 
1.8.3.1

