---
 lib/activate/dev_manager.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/lib/activate/dev_manager.c
+++ b/lib/activate/dev_manager.c
@@ -1210,17 +1210,18 @@ static int _thin_pool_callback(struct dm
 						 "global/thin_check_executable",
 						 DEFAULT_THIN_CHECK_EXECUTABLE);
 	const struct logical_volume *mlv = first_seg(data->pool_lv)->metadata_lv;
-	size_t len = strlen(dmdir) + strlen(mlv->vg->name) + strlen(mlv->name) + 3;
+	size_t len = strlen(dmdir) + 2 * strlen(mlv->vg->name) + strlen(mlv->name) + 3;
 	char meta_path[len];
 	int args;
 	char *argv[19]; /* Max supported 15 args */
-	char *split;
+	char *split, *dm_name;
 
 	if (!thin_check[0])
 		return 1; /* Checking disabled */
 
-	if (dm_snprintf(meta_path, len, "%s/%s-%s", dmdir,
-			mlv->vg->name, mlv->name) < 0) {
+	if (!(dm_name = dm_build_dm_name(data->dm->mem, mlv->vg->name,
+					 mlv->name, NULL)) ||
+	    (dm_snprintf(meta_path, len, "%s/%s", dmdir, dm_name) < 0)) {
 		log_error("Failed to build thin metadata path.");
 		return 0;
 	}
@@ -1263,6 +1264,7 @@ static int _thin_pool_callback(struct dm
 	}
 
 	dm_pool_free(data->dm->mem, split);
+	dm_pool_free(data->dm->mem, dm_name);
 
 	return ret;
 }
