From a24ad7ed432352f238f51429e3c24c600c726a8a Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:29:00 -0400
Subject: WRS: ntp-spec-add-TiS-custom-config-files.patch

---
 SPECS/ntp.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/ntp.spec b/SPECS/ntp.spec
index 52ad9aa..73ab0f7 100644
--- a/SPECS/ntp.spec
+++ b/SPECS/ntp.spec
@@ -46,6 +46,7 @@ Source13: ntpdate.service
 Source14: ntp-wait.service
 Source15: sntp.service
 Source16: sntp.sysconfig
+Source17: ntp-cgcs.conf
 
 # ntpbz #802
 Patch1: ntp-4.2.6p1-sleep.patch
@@ -462,6 +463,7 @@ sed -e 's|VENDORZONE\.|%{vendorzone}|' \
 touch -r %{SOURCE16} .%{_sysconfdir}/sysconfig/sntp
 install -p -m600 %{SOURCE8} .%{_sysconfdir}/ntp/crypto/pw
 install -p -m755 %{SOURCE10} .%{_sysconfdir}/dhcp/dhclient.d/ntp.sh
+install -p -m644 %{SOURCE17} .%{_sysconfdir}/ntp.conf
 install -p -m644 %{SOURCE12} .%{_unitdir}/ntpd.service
 install -p -m644 %{SOURCE13} .%{_unitdir}/ntpdate.service
 install -p -m644 %{SOURCE14} .%{_unitdir}/ntp-wait.service
-- 
2.7.4

