--- a/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in	2017-09-29 13:41:03.569697066 +0800
+++ b/rhel/usr_lib_systemd_system_ovs-vswitchd.service.in	2018-02-05 18:42:08.183083617 +0800
@@ -15,7 +15,11 @@
 EnvironmentFile=-/etc/sysconfig/openvswitch
 @begin_dpdk@
 ExecStartPre=-/usr/bin/chown :hugetlbfs /dev/hugepages
+ExecStartPre=-/usr/bin/chown -R :hugetlbfs /mnt/huge-2048kB
+ExecStartPre=-/usr/bin/chown -R :hugetlbfs /mnt/huge-1048576kB
 ExecStartPre=-/usr/bin/chmod 0775 /dev/hugepages
+ExecStartPre=-/usr/bin/chmod 0775 /mnt/huge-2048kB
+ExecStartPre=-/usr/bin/chmod 0775 /mnt/huge-1048576kB
 @end_dpdk@
 ExecStart=/usr/share/openvswitch/scripts/ovs-ctl \
           --no-ovsdb-server --no-monitor --system-id=random \
