From ef71415934f45a6f8ce0c5b67b03e61ded784b63 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:15:33 -0400
Subject: WRS: 0002-default-service-enabled.patch

---
 SPECS/iptables.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/iptables.spec b/SPECS/iptables.spec
index 0de8812..d83cf8c 100644
--- a/SPECS/iptables.spec
+++ b/SPECS/iptables.spec
@@ -222,6 +222,7 @@ done
 
 %post services
 %systemd_post iptables.service ip6tables.service
+/usr/bin/systemctl enable iptables.service ip6tables.service >/dev/null 2>&1
 
 %preun services
 %systemd_preun iptables.service ip6tables.service
-- 
2.7.4

