From 5b16910e95db49953a82fa583d99a0ac255dca1b Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:15:33 -0400
Subject: WRS: spec-include-custom-rules.patch

---
 SPECS/iptables.spec | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/SPECS/iptables.spec b/SPECS/iptables.spec
index b3c208d..1bcfb4d 100644
--- a/SPECS/iptables.spec
+++ b/SPECS/iptables.spec
@@ -16,6 +16,11 @@ Source4: iptables.save-legacy
 Source5: sysconfig_iptables
 Source6: sysconfig_ip6tables
 Source7: iptables.panic-legacy
+
+# WRS
+Source8: iptables.rules
+Source9: ip6tables.rules
+
 Patch1: iptables-1.4.21-rhbz_1054871.patch
 Patch2: iptables-1.4.21-libxt_cgroup.patch
 Patch3: iptables-1.4.21-wait_seconds.patch
@@ -74,8 +79,8 @@ Requires(post): systemd
 Requires(preun): systemd
 Requires(postun): systemd
 # provide and obsolete old main package
-Provides: %{name} = 1.4.16.1
-Obsoletes: %{name} < 1.4.16.1
+#Provides: %{name} = 1.4.16.1
+#Obsoletes: %{name} < 1.4.16.1
 # provide and obsolete ipv6 sub package
 Provides: %{name}-ipv6 = 1.4.11.1
 Obsoletes: %{name}-ipv6 < 1.4.11.1
@@ -178,6 +183,10 @@ sed -e 's;iptables.init;ip6tables.init;g' -e 's;IPTABLES;IP6TABLES;g' < %{buildr
 install -c -m 755 ip6tabes.save-legacy %{buildroot}/%{legacy_actions}/ip6tables/save
 install -c -m 755 ip6tabes.panic-legacy %{buildroot}/%{legacy_actions}/ip6tables/panic
 
+# WRS
+install -m 600 %{SOURCE8} %{buildroot}%{_sysconfdir}/sysconfig/iptables
+install -m 600 %{SOURCE9} %{buildroot}%{_sysconfdir}/sysconfig/ip6tables
+
 %if 0%{?rhel}
 %pre
 for p in %{_sysconfdir}/alternatives/iptables.*; do
-- 
2.7.4

