From dc2d2b15db75503e8b5f1bc639372902ad99333c Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Wed, 25 Oct 2017 16:12:09 -0400
Subject: [PATCH] Update package versioning for Titanium format

---
 SPECS/puppet-keystone.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-keystone.spec b/SPECS/puppet-keystone.spec
index 9dc1e0f..956664d 100644
--- a/SPECS/puppet-keystone.spec
+++ b/SPECS/puppet-keystone.spec
@@ -1,7 +1,7 @@
 %{!?upstream_version: %global upstream_version %{version}%{?milestone}}
 Name:           puppet-keystone
 Version:        11.3.0
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Puppet module for OpenStack Keystone
 License:        ASL 2.0
 
-- 
1.8.3.1

