From 145827dcc2c13fa18fbba2455773187dcee9b42e Mon Sep 17 00:00:00 2001
From: Kam Nasim <kam.nasim@windriver.com>
Date: Mon, 22 Jan 2018 11:43:27 -0500
Subject: [PATCH] remove the keystone admin app

---
 SPECS/puppet-keystone.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-keystone.spec b/SPECS/puppet-keystone.spec
index f0a18e8..42352df 100644
--- a/SPECS/puppet-keystone.spec
+++ b/SPECS/puppet-keystone.spec
@@ -11,6 +11,7 @@ Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_versio
 
 # WRS Patches
 Patch0001:      0001-pike-rebase-squash-titanium-patches.patch
+Patch0002:      0002-remove-the-Keystone-admin-app.patch
 
 BuildArch:      noarch
 
@@ -29,6 +30,7 @@ Puppet module for OpenStack Keystone
 
 # WRS Patches
 %patch0001 -p1
+%patch0002 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

