From 06305a07fa3028ab15f115a510e9e4a71c8730c4 Mon Sep 17 00:00:00 2001
From: Jerry Sun <jerry.sun@windriver.com>
Date: Tue, 30 Jan 2018 09:54:26 -0500
Subject: [PATCH 1/1] meta fixed puppet deprecated warnings

---
 SPECS/puppet-magnum.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-magnum.spec b/SPECS/puppet-magnum.spec
index fd77fec..207953e 100644
--- a/SPECS/puppet-magnum.spec
+++ b/SPECS/puppet-magnum.spec
@@ -10,6 +10,7 @@ URL:                    https://launchpad.net/puppet-magnum
 
 Source0:                https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 Patch0001:              0001-Pike-Rebase.patch
+Patch0002:              0002-fixed-puppet-deprecated-warnings.patch
 
 BuildArch:              noarch
 
@@ -26,6 +27,7 @@ Installs and configures OpenStack Magnum.
 %prep
 %setup -q -n openstack-magnum-%{upstream_version}
 %patch0001 -p1
+%patch0002 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

