From 402d56affb9e0767c4406260522fb356ecc420a0 Mon Sep 17 00:00:00 2001
From: jmckenna <jason.mckenna@windriver.com>
Date: Fri, 9 Feb 2018 13:51:24 -0500
Subject: [PATCH] update package patching

Signed-off-by: zhipengl <zhipengs.liu@intel.com>
---
 SPECS/dnsmasq.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/dnsmasq.spec b/SPECS/dnsmasq.spec
index 4d30b0a..689158e 100644
--- a/SPECS/dnsmasq.spec
+++ b/SPECS/dnsmasq.spec
@@ -13,7 +13,7 @@
 
 Name:           dnsmasq
 Version:        2.76
-Release:        7%{?extraversion}%{?dist}
+Release:        7.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        A lightweight DHCP/caching DNS server
 
 Group:          System Environment/Daemons
-- 
1.8.3.1

