From e257b88db5d62a1e44e94373917dd3f97de12522 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:27:13 -0400
Subject: [PATCH 3/5] WRS: patch-tftp-to-close-sockets-immediately.patch

---
 SPECS/dnsmasq.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/dnsmasq.spec b/SPECS/dnsmasq.spec
index 0641503..5cf440a 100644
--- a/SPECS/dnsmasq.spec
+++ b/SPECS/dnsmasq.spec
@@ -58,6 +58,7 @@ Patch19:	dnsmasq-2.76-misc-cleanups.patch
 
 # WRS patches
 Patch30:        dnsmasq-update-ipv6-leases-from-config.patch
+Patch31:        close-tftp-sockets-immediately.patch
 
 
 BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
@@ -116,6 +117,7 @@ query/remove a DHCP server's leases.
 
 # WRS patches
 %patch30 -p1
+%patch31 -p1
 
 # use /var/lib/dnsmasq instead of /var/lib/misc
 for file in dnsmasq.conf.example man/dnsmasq.8 man/es/dnsmasq.8 src/config.h; do
-- 
2.7.4

