From 9771ea4fdcbea1f4124564654b0541dcb8ccf780 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:11:21 -0400
Subject: [PATCH 3/5] WRS: openldap-sysconfig-file.patch

---
 SOURCES/slapd.sysconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SOURCES/slapd.sysconfig b/SOURCES/slapd.sysconfig
index 68091a5..573486d 100644
--- a/SOURCES/slapd.sysconfig
+++ b/SOURCES/slapd.sysconfig
@@ -9,7 +9,7 @@
 SLAPD_URLS="ldapi:/// ldap:///"
 
 # Any custom options
-#SLAPD_OPTIONS=""
+SLAPD_OPTIONS=""
 
 # Keytab location for GSSAPI Kerberos authentication
 #KRB5_KTNAME="FILE:/etc/openldap/ldap.keytab"
-- 
1.9.1

