From 151218c66280ec3984daf0a476e7b5cac69d305a Mon Sep 17 00:00:00 2001
From: Andy Ning <andy.ning@windriver.com>
Date: Wed, 28 Mar 2018 14:20:39 -0400
Subject: [PATCH 7/7] fix systemd tmpfiles ACL warnings

---
 SPECS/systemd.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/systemd.spec b/SPECS/systemd.spec
index 6152e33..ffd0770 100644
--- a/SPECS/systemd.spec
+++ b/SPECS/systemd.spec
@@ -661,6 +661,7 @@ Patch0705: 0705-remove-id-sas-path-symlink.patch
 Patch0706: 0706-fix-ACL-warnings-from-systemd-tmpfiles-set.patch
 Patch0707: 0707-move-vartmp-to-tmpfs.patch
 Patch0708: 0708-set-a-1GB-size-restriction-on-tpmfs.patch
+Patch0709: 0709-fix-systemd-tmpfiles-ACL-warnings.patch
 
 %global num_patches %{lua: c=0; for i,p in ipairs(patches) do c=c+1; end; print(c);}
 
-- 
2.7.4

