From 6aead74fb56ae75cc16be507165d3fc75c38fac0 Mon Sep 17 00:00:00 2001
From: slin14 <shuicheng.lin@intel.com>
Date: Thu, 9 Aug 2018 18:40:36 +0800
Subject: [PATCH] Add 0710-fix-build-error-for-unused-variable.patch

Signed-off-by: slin14 <shuicheng.lin@intel.com>
---
 SPECS/systemd.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/systemd.spec b/SPECS/systemd.spec
index ffd0770..3f7cc10 100644
--- a/SPECS/systemd.spec
+++ b/SPECS/systemd.spec
@@ -662,6 +662,7 @@ Patch0706: 0706-fix-ACL-warnings-from-systemd-tmpfiles-set.patch
 Patch0707: 0707-move-vartmp-to-tmpfs.patch
 Patch0708: 0708-set-a-1GB-size-restriction-on-tpmfs.patch
 Patch0709: 0709-fix-systemd-tmpfiles-ACL-warnings.patch
+Patch0710: 0710-fix-build-error-for-unused-variable.patch
 
 %global num_patches %{lua: c=0; for i,p in ipairs(patches) do c=c+1; end; print(c);}
 
-- 
2.7.4

