From 6f9a8e792bad1f4bb8d9dad4f2430d89a1058d0c Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:53:00 -0400
Subject: [PATCH] WRS: 0005-spec-millisec-in-syslog-date.patch

---
 SPECS/systemd.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/systemd.spec b/SPECS/systemd.spec
index 6a04c16..845d1dd 100644
--- a/SPECS/systemd.spec
+++ b/SPECS/systemd.spec
@@ -712,6 +712,9 @@ Patch0670: 0670-Refuse-dbus-message-paths-longer-than-BUS_PATH_SIZE_.patch
 Patch0671: 0671-Allocate-temporary-strings-to-hold-dbus-paths-on-the.patch
 Patch0672: 0672-sd-bus-if-we-receive-an-invalid-dbus-message-ignore-.patch
 
+#WRS Patches
+Patch0701: 0701-inject-millisec-in-syslog-date.patch
+
 %global num_patches %{lua: c=0; for i,p in ipairs(patches) do c=c+1; end; print(c);}
 
 BuildRequires:  libcap-devel
-- 
2.7.4

