From 6f9a8e792bad1f4bb8d9dad4f2430d89a1058d0c Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:53:00 -0400
Subject: [PATCH] WRS: 0005-spec-millisec-in-syslog-date.patch

---
 SPECS/systemd.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/systemd.spec b/SPECS/systemd.spec
index 6a04c16..845d1dd 100644
--- a/SPECS/systemd.spec
+++ b/SPECS/systemd.spec
@@ -704,6 +704,9 @@ Patch0662: 0662-cryptsetup-generator-don-t-return-error-if-target-di.patch
 Patch0663: 0663-cryptsetup-generator-allow-whitespace-characters-in-.patch
 Patch0664: 0664-Make-sure-the-mount-units-pulled-by-RequiresMountsFo.patch
 
+#WRS Patches
+Patch0701: 0701-inject-millisec-in-syslog-date.patch
+
 %global num_patches %{lua: c=0; for i,p in ipairs(patches) do c=c+1; end; print(c);}

 BuildRequires:  libcap-devel
--
2.7.4

