From e1b3591beb36a45f7cc4bf9340dd640b359955b7 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:53:00 -0400
Subject: [PATCH] WRS: 0001-update-package-versioning-for-TIS-format.patch

---
 SPECS/systemd.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/systemd.spec b/SPECS/systemd.spec
index 6bdbb74..3b2aa7f 100644
--- a/SPECS/systemd.spec
+++ b/SPECS/systemd.spec
@@ -7,7 +7,7 @@
 Name:           systemd
 Url:            http://www.freedesktop.org/wiki/Software/systemd
 Version:        219
-Release:        62%{?dist}
+Release:        62.el7%{?_tis_dist}.%{tis_patch_ver}
 # For a breakdown of the licensing, see README
 License:        LGPLv2+ and MIT and GPLv2+
 Summary:        A System and Service Manager
--
2.7.4

