From f57e1196deb8890c143ebd978f0b83e4c0e749f8 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:09:29 -0400
Subject: [PATCH 3/3] WRS: meta-e2fsprogs-disable-tests.patch

---
 SPECS/e2fsprogs.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/e2fsprogs.spec b/SPECS/e2fsprogs.spec
index 38ff00b..0eb885a 100644
--- a/SPECS/e2fsprogs.spec
+++ b/SPECS/e2fsprogs.spec
@@ -37,6 +37,7 @@ Patch24: e2fsprogs-1.42.10-Fix-nroff-macro-issue-in-chattr-man-page.patch
 
 # WRS patches
 Patch100: 0100-tamper-proof-bash-log.patch
+Patch101: 0101-e2fsprogs-disable-tests.patch
 
 Url: http://e2fsprogs.sourceforge.net/
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
@@ -213,6 +214,7 @@ It was originally inspired by the Multics SubSystem library.
 
 # WRS patches
 %patch100 -p1
+%patch101 -p1
 
 %build
 %configure --enable-elf-shlibs --enable-nls --disable-uuidd --disable-fsck \
-- 
1.9.1

