From 8f4ade8974fdb044f86ab72533b5ce847875089d Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 3 Jan 2017 13:37:20 -0500
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/puppet-ceph.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-ceph.spec b/SPECS/puppet-ceph.spec
index 8cfc8c1..3d7dccf 100644
--- a/SPECS/puppet-ceph.spec
+++ b/SPECS/puppet-ceph.spec
@@ -1,7 +1,7 @@
 %{!?upstream_version: %global upstream_version %{version}%{?milestone}}
 Name:           puppet-ceph
 Version:        2.4.1
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Community Developed Ceph Module
 License:        ASL 2.0
 
-- 
2.7.4

