From ff91fc6138685fcaff0222686e83ba7d083e5630 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 7 Feb 2017 14:08:21 -0500
Subject: [PATCH] Add TIS patches

---
 SPECS/puppet-ceph.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/puppet-ceph.spec b/SPECS/puppet-ceph.spec
index 3d7dccf..5b0deea 100644
--- a/SPECS/puppet-ceph.spec
+++ b/SPECS/puppet-ceph.spec
@@ -8,6 +8,8 @@ License:        ASL 2.0
 URL:            https://launchpad.net/puppet-ceph
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
+Patch0001:      0001-Roll-up-TIS-patches.patch
+Patch0002:      0002-Newton-rebase-fixes.patch
 
 BuildArch:      noarch
 
@@ -23,6 +25,8 @@ Community Developed Ceph Module
 
 %prep
 %setup -q -n openstack-ceph-%{upstream_version}
+%patch0001 -p1
+%patch0002 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.7.4

