From 0fadbf6ad0702c390632e687656c91a48a417c99 Mon Sep 17 00:00:00 2001
From: Daniel Badea <daniel.badea@windriver.com>
Date: Thu, 23 Mar 2017 08:07:56 +0000
Subject: [PATCH] ceph jewel rebase

---
 SPECS/puppet-ceph.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-ceph.spec b/SPECS/puppet-ceph.spec
index 5b0deea..23e0562 100644
--- a/SPECS/puppet-ceph.spec
+++ b/SPECS/puppet-ceph.spec
@@ -10,6 +10,7 @@ URL:            https://launchpad.net/puppet-ceph
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 Patch0001:      0001-Roll-up-TIS-patches.patch
 Patch0002:      0002-Newton-rebase-fixes.patch
+Patch0003:      0003-Ceph-Jewel-rebase.patch
 
 BuildArch:      noarch
 
@@ -27,6 +28,7 @@ Community Developed Ceph Module
 %setup -q -n openstack-ceph-%{upstream_version}
 %patch0001 -p1
 %patch0002 -p1
+%patch0003 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.7.4

