From e98b25d0168caa56d72e62e98f3c6701df704db5 Mon Sep 17 00:00:00 2001
From: Robert Church <robert.church@windriver.com>
Date: Fri, 14 Apr 2017 02:04:43 +0000
Subject: [PATCH] Add OSD support for persistent naming

---
 SPECS/puppet-ceph.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-ceph.spec b/SPECS/puppet-ceph.spec
index 23e0562..ef82b81 100644
--- a/SPECS/puppet-ceph.spec
+++ b/SPECS/puppet-ceph.spec
@@ -11,6 +11,7 @@ Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_versio
 Patch0001:      0001-Roll-up-TIS-patches.patch
 Patch0002:      0002-Newton-rebase-fixes.patch
 Patch0003:      0003-Ceph-Jewel-rebase.patch
+Patch0004:      0004-US92424-Add-OSD-support-for-persistent-naming.patch
 
 BuildArch:      noarch
 
@@ -29,6 +30,7 @@ Community Developed Ceph Module
 %patch0001 -p1
 %patch0002 -p1
 %patch0003 -p1
+%patch0004 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.7.4

