From a6572b58090f2c9bf492d8745ce7cbd0fb8ede33 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Wed, 3 Jan 2018 10:42:13 -0600
Subject: [PATCH] meta patch for patch5

---
 SPECS/puppet-ceph.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-ceph.spec b/SPECS/puppet-ceph.spec
index ef82b81..ad75ea5 100644
--- a/SPECS/puppet-ceph.spec
+++ b/SPECS/puppet-ceph.spec
@@ -12,6 +12,7 @@ Patch0001:      0001-Roll-up-TIS-patches.patch
 Patch0002:      0002-Newton-rebase-fixes.patch
 Patch0003:      0003-Ceph-Jewel-rebase.patch
 Patch0004:      0004-US92424-Add-OSD-support-for-persistent-naming.patch
+Patch0005:      0005-Remove-puppetlabs-apt-as-ceph-requirement.patch
 
 BuildArch:      noarch
 
@@ -31,6 +32,7 @@ Community Developed Ceph Module
 %patch0002 -p1
 %patch0003 -p1
 %patch0004 -p1
+%patch0005 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.7.4

