From b590f06d6f6ce2bd71d4d0389b6d51a78e225c19 Mon Sep 17 00:00:00 2001
From: Ovidiu Poncea <ovidiu.poncea@windriver.com>
Date: Thu, 20 Dec 2018 08:07:15 -0500
Subject: [PATCH] Add-StarlingX-specific-restart-command-for-Ceph-moni patch

---
 SPECS/puppet-ceph.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-ceph.spec b/SPECS/puppet-ceph.spec
index 0b728a1..e5cc64c 100644
--- a/SPECS/puppet-ceph.spec
+++ b/SPECS/puppet-ceph.spec
@@ -14,6 +14,7 @@ Patch0003:      0003-Ceph-Jewel-rebase.patch
 Patch0004:      0004-US92424-Add-OSD-support-for-persistent-naming.patch
 Patch0005:      0005-Remove-puppetlabs-apt-as-ceph-requirement.patch
 Patch0006:      0006-ceph-disk-prepare-invalid-data-disk-value.patch
+Patch0007:      0007-Add-StarlingX-specific-restart-command-for-Ceph-moni.patch
 
 BuildArch:      noarch
 
@@ -35,6 +36,7 @@ Community Developed Ceph Module
 %patch0004 -p1
 %patch0005 -p1
 %patch0006 -p1
+%patch0007 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

