From 88ffd373172ecabdda7235cce1efa3b7014a839f Mon Sep 17 00:00:00 2001
From: Daniel Badea <daniel.badea@windriver.com>
Date: Thu, 25 Apr 2019 15:42:38 +0000
Subject: [PATCH] fix ceph osd disk partition for nvme disks

---
 SPECS/puppet-ceph.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-ceph.spec b/SPECS/puppet-ceph.spec
index 95c1f0e..6b6cac2 100644
--- a/SPECS/puppet-ceph.spec
+++ b/SPECS/puppet-ceph.spec
@@ -16,6 +16,7 @@ Patch0005:      0005-Remove-puppetlabs-apt-as-ceph-requirement.patch
 Patch0006:      0006-ceph-disk-prepare-invalid-data-disk-value.patch
 Patch0007:      0007-Add-StarlingX-specific-restart-command-for-Ceph-moni.patch
 Patch0008:      0008-ceph-mimic-prepare-activate-osd.patch
+Patch0009:      0009-fix-ceph-osd-disk-partition-for-nvme-disks.patch
 
 BuildArch:      noarch
 
@@ -39,6 +40,7 @@ Community Developed Ceph Module
 %patch0006 -p1
 %patch0007 -p1
 %patch0008 -p1
+%patch0009 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

