From b8dee2da527c3d3010e2b5b4e49f87f430afa826 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Fri, 3 Nov 2017 17:48:38 -0500
Subject: [PATCH] Remove log_dir from conf files

---
 manifests/log.pp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/manifests/log.pp b/manifests/log.pp
index 8778d46..771a674 100644
--- a/manifests/log.pp
+++ b/manifests/log.pp
@@ -27,9 +27,7 @@
 #   Defaults to $::os_service_default
 #
 # [*log_dir*]
-#   (Optional) Directory where logs should be stored.
-#   If set to $::os_service_default, it will not log to any directory.
-#   Defaults to $::os_service_default
+#   WRS: Remove log_dir to ensure services log via syslog
 #
 # [*watch_log_file*]
 #   (Optional) Uses logging handler designed to watch file system (boolean value).
@@ -136,7 +134,7 @@ define oslo::log(
     'DEFAULT/log_config_append'             => { value => $log_config_append },
     'DEFAULT/log_date_format'               => { value => $log_date_format },
     'DEFAULT/log_file'                      => { value => $log_file },
-    'DEFAULT/log_dir'                       => { value => $log_dir },
+    'DEFAULT/log_dir'                       => { ensure => absent },
     'DEFAULT/watch_log_file'                => { value => $watch_log_file },
     'DEFAULT/use_syslog'                    => { value => $use_syslog },
     'DEFAULT/syslog_log_facility'           => { value => $syslog_log_facility },
-- 
1.8.3.1

