From fc1f1853e99c5afaae334b0c37296e34e9cf19fd Mon Sep 17 00:00:00 2001
From: root <root@yow-cgts4-lx.wrs.com>
Date: Mon, 15 Jan 2018 13:09:41 -0500
Subject: [PATCH 1/2] Ti version string

---
 SPECS/shim-unsigned-x64.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/shim-unsigned-x64.spec b/SPECS/shim-unsigned-x64.spec
index 0261a47..aa45c33 100755
--- a/SPECS/shim-unsigned-x64.spec
+++ b/SPECS/shim-unsigned-x64.spec
@@ -18,7 +18,7 @@
 
 Name:		shim-unsigned-%{efiarch}
 Version:	15
-Release:	2%{?dist}
+Release:	2.el8%{?_tis_dist}.%{tis_patch_ver}
 Summary:	First-stage UEFI bootloader
 ExclusiveArch:	x86_64
 License:	BSD
-- 
2.7.4

