From fb4da7f4d7d8e8565371ed236150de2e4bb47b95 Mon Sep 17 00:00:00 2001
From: root <root@yow-cgts4-lx.wrs.com>
Date: Mon, 15 Jan 2018 13:22:09 -0500
Subject: [PATCH 2/2] Add Ti certificate

---
 SPECS/shim-unsigned-x64.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/shim-unsigned-x64.spec b/SPECS/shim-unsigned-x64.spec
index aa45c33..384d339 100755
--- a/SPECS/shim-unsigned-x64.spec
+++ b/SPECS/shim-unsigned-x64.spec
@@ -30,12 +30,14 @@ Source1:	centossecureboot001.der
 Source2:	dbx.esl
 
 Source100:	shim-find-debuginfo.sh
+Source1000:     tis-shim.crt
 
 Patch0001:	0001-Make-sure-that-MOK-variables-always-get-mirrored.patch
 Patch0002:	0002-mok-fix-the-mirroring-of-RT-variables.patch
 Patch0003:	0003-mok-consolidate-mirroring-code-in-a-helper-instead-o.patch
 Patch0004:	0004-Make-VLogError-behave-as-expected.patch
 Patch0005:	0005-MokListRT-Fatal.patch
+Patch1000:	0001-Use-Titanium-certificate.patch
 
 BuildRequires:	elfutils-libelf-devel
 BuildRequires:	git openssl-devel openssl
@@ -100,6 +102,8 @@ BuildArch:	noarch
 %autosetup -S git -n shim-%{version}
 git config --unset user.email
 git config --unset user.name
+cp %{SOURCE1000} .
+
 mkdir build-%{efiarch}
 mkdir build-%{efialtarch}
 
-- 
2.7.4

