From a15d83975ed19367767e18354ea07cd5d281e265 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 27 Sep 2016 10:14:02 -0400
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/crontabs.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/crontabs.spec b/SPECS/crontabs.spec
index 7001d1a..4512bcc 100644
--- a/SPECS/crontabs.spec
+++ b/SPECS/crontabs.spec
@@ -2,7 +2,7 @@
 Summary: Root crontab files used to schedule the execution of programs
 Name: crontabs
 Version: 1.11
-Release: 6.%{snap_release}%{?dist}
+Release: 6.%{snap_release}.el7%{?_tis_dist}.%{tis_patch_ver}
 License: Public Domain and GPLv2
 Group: System Environment/Base
 URL: https://fedorahosted.org/crontabs 
-- 
1.8.3.1

