From 9221bd11aec1590df2dc3f19e9a582d76ed7adc0 Mon Sep 17 00:00:00 2001
From: Michel Thebeau <michel.thebeau@windriver.com>
Date: Mon, 25 Jul 2016 11:45:55 -0400
Subject: [PATCH] spec: add patch to remove printing of motd script name

Signed-off-by: Michel Thebeau <michel.thebeau@windriver.com>
---
 SPECS/crontabs.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/crontabs.spec b/SPECS/crontabs.spec
index d6a8c69..90098e1 100644
--- a/SPECS/crontabs.spec
+++ b/SPECS/crontabs.spec
@@ -10,6 +10,8 @@ Source0: https://fedorahosted.org/releases/c/r/crontabs/%{name}-%{version}-1.%{s
 BuildArch: noarch
 Requires: /etc/cron.d
 
+Patch1: run-parts-add-option-to-remove-printing-of-motd-scri.patch
+
 %description
 This package is used by Fedora mainly for executing files by cron.
 
@@ -25,6 +27,8 @@ your system.
 %prep
 %setup -q 
 
+%patch1 -p1
+
 %build
 #empty
 
-- 
1.8.3.1

