From b6ceef1c915827b50ce3f76da4dc47f3eb768b44 Mon Sep 17 00:00:00 2001
Message-Id: <b6ceef1c915827b50ce3f76da4dc47f3eb768b44.1527544850.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Thu, 15 Dec 2016 14:27:48 -0500
Subject: [PATCH 01/26] Fix compile issue when transparent hugepages are off

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 mm/swap.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/swap.c b/mm/swap.c
index 0982a35..6dcf38c 100644
--- a/mm/swap.c
+++ b/mm/swap.c
@@ -998,8 +998,10 @@ void release_pages(struct page **pages, int nr, bool cold)
 		if (!put_page_testzero(page))
 			continue;
 
+#ifdef CONFIG_TRANSPARENT_HUGEPAGE
 		VM_BUG_ON_PAGE(check_mmu_gather &&
 			       trans_huge_mmu_gather_count(page), page);
+#endif
 
 		if (PageLRU(page)) {
 			if (!was_thp)
-- 
1.8.3.1

