diff --git a/config.c b/config.c
index c1087a5..add7c53 100644
--- a/config.c
+++ b/config.c
@@ -6,6 +6,8 @@
   This file is part of IBSH (Iron Bars Shell) , a restricted Unix shell
   Copyright (C) 2005  Attila Nagyidai
 
+  Copyright(c) 2013-2017 Wind River Systems, Inc. All rights reserved.
+
   This program is free software; you can redistribute it and/or
   modify it under the terms of the GNU General Public License
   as published by the Free Software Foundation; either version 2
diff --git a/main.c b/main.c
index cf3ae9e..6cda04e 100644
--- a/main.c
+++ b/main.c
@@ -6,6 +6,8 @@
   This file is part of IBSH (Iron Bars Shell) , a restricted Unix shell
   Copyright (C) 2005  Attila Nagyidai
 
+  Copyright(c) 2013-2017 Wind River Systems, Inc. All rights reserved.
+
   This program is free software; you can redistribute it and/or
   modify it under the terms of the GNU General Public License
   as published by the Free Software Foundation; either version 2
