From 8f91c53ae95b97e12d4a7b16fac8f3d5195ccd52 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 20 Mar 2017 10:21:28 -0400
Subject: [PATCH 1/4] spec-include-TiS-changes.patch

Signed-off-by: Dongqi Chen <chen.dq@neusoft.com>
Signed-off-by: Yong Fu <fuyong@neusoft.com>
---
 SPECS/lighttpd.spec | 27 ++++++++++++++-------------
 1 file changed, 14 insertions(+), 13 deletions(-)

diff --git a/SPECS/lighttpd.spec b/SPECS/lighttpd.spec
index a11898f..8b029b4 100644
--- a/SPECS/lighttpd.spec
+++ b/SPECS/lighttpd.spec
@@ -57,6 +57,7 @@ Source11: http://www.lighttpd.net/favicon.ico
 Source12: http://www.lighttpd.net/light_button.png
 Source13: http://www.lighttpd.net/light_logo.png
 Source14: lighttpd-empty.png
+
 Patch0: lighttpd-1.4.55-defaultconf.patch
 Requires: system-logos
 Requires: %{name}-filesystem
@@ -162,16 +163,18 @@ autoreconf -if
     --libdir='%{_libdir}/lighttpd' \
     %{confswitch mysql} \
     %{confswitch pam} \
-    %{confswitch ldap} \
+    --without-ldap \
     %{confswitch attr} \
-    %{confswitch openssl} \
-    %{confswitch pcre} \
+    --with-openssl \
+    --with-pcre \
     %{confswitch fam} \
-    %{?with_webdavprops:--with-webdav-props} \
-    %{?with_webdavlocks:--with-webdav-locks} \
+    --without-webdav-props \
+    --without-webdav-locks \
     %{confswitch gdbm} \
-    %{confswitch memcached} \
-    %{confswitch lua} \
+    --without-memcached \
+    --without-lua \
+    --without-bzip2 \
+    --disable-static \
     %{confswitch geoip} \
     %{confswitch krb5}
 make %{?_smp_mflags}
@@ -189,13 +192,14 @@ install -D -p -m 0644 %{SOURCE2} \
     %{buildroot}%{_sysconfdir}/php.d/lighttpd.ini

 # Install our own init script (included one is old style) or systemd service
-%if %{with systemd}
+#%if %{with systemd}
 install -D -p -m 0644 %{SOURCE4} \
     %{buildroot}%{_unitdir}/lighttpd.service
-%else
+#%else
+mkdir -p /etc/rc.d/init.d
 install -D -p -m 0755 %{SOURCE3} \
     %{buildroot}%{_sysconfdir}/rc.d/init.d/lighttpd
-%endif
+#%endif

 # Install our own default web page and images
 mkdir -p %{buildroot}%{webroot}
@@ -278,11 +282,8 @@ fi
 %config %{_sysconfdir}/lighttpd/conf.d/mod.template
 %config %{_sysconfdir}/lighttpd/vhosts.d/vhosts.template
 %config(noreplace) %{_sysconfdir}/logrotate.d/lighttpd
-%if %{with systemd}
 %{_unitdir}/lighttpd.service
-%else
 %{_sysconfdir}/rc.d/init.d/lighttpd
-%endif
 %if %{with tmpfiles}
 %config(noreplace) /usr/lib/tmpfiles.d/lighttpd.conf
 %endif
--
2.17.1

