From 1c4a8d83d96eab943d1cb7b4f0d9b7175e6858f1 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 20 Mar 2017 10:21:28 -0400
Subject: [PATCH 2/4] Update-package-versioning-for-TIS-format.patch

Signed-off-by: Dongqi Chen <chen.dq@neusoft.com>
---
 SPECS/lighttpd.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/lighttpd.spec b/SPECS/lighttpd.spec
index 20c4503..3280403 100644
--- a/SPECS/lighttpd.spec
+++ b/SPECS/lighttpd.spec
@@ -44,7 +44,7 @@
 Summary: Lightning fast webserver with light system requirements
 Name: lighttpd
 Version: 1.4.55
-Release: 1%{?dist}
+Release: 1.el8%{?_tis_dist}.%{tis_patch_ver}
 License: BSD
 URL: http://www.lighttpd.net/
 Source0: http://download.lighttpd.net/lighttpd/releases-1.4.x/lighttpd-%{version}.tar.gz
-- 
2.7.4

