From 61010c2ba23abfa0e69ab884f3fff0c252544c8b Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 27 Sep 2016 11:00:04 -0400
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/scsi-target-utils.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/scsi-target-utils.spec b/SPECS/scsi-target-utils.spec
index c97ada4..1778e8f 100644
--- a/SPECS/scsi-target-utils.spec
+++ b/SPECS/scsi-target-utils.spec
@@ -14,7 +14,7 @@
 
 Name:           scsi-target-utils
 Version:        1.0.55
-Release:        4%{?dist}
+Release:        4.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        The SCSI target daemon and utility programs
 
 Group:          System Environment/Daemons
-- 
1.8.3.1

