From 2f7d1d4fd969e529e8efcdcf8ea6c0040e3392e3 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Thu, 4 Jan 2018 12:54:34 -0600
Subject: [PATCH 2/2] Apply rename patch

---
 SPECS/puppet-staging.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-staging.spec b/SPECS/puppet-staging.spec
index 0dd5d59..b6c93ce 100644
--- a/SPECS/puppet-staging.spec
+++ b/SPECS/puppet-staging.spec
@@ -15,6 +15,7 @@ License:        Apache-2.0
 URL:            https://github.com/nanliu/puppet-staging
 
 Source0:        https://github.com/nanliu/%{upstream_name}/archive/%{commit}.tar.gz#/%{upstream_name}-%{shortcommit}.tar.gz
+Patch0001:      0001-Rename-nanliu-staging-to-puppet-staging.patch
 
 BuildArch:      noarch
 
@@ -25,6 +26,7 @@ Compressed file staging and deployment
 
 %prep
 %setup -q -n %{name}-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

