From 67ee34be8bd3e7829442c97600118148f874ab8d Mon Sep 17 00:00:00 2001
From: Joseph Richard <Joseph.Richard@windriver.com>
Date: Thu, 9 Nov 2017 17:57:50 -0500
Subject: [PATCH 2/2] Add TiS patches

---
 SPECS/puppet-neutron.spec | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/SPECS/puppet-neutron.spec b/SPECS/puppet-neutron.spec
index 12252ad..684b3d6 100644
--- a/SPECS/puppet-neutron.spec
+++ b/SPECS/puppet-neutron.spec
@@ -8,6 +8,14 @@ License:        ASL 2.0
 URL:            https://launchpad.net/puppet-neutron
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
+Patch0001:      0001-Roll-up-TIS-patches.patch
+Patch0002:      0002-Remove-broken-authtoken-references.patch
+Patch0003:      0003-control-provider-network-testing-state.patch
+Patch0004:      0004-set-wsgi-pool-size-to-10-from-100.patch
+Patch0005:      0005-Add-support-for-neutron-bgp-dragent.patch
+Patch0006:      0006-Add-parameters-to-classes.patch
+Patch0007:      0007-update-neutron-bgp-to-use-class-variables.patch
+Patch0008:      0008-dvr-allow-base-mac-address-to-be-configured.patch
 
 BuildArch:      noarch
 
@@ -26,6 +34,14 @@ Puppet module for OpenStack Neutron
 
 %prep
 %setup -q -n openstack-neutron-%{upstream_version}
+%patch0001 -p1
+%patch0002 -p1
+%patch0003 -p1
+%patch0004 -p1
+%patch0005 -p1
+%patch0006 -p1
+%patch0007 -p1
+%patch0008 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

