From bd351d61990e5b7490bca78c1e5541ae895e97eb Mon Sep 17 00:00:00 2001
From: "Zhang Zhiguo" <zhangzhg@neusoft.com>
Date: Wed, 11 Dec 2019 10:37:42 +0800
Subject: Support-Centos8-OS

Signed-off-by: Zhang Zhiguo <zhangzhg@neusoft.com>

---
 SPECS/mlnx-ofa_kernel.spec | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/SPECS/mlnx-ofa_kernel.spec b/SPECS/mlnx-ofa_kernel.spec
index b8adb35..138b32b 100644
--- a/SPECS/mlnx-ofa_kernel.spec
+++ b/SPECS/mlnx-ofa_kernel.spec
@@ -48,6 +48,7 @@ BuildRequires: kernel%{?bt_ext}-devel, openssl, elfutils-libelf-devel, python3
 %global BLUENIX %(if (grep -qiE "Bluenix" /etc/issue /etc/*release* 2>/dev/null); then echo -n '1'; else echo -n '0'; fi)
 %global XENSERVER65 %(if (grep -qiE "XenServer.*6\.5" /etc/issue /etc/*release* 2>/dev/null); then echo -n '1'; else echo -n '0'; fi)
 %global RHEL8 %(if test `grep -E '^(ID="rhel"|VERSION="8)' /etc/os-release 2>/dev/null | wc -l` -eq 2; then echo -n '1'; else echo -n '0'; fi)
+%global CENTOS8 %(if test `grep -E '^(ID="centos"|VERSION="8)' /etc/os-release 2>/dev/null | wc -l` -eq 2; then echo -n '1'; else echo -n '0'; fi)
 
 %{!?KVERSION: %global KVERSION %(rpm -q kernel%{?bt_ext}-devel | sort --version-sort | tail -1 | sed 's/kernel%{?bt_ext}-devel-//')}
 %global kernel_version %{KVERSION}
@@ -80,7 +81,7 @@ BuildRequires: kernel%{?bt_ext}-devel, openssl, elfutils-libelf-devel, python3
 %global devel_pname %{_name}-devel
 %global non_kmp_pname %{_name}-modules
 
-%if %{RHEL8}
+%if %{RHEL8} || %{CENTOS8}
 %global mlnx_python_env    export MLNX_PYTHON_EXECUTABLE=python3
 %else
 %global mlnx_python_env    :
@@ -283,7 +284,7 @@ The driver sources are located at: http://www.mellanox.com/downloads/ofed/mlnx-o
 set -- *
 mkdir source
 mv "$@" source/
-%if %{RHEL8}
+%if %{RHEL8} || %{CENTOS8}
 sed -s -i -e '1s|python\>|python3|' `grep -rl '^#!.*python' source/ofed_scripts`
 %endif
 mkdir obj
-- 
2.17.1

