From 3b29074ca269b8246a7f9bd5ad8468c1b7e54dc5 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 3 Jan 2017 13:45:43 -0500
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/puppet-stdlib.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-stdlib.spec b/SPECS/puppet-stdlib.spec
index baab15b..0261e5d 100644
--- a/SPECS/puppet-stdlib.spec
+++ b/SPECS/puppet-stdlib.spec
@@ -8,7 +8,7 @@
 
 Name:           puppet-stdlib
 Version:        4.13.1
-Release:        2%{?alphatag}%{?dist}
+Release:        2.dcef77agit.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Standard library of resources for Puppet modules.
 License:        Apache-2.0
 
-- 
1.8.3.1

