From beb4d7f0c2a6d19e863194ea8ec999f815557035 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 3 Jan 2017 13:45:43 -0500
Subject: [PATCH 1/2] Update package versioning for TIS format

---
 SPECS/puppet-stdlib.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-stdlib.spec b/SPECS/puppet-stdlib.spec
index 77b06f9..4947330 100644
--- a/SPECS/puppet-stdlib.spec
+++ b/SPECS/puppet-stdlib.spec
@@ -2,7 +2,7 @@
 %define upstream_name puppetlabs-stdlib
 Name:           puppet-stdlib
 Version:        4.18.0
-Release:        2%{?dist}
+Release:        2.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Standard library of resources for Puppet modules.
 License:        ASL 2.0
 
-- 
2.7.4

