From c7d61832d0ab7eb0d8b7656d0c1bc0eda036331b Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 1 May 2017 14:39:38 -0400
Subject: [PATCH 2/2] Add TIS Patches

---
 SPECS/puppet-stdlib.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/puppet-stdlib.spec b/SPECS/puppet-stdlib.spec
index 4947330..19e9599 100644
--- a/SPECS/puppet-stdlib.spec
+++ b/SPECS/puppet-stdlib.spec
@@ -10,6 +10,8 @@ URL:            https://github.com/puppetlabs/puppetlabs-stdlib
 
 Source0:        https://github.com/puppetlabs/%{upstream_name}/archive/%{version}.tar.gz
 
+Patch0001:      0001-Filter-password-in-logs.patch
+
 BuildArch:      noarch
 
 Requires:       puppet >= 2.7.0
@@ -19,6 +21,7 @@ Standard library of resources for Puppet modules.
 
 %prep
 %setup -q -n %{upstream_name}-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.7.4

