From 2f72d134c0c9a4fdc4e83f0aefab1608da644621 Mon Sep 17 00:00:00 2001
From: Daniel Badea <daniel.badea@windriver.com>
Date: Tue, 31 Oct 2017 15:44:43 +0200
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/puppet-glance.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/puppet-glance.spec b/puppet-glance.spec
index 1e9d4b8..64b1dbe 100644
--- a/SPECS/puppet-glance.spec
+++ b/SPECS/puppet-glance.spec
@@ -1,7 +1,7 @@
 %{!?upstream_version: %global upstream_version %{version}%{?milestone}}
 Name:           puppet-glance
 Version:        11.3.0
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Puppet module for OpenStack Glance
 License:        ASL 2.0
 
-- 
2.7.4

