From 534bf798c41db10847c2035a36e49f2d9d459a00 Mon Sep 17 00:00:00 2001
From: Daniel Badea <daniel.badea@windriver.com>
Date: Tue, 31 Oct 2017 17:51:50 +0200
Subject: [PATCH] Add TIS patch

---
 SPECS/puppet-glance.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/puppet-glance.spec b/puppet-glance.spec
index 64b1dbe..d6b63d1 100644
--- a/SPECS/puppet-glance.spec
+++ b/SPECS/puppet-glance.spec
@@ -8,6 +8,7 @@ License:        ASL 2.0
 URL:            https://launchpad.net/puppet-glance
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
+Patch0001:      0001-Roll-up-TIS-patches.patch
 
 BuildArch:      noarch
 
@@ -23,6 +24,7 @@ Puppet module for OpenStack Glance
 
 %prep
 %setup -q -n openstack-glance-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.7.4

