From 56feda10321fb1e5067b78dfeced8c5adcf8748e Mon Sep 17 00:00:00 2001
From: Stefan Dinescu <stefan.dinescu@windriver.com>
Date: Thu, 16 Nov 2017 18:28:36 +0000
Subject: [PATCH] Add region-cache support

---
 SPECS/puppet-glance.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-glance.spec b/SPECS/puppet-glance.spec
index d6b63d1..1e49ae0 100644
--- a/SPECS/puppet-glance.spec
+++ b/SPECS/puppet-glance.spec
@@ -9,6 +9,7 @@ URL:            https://launchpad.net/puppet-glance
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 Patch0001:      0001-Roll-up-TIS-patches.patch
+Patch0002:      0002-Add-region-cache-support.patch
 
 BuildArch:      noarch
 
@@ -25,6 +26,7 @@ Puppet module for OpenStack Glance
 %prep
 %setup -q -n openstack-glance-%{upstream_version}
 %patch0001 -p1
+%patch0002 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.7.4

