From 7621f6c41668b1ec8b8c76f834f486ee938cd77c Mon Sep 17 00:00:00 2001
From: Dostoievski Batista <dostoievski.albinobatista@windriver.com>
Date: Fri, 17 Mar 2023 17:55:16 -0300
Subject: [PATCH] Disabling auto test

Signed-off-by: Dostoievski Batista <dostoievski.albinobatista@windriver.com>
---
 debian/rules | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/debian/rules b/debian/rules
index 1d632f2..0836972 100755
--- a/debian/rules
+++ b/debian/rules
@@ -1539,3 +1539,6 @@ binary: binary-indep binary-arch
 # Local Variables:
 # mode: makefile
 # end:
+
+override_dh_auto_test:
+	dh_auto_test --nocheck
\ No newline at end of file
-- 
2.25.1
