From 662913fede85b09f54e5272451e4dc3f295a4e05 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Tue, 3 Jan 2017 15:55:20 -0500
Subject: [PATCH] WRS: spec-10_linux-mktitle-de-brand-the-grub.cfg-menu.patch

---
 SOURCES/grub.patches | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SOURCES/grub.patches b/SOURCES/grub.patches
index 8c7d946..fcd0a51 100644
--- a/SOURCES/grub.patches
+++ b/SOURCES/grub.patches
@@ -210,3 +210,4 @@ Patch0210: 0210-Fix-duplicate-shell-entries-1226325.patch
 Patch0211: 0211-Coverity-rightly-noticed-that-grub_ipv6_get_masksize.patch
 Patch0212: 0212-Export-grub-get-kernel-settings-variables.patch
 Patch0213: 0213-Normalize-slashes-in-tftp-paths.patch
+Patch1000: 1000_linux-mktitle-de-brand-the-grub.cfg-menu.patch
-- 
1.8.3.1

