From 2d237c31ea6d2fdfeaf61aef89b4755def0495a2 Mon Sep 17 00:00:00 2001
From: slin14 <shuicheng.lin@intel.com>
Date: Fri, 17 Aug 2018 23:28:59 +0800
Subject: [PATCH] remove-debug-pkgs

Signed-off-by: slin14 <shuicheng.lin@intel.com>
---
 SPECS/grub2.spec | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/SPECS/grub2.spec b/SPECS/grub2.spec
index 88c6c09..11f6b0e 100644
--- a/SPECS/grub2.spec
+++ b/SPECS/grub2.spec
@@ -172,7 +172,6 @@ rm -f grub-%{tarversion}/util/grub-setpassword.in.orig
 %install
 set -e
 rm -fr $RPM_BUILD_ROOT
-
 %do_common_install
 %if 0%{with_efi_arch}
 %do_efi_install %{grubefiarch} %{grubefiname} %{grubeficdname}
@@ -199,25 +198,6 @@ cat << EOF > ${RPM_BUILD_ROOT}%{_sysconfdir}/prelink.conf.d/grub2.conf
 -b /usr/sbin/grub2-sparc64-setup
 EOF
 
-# Don't run debuginfo on all the grub modules and whatnot; it just
-# rejects them, complains, and slows down extraction.
-%global finddebugroot "%{_builddir}/%{?buildsubdir}/debug"
-
-%global dip RPM_BUILD_ROOT=%{finddebugroot} %{__debug_install_post}
-%define __debug_install_post (						\
-	mkdir -p %{finddebugroot}/usr					\
-	mv ${RPM_BUILD_ROOT}/usr/bin %{finddebugroot}/usr/bin		\
-	mv ${RPM_BUILD_ROOT}/usr/sbin %{finddebugroot}/usr/sbin		\
-	%{dip}								\
-	install -m 0755 -d %{buildroot}/usr/lib/ %{buildroot}/usr/src/	\
-	cp -al %{finddebugroot}/usr/lib/debug/				\\\
-		%{buildroot}/usr/lib/debug/				\
-	cp -al %{finddebugroot}/usr/src/debug/				\\\
-		%{buildroot}/usr/src/debug/ )				\
-	mv %{finddebugroot}/usr/bin %{buildroot}/usr/bin		\
-	mv %{finddebugroot}/usr/sbin %{buildroot}/usr/sbin		\
-	%{nil}
-
 %clean    
 rm -rf $RPM_BUILD_ROOT
 
-- 
2.7.4

