diff --git a/SPECS/grub2.spec b/SPECS/grub2.spec
index 4059aea..a56d930 100644
--- a/SPECS/grub2.spec
+++ b/SPECS/grub2.spec
@@ -176,7 +176,6 @@ This subpackage provides tools for support of all platforms.
 %install
 set -e
 rm -fr $RPM_BUILD_ROOT
-
 %do_common_install
 # Fix for hardcoded efidir
 sed -i.orig -e 's@/efi/EFI/redhat/@/efi/EFI/%{efidir}/@' \
@@ -184,7 +183,6 @@ sed -i.orig -e 's@/efi/EFI/redhat/@/efi/EFI/%{efidir}/@' \
 touch --reference=grub-%{tarversion}/util/grub-setpassword.in.orig \
     grub-%{tarversion}/util/grub-setpassword.in
 rm -f grub-%{tarversion}/util/grub-setpassword.in.orig
-
 %if 0%{with_efi_arch}
 %do_efi_install %{grubefiarch} %{grubefiname} %{grubeficdname}
 %endif
@@ -210,25 +208,6 @@ cat << EOF > ${RPM_BUILD_ROOT}%{_sysconfdir}/prelink.conf.d/grub2.conf
 -b /usr/sbin/grub2-sparc64-setup
 EOF
 
-# Don't run debuginfo on all the grub modules and whatnot; it just
-# rejects them, complains, and slows down extraction.
-%global finddebugroot "%{_builddir}/%{?buildsubdir}/debug"
-
-%global dip RPM_BUILD_ROOT=%{finddebugroot} %{__debug_install_post}
-%define __debug_install_post (						\
-	mkdir -p %{finddebugroot}/usr					\
-	mv ${RPM_BUILD_ROOT}/usr/bin %{finddebugroot}/usr/bin		\
-	mv ${RPM_BUILD_ROOT}/usr/sbin %{finddebugroot}/usr/sbin		\
-	%{dip}								\
-	install -m 0755 -d %{buildroot}/usr/lib/ %{buildroot}/usr/src/	\
-	cp -al %{finddebugroot}/usr/lib/debug/				\\\
-		%{buildroot}/usr/lib/debug/				\
-	cp -al %{finddebugroot}/usr/src/debug/				\\\
-		%{buildroot}/usr/src/debug/ )				\
-	mv %{finddebugroot}/usr/bin %{buildroot}/usr/bin		\
-	mv %{finddebugroot}/usr/sbin %{buildroot}/usr/sbin		\
-	%{nil}
-
 %clean    
 rm -rf $RPM_BUILD_ROOT
 
