diff --git a/SPECS/grub2.spec b/SPECS/grub2.spec
index 9c32a93..676572e 100644
--- a/SPECS/grub2.spec
+++ b/SPECS/grub2.spec
@@ -140,6 +140,16 @@ bootloader with modular architecture.  It support rich varietyof kernel formats,
 file systems, computer architectures and hardware devices.  This subpackage
 provides tools for support of all platforms.
 
+%ifarch %{efiarchs}
+%ifnarch aarch64
+%package efi-unsigned
+Summary:    Unsigned versions of GRUB EFI binaries
+
+%description efi-unsigned
+This package contains unsigned version of GRUB EFI binaries.
+%endif # aarch64
+%endif # %{efiarchs}
+
 %prep
 %setup -T -c -n grub-%{tarversion}
 %ifarch %{efiarchs}
@@ -210,6 +220,10 @@ GRUB_MODULES="${GRUB_MODULES} linuxefi"
 		-d grub-core ${GRUB_MODULES}
 ./grub-mkimage -O %{grubefiarch} -o %{grubeficdname}.orig -p /EFI/BOOT \
 		-d grub-core ${GRUB_MODULES}
+
+cp %{grubefiname}.orig %{grubefiname}.unsigned
+cp %{grubeficdname}.orig %{grubeficdname}.unsigned
+
 %ifarch aarch64
 mv %{grubefiname}.orig %{grubefiname}
 mv %{grubeficdname}.orig %{grubeficdname}
@@ -292,6 +306,12 @@ do
         TGT=$(echo $MODULE |sed "s,$RPM_BUILD_ROOT,.debugroot,")
 #        install -m 755 -D $BASE$EXT $TGT
 done
+%ifarch %{efiarchs}
+%ifnarch aarch64
+install -m 755 %{grubefiname}.unsigned $RPM_BUILD_ROOT/boot/efi/EFI/%{efidir}/%{grubefiname}.unsigned
+install -m 755 %{grubeficdname}.unsigned $RPM_BUILD_ROOT/boot/efi/EFI/%{efidir}/%{grubeficdname}.unsigned
+%endif
+%endif
 install -m 755 %{grubefiname} $RPM_BUILD_ROOT/boot/efi/EFI/%{efidir}/%{grubefiname}
 install -m 755 %{grubeficdname} $RPM_BUILD_ROOT/boot/efi/EFI/%{efidir}/%{grubeficdname}
 install -D -m 644 unicode.pf2 $RPM_BUILD_ROOT/boot/efi/EFI/%{efidir}/fonts/unicode.pf2
@@ -444,7 +464,9 @@ fi
 %files efi
 %defattr(-,root,root,-)
 %config(noreplace) %{_sysconfdir}/%{name}-efi.cfg
-%attr(0755,root,root)/boot/efi/EFI/%{efidir}
+%attr(0755,root,root) %dir /boot/efi/EFI/%{efidir}
+%attr(0755,root,root)/boot/efi/EFI/%{efidir}/%{grubefiname}
+%attr(0755,root,root)/boot/efi/EFI/%{efidir}/%{grubeficdname}
 %attr(0755,root,root)/boot/efi/EFI/%{efidir}/fonts
 %ghost %config(noreplace) /boot/efi/EFI/%{efidir}/grub.cfg
 %doc grub-%{tarversion}/COPYING
@@ -520,6 +542,14 @@ fi
 %dir %{_datarootdir}/grub/themes
 %exclude %{_datarootdir}/grub/themes/starfield
 
+%ifarch %{efiarchs}
+%ifnarch aarch64
+%files efi-unsigned
+/boot/efi/EFI/%{efidir}/%{grubefiname}.unsigned
+/boot/efi/EFI/%{efidir}/%{grubeficdname}.unsigned
+%endif
+%endif
+
 %changelog
 * Sun Nov 20 2016 Johnny Hughes <johnny@centos.org> - 2.02-0.44
 - Roll in CentOS Secureboot keys

