diff --git a/SPECS/grub2.spec b/SPECS/grub2.spec
index 11f6b0e..613f2e1 100644
--- a/SPECS/grub2.spec
+++ b/SPECS/grub2.spec
@@ -49,11 +49,6 @@ BuildRequires:  /usr/lib64/crt1.o glibc-static glibc-devel
 BuildRequires:  /usr/lib64/crt1.o glibc-static(x86-64) glibc-devel(x86-64)
 # glibc32 is what will be in the buildroots, but glibc-static(x86-32) is what
 # will be in an epel-7 (i.e. centos) mock root.  I think.
-%if 0%{?centos}%{?mock}
-BuildRequires:  /usr/lib/crt1.o glibc-static(x86-32) glibc-devel(x86-32)
-%else
-BuildRequires:  /usr/lib/crt1.o glibc32
-%endif
 %else
 # ppc64 builds need the ppc crt1.o
 BuildRequires:  /usr/lib/crt1.o glibc-static glibc-devel
