diff --git a/SPECS/grub2.spec b/SPECS/grub2.spec
index 53778b3..4829e91 100644
--- a/SPECS/grub2.spec
+++ b/SPECS/grub2.spec
@@ -149,6 +149,14 @@ Summary:    Unsigned versions of GRUB EFI binaries
 
 %description efi-unsigned
 This package contains unsigned version of GRUB EFI binaries.
+
+%package efi-pxeboot
+Summary:    PXE bootable GRUB EFI binaries
+
+%description efi-pxeboot
+This package contains the version of EFI GRUB that is served by the pxeboot
+server
+
 %endif # aarch64
 %endif # %{efiarchs}
 
@@ -312,6 +320,7 @@ done
 %ifnarch aarch64
 install -m 755 %{grubefiname}.unsigned $RPM_BUILD_ROOT/boot/efi/EFI/%{efidir}/%{grubefiname}.unsigned
 install -m 755 %{grubeficdname}.unsigned $RPM_BUILD_ROOT/boot/efi/EFI/%{efidir}/%{grubeficdname}.unsigned
+install -D -m 755 %{grubefiname} $RPM_BUILD_ROOT/pxeboot/EFI/%{grubefiname}
 %endif
 %endif
 install -m 755 %{grubefiname} $RPM_BUILD_ROOT/boot/efi/EFI/%{efidir}/%{grubefiname}
@@ -546,9 +555,14 @@ fi
 
 %ifarch %{efiarchs}
 %ifnarch aarch64
+
 %files efi-unsigned
 /boot/efi/EFI/%{efidir}/%{grubefiname}.unsigned
 /boot/efi/EFI/%{efidir}/%{grubeficdname}.unsigned
+
+%files efi-pxeboot
+/pxeboot/EFI/%{grubefiname}
+
 %endif
 %endif
 
