From 432caafd03704396941acec4ff059390ad749719 Mon Sep 17 00:00:00 2001
From: root <root@yow-cgts4-lx.wrs.com>
Date: Mon, 22 Jan 2018 16:05:57 -0500
Subject: macros

---
 SOURCES/grub.macros | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/SOURCES/grub.macros b/SOURCES/grub.macros
index d59f889..ed426f3 100644
--- a/SOURCES/grub.macros
+++ b/SOURCES/grub.macros
@@ -122,12 +122,6 @@
 %global legacy_package_arch pc
 %global platform pc
 
-%global alt_efi_arch ia32
-%global alt_target_cpu_name i386
-%global alt_grub_target_name i386-efi
-%global alt_platform efi
-%global alt_package_arch efi-ia32
-
 %global alt_efi_host_cflags %{expand:%%(echo %{efi_host_cflags})}
 %global alt_efi_target_cflags					\\\
 	%{expand:%%(echo %{target_cflags} |			\\\
@@ -439,6 +433,7 @@ cd ..								\
 %define do_efi_install()					\
 cd grub-%{1}-%{tarversion}					\
 make DESTDIR=$RPM_BUILD_ROOT install				\
+find $RPM_BUILD_ROOT -name "grub2-bios-setup*" | xargs rm -f    \
 if [ -f $RPM_BUILD_ROOT%{_infodir}/grub.info ]; then		\
 	rm -f $RPM_BUILD_ROOT%{_infodir}/grub.info		\
 fi								\
-- 
2.7.4

