From 69f56affb4e910470072b79824f3c6ecccefa4c2 Mon Sep 17 00:00:00 2001
From: Bin Qian <bin.qian@windriver.com>
Date: Wed, 6 Dec 2017 10:45:42 -0500
Subject: [PATCH 1/1] WRS: Add tboot support

---
 SPECS/grub2.spec | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/SPECS/grub2.spec b/SPECS/grub2.spec
index 4d92bf9..5719df6 100644
--- a/SPECS/grub2.spec
+++ b/SPECS/grub2.spec
@@ -65,6 +65,7 @@ Source6:	gitignore
 %include %{SOURCE3}
 
 Patch9999: 0001-centos-grub2-password-fix.patch
+Patch10001: 10001-add-tboot.patch
 
 BuildRequires:  flex bison binutils python
 BuildRequires:  ncurses-devel xz-devel bzip2-devel
@@ -119,7 +120,7 @@ provides support for EFI systems.
 Summary:	Modules used to build custom grub.efi images
 Group:		System Environment/Base
 Requires:	%{name}-tools = %{epoch}:%{version}-%{release}
-Obsoletes:	grub2-efi <= 1:2.00-20%{?dist}
+# Obsoletes:	grub2-efi <= 1:2.00-20%{?dist}
 
 %description efi-modules
 The GRand Unified Bootloader (GRUB) is a highly configurable and customizable
@@ -219,6 +220,10 @@ GRUB_MODULES="	all_video boot btrfs cat chain configfile echo efifwsetup \
 		part_apple part_msdos part_gpt password_pbkdf2 png \
 		reboot search search_fs_uuid search_fs_file search_label \
 		sleep syslinuxcfg test tftp regexp video xfs"
+
+# WRS: Adding new modules for tboot support
+GRUB_MODULES="${GRUB_MODULES} multiboot multiboot2"
+
 %ifarch aarch64
 GRUB_MODULES="${GRUB_MODULES} linux"
 %else
-- 
1.8.3.1

