From 432caafd03704396941acec4ff059390ad749719 Mon Sep 17 00:00:00 2001
From: root <root@yow-cgts4-lx.wrs.com>
Date: Mon, 22 Jan 2018 16:05:57 -0500
Subject: macros

---
 SOURCES/grub.macros | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/SOURCES/grub.macros b/SOURCES/grub.macros
index 10f74df..075727c 100644
--- a/SOURCES/grub.macros
+++ b/SOURCES/grub.macros
@@ -82,15 +82,6 @@
 %global legacy_package_arch pc
 %global platform pc
 
-%global alt_efi_arch ia32
-%global alt_target_cpu_name i386
-%global alt_grub_target_name i386-efi
-%global alt_platform efi
-%global alt_package_arch efi-ia32
-%global alt_efi_cflags %(echo %{efi_cflags} |			\\\
-	sed							\\\
-		-e 's/-m64//g'				\\\
-	) %{nil}
 %endif
 %ifarch %{ix86}
 %global target_cpu_name %{_arch}
@@ -401,6 +392,7 @@ cd grub-%{1}-%{tarversion}					\
 install -m 700 -d $RPM_BUILD_ROOT/boot/efi/EFI/%{efidir}/	\
 install -m 700 -d $RPM_BUILD_ROOT/boot/grub2/			\
 make DESTDIR=$RPM_BUILD_ROOT install				\
+find $RPM_BUILD_ROOT -name "grub2-bios-setup*" | xargs rm -f \
 if [ -f $RPM_BUILD_ROOT%{_infodir}/grub.info ]; then		\
 	rm -f $RPM_BUILD_ROOT%{_infodir}/grub.info		\
 fi								\
-- 
2.7.4

