From 8fb898476d6a86b2b280bd63f1557dfe4d1360a4 Mon Sep 17 00:00:00 2001
From: root <root@yow-cgts4-lx.wrs.com>
Date: Tue, 23 Jan 2018 09:24:16 -0500
Subject: [PATCH] ship .lst files

---
 SPECS/grub2.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/grub2.spec b/SPECS/grub2.spec
index 613f2e1..7e11008 100644
--- a/SPECS/grub2.spec
+++ b/SPECS/grub2.spec
@@ -299,6 +299,8 @@ fi
 %ifarch x86_64
 %files common -f grub.lang
 %dir %{_libdir}/grub/
+%dir %{_libdir}/grub/%{grubefiarch}/
+%{_libdir}/grub/%{grubefiarch}/*.lst
 %dir %{_datarootdir}/grub/
 %dir %{_datarootdir}/grub/themes/
 %exclude %{_datarootdir}/grub/themes/*
-- 
2.7.4

