From 9056b35661962a311aafa1937b5d8e236f44b119 Mon Sep 17 00:00:00 2001
From: root <root@yow-cgts4-lx.wrs.com>
Date: Tue, 23 Jan 2018 13:29:11 -0500
Subject: [PATCH] debrand the grub_cfg menu

---
 SOURCES/grub.patches | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SOURCES/grub.patches b/SOURCES/grub.patches
index bac4594..d7475f0 100644
--- a/SOURCES/grub.patches
+++ b/SOURCES/grub.patches
@@ -286,3 +286,4 @@ Patch0285: 0285-editenv-handle-relative-symlinks.patch
 Patch0286: 0286-efinet-also-use-the-firmware-acceleration-for-http.patch
 Patch0287: 0287-Make-root_url-reflect-the-protocol-hostname-of-our-b.patch
 Patch0289: 0288-efi-uga-Fix-PCIe-LER-when-GRUB2-accesses-non-enabled.patch
+Patch1000: 1000_linux-mktitle-de-brand-the-grub.cfg-menu.patch
-- 
2.7.4

