From fadbfe260a2678e1a6e79f8df4372b2eaee5dc9f Mon Sep 17 00:00:00 2001
From: root <root@yow-cgts4-lx.wrs.com>
Date: Tue, 23 Jan 2018 14:46:01 -0500
Subject: add unsigned package

---
 SOURCES/grub.macros | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/SOURCES/grub.macros b/SOURCES/grub.macros
index ed426f3..0cbfa4e 100644
--- a/SOURCES/grub.macros
+++ b/SOURCES/grub.macros
@@ -278,6 +278,13 @@ Provides:	%{name}-efi-cdboot = %{evr}				\
 %{expand:%%description %{1}-cdboot}					\
 %{desc}									\
 This subpackage provides optional components of grub used with removeable media on %{1} systems.\
+                                                                        \
+%package %{1}-unsigned                                                  \
+Summary:    Unsigned versions of GRUB EFI binaries                      \
+                                                                        \
+%description %{1}-unsigned                                              \
+This package contains unsigned version of GRUB EFI binaries.            \
+                                                                        \
 %{nil}
 
 %global do_common_setup()					\
@@ -345,6 +352,8 @@ done								\
 	-p /EFI/%{efi_vendor} -d grub-core ${GRUB_MODULES}	\
 %{4}./grub-mkimage -O %{1} -o %{3}.orig				\\\
 	-p /EFI/BOOT -d grub-core ${GRUB_MODULES}		\
+cp %{2}.orig %{2}.unsigned                                      \
+cp %{3}.orig %{3}.unsigned                                      \
 %{expand:%%{pesign -s -i %%{2}.orig -o %%{2} -a %%{5} -c %%{6} -n %%{7}}}	\
 %{expand:%%{pesign -s -i %%{3}.orig -o %%{3} -a %%{5} -c %%{6} -n %%{7}}}	\
 %{nil}
@@ -444,6 +453,8 @@ find $RPM_BUILD_ROOT -iname "*.module" -exec chmod a-x {} '\;'	\
 touch $RPM_BUILD_ROOT%{efi_esp_dir}/grub.cfg			\
 ln -sf ..%{efi_esp_dir}/grub.cfg				\\\
 	$RPM_BUILD_ROOT%{_sysconfdir}/%{name}-efi.cfg		\
+install -m 700 %{2} $RPM_BUILD_ROOT%{efi_esp_dir}/%{2}.unsigned \
+install -m 700 %{3} $RPM_BUILD_ROOT%{efi_esp_dir}/%{3}.unsigned \
 install -m 700 %{2} $RPM_BUILD_ROOT%{efi_esp_dir}/%{2}		\
 install -m 700 %{3} $RPM_BUILD_ROOT%{efi_esp_dir}/%{3}		\
 install -D -m 700 unicode.pf2					\\\
@@ -546,4 +557,8 @@ touch ${RPM_BUILD_ROOT}/boot/%{name}/grub.cfg			\
 %defattr(0700,root,root,-)					\
 %attr(0700,root,root)%{efi_esp_dir}/%{3}			\
 %attr(0700,root,root)%{efi_esp_dir}/fonts			\
+                                                                \
+%{expand:%%files %{1}-unsigned}                                 \
+%{efi_esp_dir}/%{grubefiname}.unsigned                          \
+%{efi_esp_dir}/%{grubeficdname}.unsigned                        \
 %{nil}
-- 
2.7.4

