From ca7e4edddd48b0ffd0520a2ccb271bbe036aa1e5 Mon Sep 17 00:00:00 2001
From: root <root@yow-cgts4-lx.wrs.com>
Date: Tue, 23 Jan 2018 14:57:29 -0500
Subject: add pxeboot package

---
 SOURCES/grub.macros | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/SOURCES/grub.macros b/SOURCES/grub.macros
index 0cbfa4e..60bb1b3 100644
--- a/SOURCES/grub.macros
+++ b/SOURCES/grub.macros
@@ -285,6 +285,13 @@ Summary:    Unsigned versions of GRUB EFI binaries                      \
 %description %{1}-unsigned                                              \
 This package contains unsigned version of GRUB EFI binaries.            \
                                                                         \
+%package %{1}-pxeboot                                                   \
+Summary:    PXE bootable GRUB EFI binaries                              \
+                                                                        \
+%description %{1}-pxeboot                                               \
+This package contains the version of EFI GRUB that is served by the pxeboot \
+server                                                                  \
+                                                                        \
 %{nil}
 
 %global do_common_setup()					\
@@ -455,6 +462,7 @@ ln -sf ..%{efi_esp_dir}/grub.cfg				\\\
 	$RPM_BUILD_ROOT%{_sysconfdir}/%{name}-efi.cfg		\
 install -m 700 %{2} $RPM_BUILD_ROOT%{efi_esp_dir}/%{2}.unsigned \
 install -m 700 %{3} $RPM_BUILD_ROOT%{efi_esp_dir}/%{3}.unsigned \
+install -D -m 755 %{2} $RPM_BUILD_ROOT/pxeboot/EFI/%{2}         \
 install -m 700 %{2} $RPM_BUILD_ROOT%{efi_esp_dir}/%{2}		\
 install -m 700 %{3} $RPM_BUILD_ROOT%{efi_esp_dir}/%{3}		\
 install -D -m 700 unicode.pf2					\\\
@@ -561,4 +569,8 @@ touch ${RPM_BUILD_ROOT}/boot/%{name}/grub.cfg			\
 %{expand:%%files %{1}-unsigned}                                 \
 %{efi_esp_dir}/%{grubefiname}.unsigned                          \
 %{efi_esp_dir}/%{grubeficdname}.unsigned                        \
+								\
+%{expand:%%files %{1}-pxeboot}                                  \
+/pxeboot/EFI/%{grubefiname}                                     \
+								\
 %{nil}
-- 
2.7.4

