From 25964e485c769efcccf853babcc8263eb1c89baf Mon Sep 17 00:00:00 2001
From: jmckenna <jason.mckenna@windriver.com>
Date: Wed, 28 Mar 2018 12:53:34 -0400
Subject: WRS: 0010-grub2-add-tboot.patch

---
 SOURCES/grub.macros  | 5 +++++
 SOURCES/grub.patches | 1 +
 2 files changed, 6 insertions(+)

diff --git a/SOURCES/grub.macros b/SOURCES/grub.macros
index 9ef91d6..ffdd23c 100644
--- a/SOURCES/grub.macros
+++ b/SOURCES/grub.macros
@@ -82,6 +82,10 @@
 %global legacy_package_arch pc
 %global platform pc
 
+%global wrs_modules " multiboot multiboot2"
+%else
+%global wrs_modules ""
+
 %endif
 %ifarch %{ix86}
 %global target_cpu_name %{_arch}
@@ -347,6 +351,7 @@ GRUB_MODULES="	all_video boot btrfs cat chain configfile echo	\\\
 		search_label serial sleep syslinuxcfg test tftp \\\
 		video xfs"					\
 GRUB_MODULES+=%{efi_modules}					\
+GRUB_MODULES+=%{wrs_modules}                                    \
 %{expand:%%{mkimage %{1} %{2} %{3} %{4} %{5} %{6} %{7}}}	\
 %{nil}
 
diff --git a/SOURCES/grub.patches b/SOURCES/grub.patches
index d7475f0..e24bd8c 100644
--- a/SOURCES/grub.patches
+++ b/SOURCES/grub.patches
@@ -287,3 +287,4 @@ Patch0286: 0286-efinet-also-use-the-firmware-acceleration-for-http.patch
 Patch0287: 0287-Make-root_url-reflect-the-protocol-hostname-of-our-b.patch
 Patch0289: 0288-efi-uga-Fix-PCIe-LER-when-GRUB2-accesses-non-enabled.patch
 Patch1000: 1000_linux-mktitle-de-brand-the-grub.cfg-menu.patch
+Patch1001: 1001-add-tboot.patch
-- 
2.7.4

