From 31310ca5b5d10aff6bd4480bc78ef75504711ab1 Mon Sep 17 00:00:00 2001
From: root <root@yow-cgts1-lx.wrs.com>
Date: Fri, 6 Apr 2018 11:36:01 -0400
Subject: Don't strip files from non-EFI build of grub

---
 SOURCES/grub.macros | 2 --
 1 file changed, 2 deletions(-)

diff --git a/SOURCES/grub.macros b/SOURCES/grub.macros
index 8fcb272..cd2da06 100644
--- a/SOURCES/grub.macros
+++ b/SOURCES/grub.macros
@@ -488,8 +488,6 @@ cd ..								\
 %defattr(-,root,root)						\
 %dir %{_libdir}/grub/%{2}/					\
 %{_libdir}/grub/%{2}/*						\
-%exclude %{_libdir}/grub/%{2}/*.module				\
-%exclude %{_libdir}/grub/%{2}/{boot,boot_hybrid,cdboot,diskboot,lzma_decompress,pxeboot}.{image,img}						\
 %exclude %{_libdir}/grub/%{2}/*.o				\
 %{nil}
 
-- 
2.7.4

