From 3d135a9e55dd497ab6d19f67078a85e0ef183fd7 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 26 Sep 2016 17:14:46 -0400
Subject: [PATCH 1/1] Update package versioning for TIS format

---
 SPECS/libmlx4.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/libmlx4.spec b/SPECS/libmlx4.spec
index fc5101d..c1eb78d 100644
--- a/SPECS/libmlx4.spec
+++ b/SPECS/libmlx4.spec
@@ -9,7 +9,7 @@
 
 Name: libmlx4
 Version: 41mlnx1
-Release: OFED.4.1.0.1.0.42120
+Release: OFED.4.1.0.1.0.42120%{?_tis_dist}.%{tis_patch_ver}
 Summary: Mellanox ConnectX InfiniBand HCA Userspace Driver
 
 Group: System Environment/Libraries
-- 
1.8.3.1

