From 3c4f2fef7d1d321a34a9777946879f187fdad2bc Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:23:24 -0400
Subject: [PATCH 2/2] WRS: 0002-grubby-Generic-name-for-Titanium.patch

---
 SPECS/grubby.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/grubby.spec b/SPECS/grubby.spec
index 22bff31..882339a 100644
--- a/SPECS/grubby.spec
+++ b/SPECS/grubby.spec
@@ -83,6 +83,9 @@ Patch0073: 0073-Fix-info-for-s390x-s390-1285601.patch
 Patch0074: 0074-Add-s390-s390x-set-default-index-test-1285601.patch
 Patch0075: 0075-Fix-setDefaultImage-for-s390-s390x-1285601.patch
 
+# WRS Titanium patches
+Patch1000: 1000-Generic-name-for-Titanium.patch
+
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 BuildRequires: pkgconfig glib2-devel popt-devel 
 BuildRequires: libblkid-devel git
-- 
1.9.1

