From 3c4f2fef7d1d321a34a9777946879f187fdad2bc Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:23:24 -0400
Subject: [PATCH 2/2] 0002-grubby-Generic-name-for-Titanium.patch

---
 SPECS/grubby.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/grubby.spec b/SPECS/grubby.spec
index a42642f..28de109 100644
--- a/SPECS/grubby.spec
+++ b/SPECS/grubby.spec
@@ -70,6 +70,8 @@ Patch0054: 0054-Make-installkernel-to-use-kernel-install-scripts-on-.patch
 Patch0055: 0055-Add-usr-libexec-rpm-sort.patch
 Patch0056: 0056-Improve-man-page-for-info-option.patch
 
+Patch1000: 1000-Generic-name-for-Titanium.patch
+
 BuildRequires: gcc
 BuildRequires: pkgconfig glib2-devel popt-devel 
 BuildRequires: libblkid-devel git-core sed make
-- 
2.7.4

