From 40211dceb45803c4bb4ff2afbc993f230872ad51 Mon Sep 17 00:00:00 2001
From: Bin Qian <bin.qian@windriver.com>
Date: Tue, 13 Feb 2018 23:02:49 -0500
Subject: [PATCH 1/1] grubby add multiboot2

---
 SPECS/grubby.spec | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/SPECS/grubby.spec b/SPECS/grubby.spec
index 28de109..e7bee82 100644
--- a/SPECS/grubby.spec
+++ b/SPECS/grubby.spec
@@ -71,6 +71,7 @@ Patch0055: 0055-Add-usr-libexec-rpm-sort.patch
 Patch0056: 0056-Improve-man-page-for-info-option.patch
 
 Patch1000: 1000-Generic-name-for-Titanium.patch
+Patch1001: 1001-Add-support-for-updating-grub-cfg-with-multiboot-2.patch
 
 BuildRequires: gcc
 BuildRequires: pkgconfig glib2-devel popt-devel 
@@ -88,6 +89,11 @@ Requires: s390utils-base
 %endif
 Requires: findutils
 
+BuildRequires: python-setuptools
+Requires: python-devel
+
+%define pythonroot  /usr/lib64/python2.7/site-packages
+
 %description
 This package provides a grubby compatibility script that manages
 BootLoaderSpec files and is meant to only be used for legacy compatibility
@@ -128,6 +134,11 @@ sed -e "s,@@LIBEXECDIR@@,%{_libexecdir}/grubby,g" %{SOURCE2} \
 sed -e "s,@@LIBEXECDIR@@,%{_libexecdir}/installkernel,g" %{SOURCE3} \
 	> %{buildroot}%{_sbindir}/installkernel
 
+install -d -m 0755 $RPM_BUILD_ROOT/%{pythonroot}/grubby
+install -m 644 grub_cfg_update.py $RPM_BUILD_ROOT/%{pythonroot}/grubby/grub_cfg_update.py
+install -m 644 __init__.py $RPM_BUILD_ROOT/%{pythonroot}/grubby/__init__.py
+install -m 500 grub-cfg-update $RPM_BUILD_ROOT/usr/sbin/grub-cfg-update
+
 %package deprecated
 Summary: Legacy command line tool for updating bootloader configs
 Conflicts:	%{name} <= 8.40-13
@@ -164,6 +175,11 @@ current boot environment.
 %attr(0755,root,root) %{_sbindir}/grubby
 %attr(0755,root,root) %{_sbindir}/installkernel
 %attr(0755,root,root) %{_sbindir}/new-kernel-pkg
+%{pythonroot}/grubby
+%{_prefix}/sbin/grub-cfg-update
+%{pythonroot}/grubby/grub_cfg_update.py
+%{pythonroot}/grubby/__init__.py
+
 %{_mandir}/man8/*.8*
 
 %changelog
-- 
2.7.4

