From 40211dceb45803c4bb4ff2afbc993f230872ad51 Mon Sep 17 00:00:00 2001
From: Bin Qian <bin.qian@windriver.com>
Date: Tue, 13 Feb 2018 23:02:49 -0500
Subject: [PATCH 1/1] grubby add multiboot2

---
 SPECS/grubby.spec | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/SPECS/grubby.spec b/SPECS/grubby.spec
index 071463a..a3bb371 100644
--- a/SPECS/grubby.spec
+++ b/SPECS/grubby.spec
@@ -85,6 +85,7 @@ Patch0075: 0075-Fix-setDefaultImage-for-s390-s390x-1285601.patch
 
 # WRS Titanium patches
 Patch1000: 1000-Generic-name-for-Titanium.patch
+Patch1001: 1001-Add-support-for-updating-grub-cfg-with-multiboot-2.patch 
 
 BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 BuildRequires: pkgconfig glib2-devel popt-devel 
@@ -101,6 +102,10 @@ Requires: s390utils-base
 Requires: uboot-tools
 %endif
 Requires: system-release
+BuildRequires: python-setuptools
+Requires: python-devel
+
+%define pythonroot  /usr/lib64/python2.7/site-packages
 
 %description
 grubby  is  a command line tool for updating and displaying information about 
@@ -140,6 +145,11 @@ mkdir -p $RPM_BUILD_ROOT/boot
 echo " " >> $RPM_BUILD_ROOT/boot/boot.scr
 %endif
 
+install -d -m 0755 $RPM_BUILD_ROOT/%{pythonroot}/grubby
+install -m 644 grub_cfg_update.py $RPM_BUILD_ROOT/%{pythonroot}/grubby/grub_cfg_update.py
+install -m 644 __init__.py $RPM_BUILD_ROOT/%{pythonroot}/grubby/__init__.py
+install -m 500 grub-cfg-update $RPM_BUILD_ROOT/usr/sbin/grub-cfg-update
+
 %clean
 rm -rf $RPM_BUILD_ROOT
 
@@ -152,6 +162,11 @@ rm -rf $RPM_BUILD_ROOT
 %{_prefix}/sbin/installkernel
 %{_prefix}/sbin/new-kernel-pkg
 %{_prefix}/sbin/grubby
+%{pythonroot}/grubby
+%{_prefix}/sbin/grub-cfg-update
+%{pythonroot}/grubby/grub_cfg_update.py
+%{pythonroot}/grubby/__init__.py
+
 %{_mandir}/man8/*.8*
 /usr/libexec/grubby/prune_debug
 %ghost %config(noreplace) %{_sysconfdir}/sysconfig/kernel
-- 
1.8.3.1

