From 1f430bb5fbae2d5135ad4e547de3ac30f32b299b Mon Sep 17 00:00:00 2001
Message-Id: <1f430bb5fbae2d5135ad4e547de3ac30f32b299b.1502820262.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Tue, 15 Aug 2017 14:04:06 -0400
Subject: [PATCH 1/1] Update package versioning for TIS format

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 SPECS/libvirt-python.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/libvirt-python.spec b/SPECS/libvirt-python.spec
index c2ad17f..6108b7e 100644
--- a/SPECS/libvirt-python.spec
+++ b/SPECS/libvirt-python.spec
@@ -7,7 +7,7 @@
 Summary: The libvirt virtualization API python2 binding
 Name: libvirt-python
 Version: 3.5.0
-Release: 1%{?dist}%{?extra_release}
+Release: 1%{?_tis_dist}.%{tis_patch_ver}
 Source0: http://libvirt.org/sources/python/%{name}-%{version}.tar.gz
 Url: http://libvirt.org
 License: LGPLv2+
--
1.8.3.1


