From 73815bebec5f3e65dbe4b7bcd7270604ca0da266 Mon Sep 17 00:00:00 2001
From: Saju Oommen <saju.oommen@windriver.com>
Date: Mon, 15 Jan 2018 14:19:08 -0500
Subject: remove-custom-nslcd-conf-file

---
 SPECS/nss-pam-ldapd.spec | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/SPECS/nss-pam-ldapd.spec b/SPECS/nss-pam-ldapd.spec
index 18e6482..6313d0f 100644
--- a/SPECS/nss-pam-ldapd.spec
+++ b/SPECS/nss-pam-ldapd.spec
@@ -49,7 +49,6 @@ Source1:	http://arthurdejong.org/nss-pam-ldapd/nss-pam-ldapd-%{version}.tar.gz.s
 Source2:	nslcd.init
 Source3:	nslcd.tmpfiles
 Source4:	nslcd.service
-Source5:	nslcd.conf
 Patch1:		nss-pam-ldapd-0.8.12-validname.patch
 Patch2:         nss-pam-ldapd-0.8.12-In-nslcd-log-EPIPE-only-on-debug-level.patch
 Patch3:		nss-pam-ldapd-0.8.12-uid-overflow.patch
@@ -179,9 +178,6 @@ mkdir -p -m 0755 $RPM_BUILD_ROOT/%{_tmpfilesdir}
 install -p -m 0644 %{SOURCE3} $RPM_BUILD_ROOT/%{_tmpfilesdir}/%{name}.conf
 %endif
 
-# WRS
-install -m 644 %{SOURCE5} $RPM_BUILD_ROOT/%{_sysconfdir}/nslcd.conf
-
 %clean
 rm -rf $RPM_BUILD_ROOT
 
-- 
2.7.4

