From 87b414eb890b8314e79cd41e50d30c560da1c413 Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Fri, 27 Oct 2017 14:44:17 -0400
Subject: [PATCH 2/4] upstreamable: fix syntax error for until_complete_real

---
 manifests/cron/archive_deleted_rows.pp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/manifests/cron/archive_deleted_rows.pp b/manifests/cron/archive_deleted_rows.pp
index 8cd3e45..8446339 100644
--- a/manifests/cron/archive_deleted_rows.pp
+++ b/manifests/cron/archive_deleted_rows.pp
@@ -67,6 +67,7 @@ class nova::cron::archive_deleted_rows (
 
   include ::nova::deps
 
+  $until_complete_real = ''
   if $until_complete {
     $until_complete_real = '--until_complete'
   }
-- 
1.8.3.1

