---
 lib/runtime |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/runtime
+++ b/lib/runtime
@@ -863,7 +863,7 @@ fi
 # Log command
 if [ "$LOGTOFILE" = "yes" ]
 then
-  log_to_file "$(date '+%b %d %H:%M:%S') $(uname -n | sed 's|\..*$||') ldapscripts: $(basename "$0")($USER): $0 $*"
+  log_to_file "$(date '+%FT%T') $(uname -n | sed 's|\..*$||') ldapscripts: $(basename "$0")($USER): $0 $*"
 fi
 if [ "$LOGTOSYSLOG" = "yes" ]
 then
