From 02fb08dbc67712c70f65f0098f73a72820a7182d Mon Sep 17 00:00:00 2001
From: Angie Wang <Angie.Wang@windriver.com>
Date: Mon, 29 Jan 2018 13:24:18 -0500
Subject: [PATCH 1/1] Fix panko puppet warning

---
 manifests/api.pp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/manifests/api.pp b/manifests/api.pp
index bd72fea..7f6620c 100644
--- a/manifests/api.pp
+++ b/manifests/api.pp
@@ -89,7 +89,7 @@ class panko::api (
   $es_index_name                = $::os_service_default,
 ) inherits panko::params {
 
-  warning('This Class is deprecated and will be removed in future releases.')
+  # warning('This Class is deprecated and will be removed in future releases.')
 
   include ::panko::deps
   include ::panko::policy
-- 
1.8.3.1

