From 6098a4fc06f27ce855fe777afb50b13e23f11e27 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 3 Jan 2017 13:44:33 -0500
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/puppet-ceilometer.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-ceilometer.spec b/SPECS/puppet-ceilometer.spec
index 76697c4..6b3a1b4 100644
--- a/SPECS/puppet-ceilometer.spec
+++ b/SPECS/puppet-ceilometer.spec
@@ -1,7 +1,7 @@
 %{!?upstream_version: %global upstream_version %{version}%{?milestone}}
 Name:           puppet-ceilometer
 Version:        11.3.0
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Puppet module for OpenStack Ceilometer
 License:        ASL 2.0
 
-- 
1.8.3.1

