From e2c17352768be4f66a5769876e78d0ba28c8abad Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Tue, 14 Nov 2017 09:51:47 -0600
Subject: [PATCH] Add Rebase Patches

---
 SPECS/puppet-ceilometer.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/puppet-ceilometer.spec b/SPECS/puppet-ceilometer.spec
index 5f93f86..a869999 100644
--- a/SPECS/puppet-ceilometer.spec
+++ b/SPECS/puppet-ceilometer.spec
@@ -9,6 +9,8 @@ URL:            https://launchpad.net/puppet-ceilometer
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 
+Patch0001:      0001-Roll-up-TIS-patches.patch
+
 BuildArch:      noarch
 
 Requires:       puppet-apache
@@ -24,6 +26,7 @@ Puppet module for OpenStack Ceilometer
 
 %prep
 %setup -q -n openstack-ceilometer-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

