From 5ecbc776a6fdbd7065b6145c7fd82c03217b8205 Mon Sep 17 00:00:00 2001
From: Angie Wang <Angie.Wang@windriver.com>
Date: Mon, 29 Jan 2018 14:20:19 -0500
Subject: [PATCH 1/1] meta fix ceilometer puppet warnings

---
 SPECS/puppet-ceilometer.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-ceilometer.spec b/SPECS/puppet-ceilometer.spec
index a869999..5356f91 100644
--- a/SPECS/puppet-ceilometer.spec
+++ b/SPECS/puppet-ceilometer.spec
@@ -10,6 +10,7 @@ URL:            https://launchpad.net/puppet-ceilometer
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 
 Patch0001:      0001-Roll-up-TIS-patches.patch
+Patch0002:      0002-Fix-ceilometer-puppet-warnings.patch
 
 BuildArch:      noarch
 
@@ -27,6 +28,7 @@ Puppet module for OpenStack Ceilometer
 %prep
 %setup -q -n openstack-ceilometer-%{upstream_version}
 %patch0001 -p1
+%patch0002 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

