From 2d2bc32008adca8f1dff6a9539003aaea0615542 Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Fri, 3 Nov 2017 12:15:00 -0400
Subject: [PATCH 1/4] Update package versioning for TIS format

---
 SPECS/puppet-nova.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-nova.spec b/SPECS/puppet-nova.spec
index 5950895..4dd32c5 100644
--- a/SPECS/puppet-nova.spec
+++ b/SPECS/puppet-nova.spec
@@ -1,7 +1,7 @@
 %{!?upstream_version: %global upstream_version %{version}%{?milestone}}
 Name:           puppet-nova
 Version:        11.4.0
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Puppet module for OpenStack Nova
 License:        ASL 2.0
 
-- 
1.8.3.1

