From f648b2a96fb175766cded083516d4cd73e2207b1 Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Fri, 3 Nov 2017 12:23:09 -0400
Subject: [PATCH 3/4] upstreamable: fix syntax error for until_complete_real

---
 SPECS/puppet-nova.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-nova.spec b/SPECS/puppet-nova.spec
index e209d03..9bd97b0 100644
--- a/SPECS/puppet-nova.spec
+++ b/SPECS/puppet-nova.spec
@@ -10,6 +10,7 @@ URL:            https://launchpad.net/puppet-nova
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 
 Patch0001:      0001-Roll-up-TIS-patches.patch
+Patch0002:      0002-upstreamable-fix-syntax-error-for-until_complete_rea.patch
 
 BuildArch:      noarch
 
@@ -31,6 +32,7 @@ Puppet module for OpenStack Nova
 %setup -q -n openstack-nova-%{upstream_version}
 
 %patch0001 -p1
+%patch0002 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

