From e6242033b356aad8a27599184e98033bf2d2cf1d Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Fri, 3 Nov 2017 16:45:06 -0400
Subject: [PATCH] fix pci and compute::pci

---
 SPECS/puppet-nova.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-nova.spec b/SPECS/puppet-nova.spec
index f20b9a5..b7d70b5 100644
--- a/SPECS/puppet-nova.spec
+++ b/SPECS/puppet-nova.spec
@@ -13,6 +13,7 @@ Patch0001:      0001-Roll-up-TIS-patches.patch
 Patch0002:      0002-upstreamable-fix-syntax-error-for-until_complete_rea.patch
 Patch0003:      0003-Create-nova_cell0-database.patch
 Patch0004:      0004-Filter-out-warnings-in-nova-manage-cell_v2-list_cell.patch
+Patch0005:      0005-fix-pci-and-compute-pci.patch
 
 BuildArch:      noarch
 
@@ -37,6 +38,7 @@ Puppet module for OpenStack Nova
 %patch0002 -p1
 %patch0003 -p1
 %patch0004 -p1
+%patch0005 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

