commit 9ff6f16a0f440832ef6b52162a717f2c303baace
Author: Shoaib Nasir <shoaib.nasir@windriver.com>
Date:   Thu Mar 15 15:00:46 2018 -0400

    WRS: 0009-Remove-SerialConsole-from-NovaConf.patch

diff --git a/SPECS/puppet-nova.spec b/SPECS/puppet-nova.spec
index fc7ec1f..dea8857 100644
--- a/SPECS/puppet-nova.spec
+++ b/SPECS/puppet-nova.spec
@@ -17,6 +17,7 @@ Patch0005:      0005-fix-pci-and-compute-pci.patch
 Patch0006:      0006-Create-Nova-ironic-conf.patch
 Patch0007:      0007-Remove-joshuabaird-ipaclient-from-puppet-nova-requir.patch
 Patch0008:      0008-Adding-pci_weight_multiple-to-nova-scheduler-filter.patch
+Patch0009:      0009-Remove-SerialConsole-from-NovaConf.patch
 
 BuildArch:      noarch
 
@@ -45,6 +46,7 @@ Puppet module for OpenStack Nova
 %patch0006 -p1
 %patch0007 -p1
 %patch0008 -p1
+%patch0009 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
