From 83386553c2e10f04d3080fbd9f8b9b8a7aaa1025 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 3 Jan 2017 13:29:01 -0500
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/puppet-heat.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-heat.spec b/SPECS/puppet-heat.spec
index ce1d9a5..710e16c 100644
--- a/SPECS/puppet-heat.spec
+++ b/SPECS/puppet-heat.spec
@@ -1,7 +1,7 @@
 %{!?upstream_version: %global upstream_version %{version}%{?milestone}}
 Name:           puppet-heat
 Version:        11.3.0
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Puppet module for OpenStack Heat
 License:        ASL 2.0
 
-- 
1.8.3.1

