From cdaacdb829690f7ddfd73d9184b0ebbd24cf363c Mon Sep 17 00:00:00 2001
From: Steven Webster <steven.webster@windriver.com>
Date: Fri, 29 Apr 2022 14:10:47 -0400
Subject: [PATCH] Update package versioning for TIS format

Signed-off-by: Steven Webster <steven.webster@windriver.com>
---
 SPECS/puppet-firewall.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-firewall.spec b/SPECS/puppet-firewall.spec
index 6d42163..f5925f1 100644
--- a/SPECS/puppet-firewall.spec
+++ b/SPECS/puppet-firewall.spec
@@ -8,7 +8,7 @@

 Name:           puppet-firewall
 Version:        1.8.2
-Release:        2%{?alphatag}%{?dist}
+Release:        2.2f892e9git.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Manages Firewalls such as iptables
 License:        ASL 2.0

--
2.29.2

