From 65c3c74fd119db0309d68430ed89652666c884d5 Mon Sep 17 00:00:00 2001
From: systemd team <systemd-maint@redhat.com>
Date: Tue, 10 Oct 2017 17:06:10 -0400
Subject: fix ACL warnings from systemd tmpfiles set

---
 tmpfiles.d/systemd.conf.m4 | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tmpfiles.d/systemd.conf.m4 b/tmpfiles.d/systemd.conf.m4
index 0575408..d984912 100644
--- a/tmpfiles.d/systemd.conf.m4
+++ b/tmpfiles.d/systemd.conf.m4
@@ -27,8 +27,8 @@ d /run/log 0755 root root -
 z /run/log/journal 2755 root systemd-journal - -
 Z /run/log/journal/%m ~2750 root systemd-journal - -
 m4_ifdef(`HAVE_ACL',``
-a+ /run/log/journal/%m - - - - d:group:adm:r-x,d:group:wheel:r-x
-A+ /run/log/journal/%m - - - - group:adm:r-x,group:wheel:r-x
+a+ /run/log/journal/%m - - - - d:group:wrs_protected:r-x,d:group:wheel:r-x
+A+ /run/log/journal/%m - - - - group:wrs_protected:r-x,group:wheel:r-x
 '')m4_dnl
 
 z /var/log/journal 2755 root systemd-journal - -
@@ -37,8 +37,8 @@ z /var/log/journal/%m/system.journal 0640 root systemd-journal - -
 m4_ifdef(`HAVE_ACL',``
 a+ /var/log/journal    - - - - d:group:adm:r-x,d:group:wheel:r-x
 a+ /var/log/journal    - - - - group:adm:r-x,group:wheel:r-x
-a+ /var/log/journal/%m - - - - d:group:adm:r-x,d:group:wheel:r-x
-a+ /var/log/journal/%m - - - - group:adm:r-x,group:wheel:r-x
+a+ /var/log/journal/%m - - - - d:group:wrs_protected:r-x,d:group:wheel:r-x
+a+ /var/log/journal/%m - - - - group:wrs_protected:r-x,group:wheel:r-x
 a+ /var/log/journal/%m/system.journal - - - - group:adm:r--,group:wheel:r--
 '')m4_dnl
 
-- 
1.9.1

