From e7b8b0d6308c2afcdbd17733226e7aaf7f876b09 Mon Sep 17 00:00:00 2001
From: systemd team <systemd-maint@redhat.com>
Date: Thu, 12 Oct 2017 18:06:58 -0400
Subject: [PATCH] set a 1GB size restriction on tpmfs, to prevent OOM Kernel
 failures

---
 units/tmp.mount | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/units/tmp.mount b/units/tmp.mount
index af0cf4a..eda2334 100644
--- a/units/tmp.mount
+++ b/units/tmp.mount
@@ -18,7 +18,7 @@ Before=local-fs.target umount.target
 What=tmpfs
 Where=/tmp
 Type=tmpfs
-Options=mode=1777,strictatime
+Options=mode=1777,strictatime,size=1G
 
 # Make 'systemctl enable tmp.mount' work:
 [Install]
-- 
1.8.3.1

