From 93b0c27355df9fd944ad627413135c1da952c0a9 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:05 -0400
Subject: [PATCH 4/7] WRS: dhclient-dhcp6-set-hostname.patch

---
 SOURCES/dhclient-script | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/SOURCES/dhclient-script b/SOURCES/dhclient-script
index 9f725a0..e4dbc20 100644
--- a/SOURCES/dhclient-script
+++ b/SOURCES/dhclient-script
@@ -598,6 +598,10 @@ dh6config() {
             add_ipv6_addr_with_DAD
 
             make_resolv_conf
+
+            if [ -n "${new_fqdn_hostname}" ] && need_hostname; then
+                hostname ${new_fqdn_hostname} || echo "See -nc option in dhclient(8) man page."
+            fi
             ;;
 
         RENEW6|REBIND6)
@@ -615,6 +619,10 @@ dh6config() {
                [ ! "${new_dhcp6_domain_search}" = "${old_dhcp6_domain_search}" ]; then
                 make_resolv_conf
             fi
+
+            if [ -n "${new_fqdn_hostname}" ] && need_hostname; then
+                hostname ${new_fqdn_hostname} || echo "See -nc option in dhclient(8) man page."
+            fi
             ;;
 
         DEPREF6)
-- 
1.9.1

