From 49fde785386db1dd3b13b1a3902c0b63aa7ff673 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:05 -0400
Subject: [PATCH 7/7] WRS: dhclient-disable-NSUPDATE.patch

---
 SPECS/dhcp.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/dhcp.spec b/SPECS/dhcp.spec
index 7301e63..92e7b30 100644
--- a/SPECS/dhcp.spec
+++ b/SPECS/dhcp.spec
@@ -110,6 +110,7 @@ Patch103: dhclient-ipv6-bind-to-interface.patch
 Patch104: dhclient-ipv6-conditionally-set-hostname.patch
 Patch105: dhclient-handle-wrs-install-uuid.patch
 Patch106: dhclient-dhcp6-wrs-install-uuid.patch
+Patch107: dhclient-disable-NSUPDATE.patch        
 
 BuildRequires: autoconf
 BuildRequires: automake
@@ -428,6 +429,7 @@ rm -rf includes/isc-dhcp
 %patch104 -p1
 %patch105 -p1
 %patch106 -p1
+%patch107 -p1
 
 # Update paths in all man pages
 for page in client/dhclient.conf.5 client/dhclient.leases.5 \
@@ -449,6 +451,7 @@ done
 #libtoolize --copy --force
 autoreconf --verbose --force --install
 
+# WR: failover and tracing need to be disabled in order to disable NSUPDATE
 CFLAGS="%{optflags} -fno-strict-aliasing" \
 %configure \
     --with-srv-lease-file=%{_localstatedir}/lib/dhcpd/dhcpd.leases \
@@ -463,6 +466,7 @@ CFLAGS="%{optflags} -fno-strict-aliasing" \
     --with-ldap \
     --with-ldapcrypto \
     --with-libbind=%{_includedir} --with-libbind-libs=%{_libdir} \
+    --disable-tracing --disable-failover \
     --disable-static \
 %if %sdt
     --enable-systemtap \
-- 
1.9.1

