From 075dd032f651d469e639eb1f25f3d5b7f5ff5485 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:49:27 -0400
Subject: [PATCH 13/13] WRS: add_build_require_on_systemd.patch

---
 SPECS/initscripts.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/initscripts.spec b/SPECS/initscripts.spec
index 09674a7..cdc282e 100644
--- a/SPECS/initscripts.spec
+++ b/SPECS/initscripts.spec
@@ -41,6 +41,7 @@ Requires(pre): /usr/sbin/groupadd
 Requires(post): /sbin/chkconfig, coreutils
 Requires(preun): /sbin/chkconfig
 BuildRequires: glib2-devel popt-devel gettext pkgconfig
+BuildRequires: systemd-devel
 Provides: /sbin/service
 
 Patch4: support-interface-scriptlets.patch
-- 
1.9.1

