Index: resource-agents-3.9.5/heartbeat/exportfs
===================================================================
--- resource-agents-3.9.5/heartbeat/exportfs	2013-02-07 07:17:42.000000000 -0500
+++ resource-agents-3.9.5/heartbeat/exportfs	2015-12-18 12:40:18.382930869 -0500
@@ -184,7 +184,9 @@
 
 is_exported() {
 	local dir=$1
-	local spec=$2
+        # Because clientspec contains square brackets when using IPv6, and the exports entry does not,
+        # it is necessary to remove the square brackets to compare them with each other.
+	local spec=$(echo $2|sed -r 's/(\[|\])//g')
 	exportfs |
 	        sed -e '$! N; s/\n[[:space:]]\+/ /; t; s/[[:space:]]\+\([^[:space:]]\+\)\(\n\|$\)/ \1\2/g; P;D;' |
 			grep -q -x -F "$dir $spec"
