---
 heartbeat/ocf-shellfuncs.in |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/heartbeat/ocf-shellfuncs.in
+++ b/heartbeat/ocf-shellfuncs.in
@@ -174,9 +174,9 @@ hadate() {
 set_logtag() {
 	if [ -z "$HA_LOGTAG" ]; then
 		if [ -n "$OCF_RESOURCE_INSTANCE" ]; then
-			HA_LOGTAG="$__SCRIPT_NAME($OCF_RESOURCE_INSTANCE)[$$]"
+			HA_LOGTAG="OCF_$__SCRIPT_NAME($OCF_RESOURCE_INSTANCE)[$$]"
 		else
-			HA_LOGTAG="$__SCRIPT_NAME[$$]"
+			HA_LOGTAG="OCF_$__SCRIPT_NAME[$$]"
 		fi
 	fi
 }
