---
 heartbeat/Filesystem |    4 ++++
 1 file changed, 4 insertions(+)

--- a/heartbeat/Filesystem
+++ b/heartbeat/Filesystem
@@ -727,6 +727,10 @@ signal_processes() {
 }
 try_umount() {
 	local SUB=$1
+
+	# We need to ensure we umount in namespaces, too
+	/usr/sbin/umount-in-namespace $SUB
+
 	$UMOUNT $umount_force $SUB
 	list_mounts | grep -q " $SUB " >/dev/null 2>&1 || {
 		ocf_log info "unmounted $SUB successfully"
