From 2a3ff36dc325a09ae8601575a0699bdcedcf463b Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Mon, 16 Oct 2017 12:06:28 -0500
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/puppet-openstacklib.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-openstacklib.spec b/SPECS/puppet-openstacklib.spec
index 7f3bbd8..8bf2c1b 100644
--- a/SPECS/puppet-openstacklib.spec
+++ b/SPECS/puppet-openstacklib.spec
@@ -1,7 +1,7 @@
 %{!?upstream_version: %global upstream_version %{version}%{?milestone}}
 Name:           puppet-openstacklib
 Version:        11.5.0
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Puppet OpenStack Libraries
 License:        ASL 2.0
 
-- 
2.7.4

