From 0eea2c3205d9d38e6118e8c97bebae0a87674054 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Wed, 11 Jan 2017 12:30:10 -0500
Subject: [PATCH] Add TIS patch

---
 SPECS/puppet-openstacklib.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-openstacklib.spec b/SPECS/puppet-openstacklib.spec
index 0ed0c1d..38faf85 100644
--- a/SPECS/puppet-openstacklib.spec
+++ b/SPECS/puppet-openstacklib.spec
@@ -8,6 +8,7 @@ License:        ASL 2.0
 URL:            https://launchpad.net/puppet-openstacklib
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
+Patch0001:      0001-Roll-up-TIS-patches.patch
 
 BuildArch:      noarch
 
@@ -24,6 +25,7 @@ Puppet OpenStack Libraries
 
 %prep
 %setup -q -n openstack-openstacklib-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

