From d977fad6b6844de7a22ab62e7456f27c95a56a64 Mon Sep 17 00:00:00 2001
From: chenyan <yan.chen@intel.com>
Date: Thu, 26 Mar 2020 16:17:19 +0800
Subject: [PATCH] update for openstackclient Train

Signed-off-by: chenyan <yan.chen@intel.com>
---
 SPECS/puppet-openstacklib.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-openstacklib.spec b/SPECS/puppet-openstacklib.spec
index ce3957f..9f42722 100644
--- a/SPECS/puppet-openstacklib.spec
+++ b/SPECS/puppet-openstacklib.spec
@@ -9,6 +9,7 @@ URL:            https://launchpad.net/puppet-openstacklib
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 Patch0001:      0001-Roll-up-TIS-patches.patch
+Patch0002:      0002-update-for-openstackclient-Train.patch
 
 BuildArch:      noarch
 
@@ -26,6 +27,7 @@ Puppet OpenStack Libraries
 %prep
 %setup -q -n openstack-openstacklib-%{upstream_version}
 %patch0001 -p1
+%patch0002 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.17.1

