From: Jim Somerville <jim.somerville@windriver.com>
Date: Thu, 7 Apr 2022 18:22:07 +0000
Subject: Make /boot/efi mount as rw

Signed-off-by: Jim Somerville <jim.somerville@windriver.com>
---
 init-ostree-install.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/init-ostree-install.sh b/init-ostree-install.sh
index 1a53f3c..7f7f925 100644
--- a/init-ostree-install.sh
+++ b/init-ostree-install.sh
@@ -1549,6 +1549,12 @@ if [ -d /sys/firmware/efi/efivars ] ;then
     fi
 fi
 
+############################################
+# StarlingX: Do not mount the otaefi partition ro in fstab
+# Caution... If someone resets the /etc/fstab with OSTree this change is lost...
+echo "Removing ro from otaefi mount in fstab"
+sed -i "s:\(LABEL=otaefi.*\) ro:\1 defaults:g" ${PHYS_SYSROOT}/ostree/?/etc/fstab
+
 if [ "$INSTPOST" = "shell" ] ; then
 	echo " Entering interactive install shell, please exit to continue when done"
 	shell_start
