commit 7f052d7778c72d41a5a559f9f81617e39ef3016c
Author: Shoaib Nasir <shoaib.nasir@windriver.com>
Date:   Mon May 14 16:18:45 2018 -0400

    WRS: 0001-Update-package-versioning-for-TIS-format.patch

diff --git a/SPECS/kubernetes.spec b/SPECS/kubernetes.spec
index fd43ef4..ab48da4 100644
--- a/SPECS/kubernetes.spec
+++ b/SPECS/kubernetes.spec
@@ -45,7 +45,7 @@
 ##############################################
 Name:           kubernetes
 Version:        %{kube_version}
-Release:        1%{?dist}
+Release:        1%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Container cluster management
 License:        ASL 2.0
 URL:            https://%{import_path}
@@ -807,9 +807,9 @@ Summary: Kubernetes services for node host
 
 %if 0%{?fedora} >= 27
 Requires: (docker or docker-ce)
-Suggests: docker
+Suggests: docker-ce
 %else
-Requires: docker
+Requires: docker-ce
 %endif
 Requires: conntrack-tools
 
@@ -891,6 +891,7 @@ mv $(ls | grep -v "^src$") src/k8s.io/kubernetes/.
 ###############
 
 %build
+export PBR_VERSION=%{version}
 pushd src/k8s.io/kubernetes/
 export KUBE_GIT_TREE_STATE="clean"
 export KUBE_GIT_COMMIT=%{commit}
@@ -915,6 +916,7 @@ popd
 popd
 
 %install
+export PBR_VERSION=%{version}
 pushd src/k8s.io/kubernetes/
 . hack/lib/init.sh
 kube::golang::setup_env
