From c25a30b4a0c7347234c2af4afab099b5735bbf71 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:05:59 -0400
Subject: [PATCH] WRS: 0001-Update-package-versioning-for-TIS-format.patch

---
 SPECS/net-snmp.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/net-snmp.spec b/SPECS/net-snmp.spec
index a59db4f..8a24ba1 100644
--- a/SPECS/net-snmp.spec
+++ b/SPECS/net-snmp.spec
@@ -11,7 +11,7 @@
 Summary: A collection of SNMP protocol tools and libraries
 Name: net-snmp
 Version: 5.7.2
-Release: 33%{?dist}.2
+Release: 33.el7_5.2%{?_tis_dist}.%{tis_patch_ver}
 Epoch: 1
 
 License: BSD
-- 
2.7.4

