From 28917eb0762c08ffa8255fd29555e1d75681d6f8 Mon Sep 17 00:00:00 2001
From: Tao Liu <tao.liu@windriver.com>
Date: Thu, 4 Jan 2018 11:43:15 -0500
Subject: snmp spec change config file permission

---
 SPECS/net-snmp.spec | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/SPECS/net-snmp.spec b/SPECS/net-snmp.spec
index 6fc0da2..38afcf1 100644
--- a/SPECS/net-snmp.spec
+++ b/SPECS/net-snmp.spec
@@ -558,8 +558,8 @@ rm -rf ${RPM_BUILD_ROOT}
 %doc local/passtest local/ipf-mod.pl
 %doc README.thread AGENT.txt PORTING local/README.mib2c
 %dir %{_sysconfdir}/snmp
-%config(noreplace) %attr(0600,root,root) %{_sysconfdir}/snmp/snmpd.conf
-%config(noreplace) %attr(0600,root,root) %{_sysconfdir}/snmp/snmptrapd.conf
+%config(noreplace) %attr(0640,root,root) %{_sysconfdir}/snmp/snmpd.conf
+%config(noreplace) %attr(0640,root,root) %{_sysconfdir}/snmp/snmptrapd.conf
 %{_bindir}/snmpconf
 %{_bindir}/agentxtrap
 %{_bindir}/net-snmp-create-v3-user
-- 
2.7.4

