From 492d89dca4ca498b77847abc1f1313a800fec85e Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:05:58 -0400
Subject: [PATCH] WRS: snmp-spec-add-init-script.patch

---
 SPECS/net-snmp.spec | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/SPECS/net-snmp.spec b/SPECS/net-snmp.spec
index da7706b..035d2f8 100644
--- a/SPECS/net-snmp.spec
+++ b/SPECS/net-snmp.spec
@@ -41,6 +41,7 @@ Source11: snmptrapd.service
 
 # WRS
 Source12: snmpd.conf.cgcs
+Source13: snmpd.cgcs
 
 Patch1: net-snmp-5.7.2-pie.patch
 Patch2: net-snmp-5.5-dir-fix.patch
@@ -429,6 +430,9 @@ install -m 644 %SOURCE6 ${RPM_BUILD_ROOT}%{_sysconfdir}/snmp/snmptrapd.conf
 install -m 644 %SOURCE12 ${RPM_BUILD_ROOT}%{_sysconfdir}/snmp/snmpd.conf
 chmod 600 ${RPM_BUILD_ROOT}%{_sysconfdir}/snmp/snmptrapd.conf
 
+mkdir -p ${RPM_BUILD_ROOT}%{_sysconfdir}/rc.d/init.d
+install -m 755 %SOURCE13 ${RPM_BUILD_ROOT}%{_sysconfdir}/rc.d/init.d/snmpd
+
 install -d ${RPM_BUILD_ROOT}%{_initrddir}
 install -m 755 %SOURCE2 ${RPM_BUILD_ROOT}%{_initrddir}/snmpd
 install -m 755 %SOURCE3 ${RPM_BUILD_ROOT}%{_initrddir}/snmptrapd
@@ -568,6 +572,7 @@ rm -rf ${RPM_BUILD_ROOT}
 %dir %{_localstatedir}/run/net-snmp
 %{_prefix}/lib/tmpfiles.d/net-snmp.conf
 %{_unitdir}/snmp*
+%{_sysconfdir}/rc.d/init.d/snmpd
 %config(noreplace) %{_sysconfdir}/sysconfig/snmpd
 %config(noreplace) %{_sysconfdir}/sysconfig/snmptrapd
 %attr(0755,root,root) %{_bindir}/net-snmp-config*
-- 
2.7.4

