diff --git a/SPECS/net-snmp.spec b/SPECS/net-snmp.spec
index 12323a4..1e23fba 100644
--- a/SPECS/net-snmp.spec
+++ b/SPECS/net-snmp.spec
@@ -331,7 +331,7 @@ rm testing/fulltests/default/T200*
 %endif
 
 %build
-MIBS="host agentx smux \
+MIBS="agentx smux \
      ucd-snmp/diskio tcp-mib udp-mib mibII/mta_sendmail \
      ip-mib/ipv4InterfaceTable ip-mib/ipv6InterfaceTable \
      ip-mib/ipAddressPrefixTable/ipAddressPrefixTable \
@@ -352,6 +352,7 @@ MIBS="$MIBS ucd-snmp/lmsensorsMib"
     --with-logfile="/var/log/snmpd.log" \
     --with-persistent-directory="/var/lib/net-snmp" \
     --with-mib-modules="$MIBS" \
+    --with-out-mib-modules=host \
 %if %{netsnmp_tcp_wrappers}
     --with-libwrap=yes \
 %endif
