From 061c8b4c1db3fcb4a9e60eb826d1ac474067aae3 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 27 Sep 2016 10:55:15 -0400
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/watchdog.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/watchdog.spec b/SPECS/watchdog.spec
index 75ce3dd..88c4245 100644
--- a/SPECS/watchdog.spec
+++ b/SPECS/watchdog.spec
@@ -2,7 +2,7 @@
 Summary:          Software and/or Hardware watchdog daemon
 Name:             watchdog
 Version:          5.13
-Release:          12%{?dist}
+Release:          12.el7%{?_tis_dist}.%{tis_patch_ver}
 License:          GPLv2+
 
 URL:              http://sourceforge.net/projects/watchdog/
-- 
1.8.3.1

