From 26fa177280c21f0c2048c90ae8fdd3ded7a60113 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Thu, 26 Oct 2017 10:44:17 -0400
Subject: [PATCH] Add ipaddress patch

---
 SPECS/facter.spec | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/SPECS/facter.spec b/SPECS/facter.spec
index aa4a0db..2829879 100644
--- a/SPECS/facter.spec
+++ b/SPECS/facter.spec
@@ -32,10 +32,11 @@ Source1:        https://downloads.puppetlabs.com/%{name}/%{name}-%{version}.tar.
 BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
 
 # WRS patches
-Patch0:         ps.patch
-Patch1:         personality.patch
-Patch2:         centos_remove-net-commands-that-can-timeout.patch
-Patch3:         centos_fix-ipv6-regex.patch
+Patch0001:      0001-ps.patch
+Patch0002:      0002-personality.patch
+Patch0003:      0003-centos_remove-net-commands-that-can-timeout.patch
+Patch0004:      0004-centos_fix-ipv6-regex.patch
+Patch0005:      0005-Hardcode-ipaddress-fact-to-localhost.patch
 
 
 # Upstream claims to only support 1.8.7 and higher
@@ -79,10 +80,11 @@ key off the values returned by facts.
 %prep
 %setup -q
 # WRS apply patches
-%patch0 -p1
-%patch1 -p1
-%patch2 -p2 
-%patch3 -p2
+%patch0001 -p1
+%patch0002 -p1
+%patch0003 -p2
+%patch0004 -p2
+%patch0005 -p1
 
 %build
 # Nothing to build
-- 
1.8.3.1

