From c5127c3365ebcf6124b4e6fad15cccd7ea2e7f20 Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Tue, 8 May 2018 13:08:38 -0400
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/puppet-memcached.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-memcached.spec b/SPECS/puppet-memcached.spec
index 8df048d..6b69f32 100644
--- a/SPECS/puppet-memcached.spec
+++ b/SPECS/puppet-memcached.spec
@@ -8,7 +8,7 @@
 
 Name:           puppet-memcached
 Version:        3.0.2
-Release:        1%{?alphatag}%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Manage memcached via Puppet
 License:        ASL 2.0
 
-- 
1.8.3.1

