From ea31aa96f2b67378a3879af60d1a113482515052 Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Wed, 1 Nov 2017 18:38:36 -0400
Subject: [PATCH 3/4] Create nova_cell0 database

---
 manifests/db/postgresql.pp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/manifests/db/postgresql.pp b/manifests/db/postgresql.pp
index c037886..2771074 100644
--- a/manifests/db/postgresql.pp
+++ b/manifests/db/postgresql.pp
@@ -42,6 +42,15 @@ class nova::db::postgresql(
     privileges    => $privileges,
   }
 
+  # WRS: always create nova_cell0 database
+  ::openstacklib::db::postgresql { 'nova_cell0':
+    password_hash => postgresql_password($user, $password),
+    dbname        => "${dbname}_cell0",
+    user          => $user,
+    encoding      => $encoding,
+    privileges    => $privileges,
+  }
+
   Anchor['nova::db::begin']
   ~> Class['nova::db::postgresql']
   ~> Anchor['nova::db::end']
-- 
1.8.3.1

