commit 10632d76afeaf052f604bb7e983e07c631bbb387
Author: Shoaib Nasir <shoaib.nasir@windriver.com>
Date:   Thu Mar 15 14:54:50 2018 -0400

    WRS: Patch9 - Remove-SerialConsole-from-novaconf.patch

diff --git a/manifests/compute/serial.pp b/manifests/compute/serial.pp
index 83d1526..6a34abb 100644
--- a/manifests/compute/serial.pp
+++ b/manifests/compute/serial.pp
@@ -25,13 +25,6 @@ class nova::compute::serial(
 
   include ::nova::deps
 
-  nova_config {
-    'serial_console/enabled':             value => true;
-    'serial_console/port_range':          value => $port_range;
-    'serial_console/base_url':            value => $base_url;
-    'serial_console/proxyclient_address': value => $proxyclient_address;
-  }
-
   nova_ironic_config {
     'serial_console/enabled':             value => true;
     'serial_console/port_range':          value => $port_range;
diff --git a/manifests/serialproxy.pp b/manifests/serialproxy.pp
index f52b128..5ea2cb4 100644
--- a/manifests/serialproxy.pp
+++ b/manifests/serialproxy.pp
@@ -35,10 +35,6 @@ class nova::serialproxy(
   include ::nova::deps
   include ::nova::params
 
-  nova_config {
-    'serial_console/serialproxy_port':    value => $serialproxy_port;
-    'serial_console/serialproxy_host':    value => $serialproxy_host;
-  }
   nova_ironic_config {
     'serial_console/serialproxy_port':    value => $serialproxy_port;
     'serial_console/serialproxy_host':    value => $serialproxy_host;
