From e1b8dd12605b5654bd94011cb6c587fd0c0bc9af Mon Sep 17 00:00:00 2001
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Tue, 20 Dec 2016 11:58:26 -0500
Subject: [PATCH 2/3] neuter HAVE_IEEE_GETQCN

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 source/include/linux/dcbnl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/include/linux/dcbnl.h b/source/include/linux/dcbnl.h
index 2c8ca62..eabf6da 100644
--- a/source/include/linux/dcbnl.h
+++ b/source/include/linux/dcbnl.h
@@ -5,7 +5,7 @@
 
 #include_next <linux/dcbnl.h>
 
-#ifndef HAVE_IEEE_GETQCN
+#if 0
 
 #ifndef HAVE_STRUCT_IEEE_QCN
 enum dcbnl_cndd_states {
-- 
1.8.3.1

