From 43675576fe1d5780e042852365981bd2b8fa9e0e Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:32:16 -0400
Subject: [4/6] WRS: 0001-Further-parallelize-openssh-build.patch

---
 SPECS/openssh.spec | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/SPECS/openssh.spec b/SPECS/openssh.spec
index be83a63..39bbcac 100644
--- a/SPECS/openssh.spec
+++ b/SPECS/openssh.spec
@@ -606,7 +606,7 @@ fi
 perl -pi -e "s|-lcrypto|%{_libdir}/libcrypto.a|g" Makefile
 %endif
 
-make
+make -j"%(nproc)"
 
 # Define a variable to toggle gnome1/gtk2 building.  This is necessary
 # because RPM doesn't handle nested %if statements.
@@ -632,7 +632,7 @@ popd
 pushd pam_ssh_agent_auth-%{pam_ssh_agent_ver}
 LDFLAGS="$SAVE_LDFLAGS"
 %configure --with-selinux --libexecdir=/%{_libdir}/security --with-mantype=man
-make
+make -j"%(nproc)"
 popd
 %endif
 
-- 
2.7.4

