From 788beea32cfa09f61c27db117ec3b3b0500fbad8 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:32:16 -0400
Subject: [2/6] WRS: openssh-spec-file-add-init.patch

---
 SPECS/openssh.spec | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/SPECS/openssh.spec b/SPECS/openssh.spec
index e137156..d10e73e 100644
--- a/SPECS/openssh.spec
+++ b/SPECS/openssh.spec
@@ -728,9 +728,6 @@ getent passwd sshd >/dev/null || \
 %preun server
 %systemd_preun sshd.service sshd.socket
 
-%postun server
-%systemd_postun_with_restart sshd.service
-
 %files
 %defattr(-,root,root)
 %{!?_licensedir:%global license %%doc}
@@ -793,8 +790,6 @@ getent passwd sshd >/dev/null || \
 %attr(0644,root,root) %{_unitdir}/sshd.socket
 %attr(0644,root,root) %{_unitdir}/sshd-keygen.service
 
-%files server-sysvinit
-%defattr(-,root,root)
 %attr(0755,root,root) /etc/rc.d/init.d/sshd
 %endif
 
-- 
2.7.4

