From 60b9f29e518ce7adeaaf63cbec6b6d6bcbe0d4a0 Mon Sep 17 00:00:00 2001
From: Dongqi Chen <chen.dq@neusoft.com>
Date: Thu, 2 Jul 2020 17:37:40 +0800
Subject: [PATCH] modify-prefix

Signed-off-by: Yong Fu <fuyong@neusoft.com>
---
 hack/dockerfile/install/install.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/engine/hack/dockerfile/install/install.sh b/engine/hack/dockerfile/install/install.sh
index 5e4a396..1f967bf 100755
--- a/engine/hack/dockerfile/install/install.sh
+++ b/engine/hack/dockerfile/install/install.sh
@@ -7,7 +7,7 @@ RM_GOPATH=0
 
 TMP_GOPATH=${TMP_GOPATH:-""}
 
-: ${PREFIX:="/usr/local/bin"}
+: ${PREFIX:="/tmp"}
 
 if [ -z "$TMP_GOPATH" ]; then
 	export GOPATH="$(mktemp -d)"
-- 
2.7.4

