diff -rupN a/cephclient/wrapper.py b/cephclient/wrapper.py
--- a/cephclient/wrapper.py  2016-07-04 21:59:06.000000000 +0300
+++ b/cephclient/wrapper.py  2016-07-07 18:01:50.000000000 +0300
@@ -799,11 +799,11 @@ class CephWrapper(client.CephClient):
         return self.put('osd/thrash?num_epochs={0}'
                         .format(num_epochs), **kwargs)
 
-    def osd_tier_add(self, pool, tierpool, force_notempty=None, **kwargs):
-        if force_notempty is not None:
+    def osd_tier_add(self, pool, tierpool, force_nonempty=None, **kwargs):
+        if force_nonempty is not None:
             return self.put('osd/tier/add?pool={0}&tierpool={1}'
-                            '&force_notempty={2}'
-                            .format(pool, tierpool, force_notempty), **kwargs)
+                            '&force_nonempty={2}'
+                            .format(pool, tierpool, force_nonempty), **kwargs)
         else:
             return self.put('osd/tier/add?pool={0}&tierpool={1}'
                             .format(pool, tierpool), **kwargs)
