From 2cc29381484f15cb9e60d94491ebb260402edea1 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Mon, 2 Oct 2017 08:29:19 -0500
Subject: [PATCH 1/1] Update package versioning for TIS format

---
 SPECS/puppet.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet.spec b/SPECS/puppet.spec
index 777abe2..023f74f 100644
--- a/SPECS/puppet.spec
+++ b/SPECS/puppet.spec
@@ -19,7 +19,7 @@
 
 Name:           puppet
 Version:        4.8.2
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        A network tool for managing many disparate systems
 License:        ASL 2.0
 URL:            http://puppetlabs.com
-- 
1.8.3.1

