From cb258865db8bd9c37e22e45181b5dd3c9b3fee85 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 7 Mar 2017 13:38:12 -0500
Subject: [PATCH] Add TIS content

---
 SPECS/syslog-ng.spec | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/SPECS/syslog-ng.spec b/SPECS/syslog-ng.spec
index b54632c..a6d3f68 100644
--- a/SPECS/syslog-ng.spec
+++ b/SPECS/syslog-ng.spec
@@ -12,10 +12,12 @@ URL: http://www.balabit.com/network-security/syslog-ng
 Source0: http://www.balabit.com/downloads/files/syslog-ng/open-source-edition/%{version}/source/%{name}_%{version}.tar.gz
 Source1: syslog-ng.conf
 Source2: syslog-ng.logrotate
+Source3: remotelogging.conf
 
 Patch0: syslog-ng-3.5.0-syslog-ng.service.patch
 Patch1: syslog-ng-3.4.0beta1-tests-functional-control.py.patch
 Patch2: syslog-ng-3.3.6-tests-functional-sql-test.patch
+Patch3: syslog-ng-service-pid-file-pmond.patch
 
 BuildRequires: systemd-units
 BuildRequires: pkgconfig
@@ -148,6 +150,7 @@ developing applications that use %{name}.
 %patch0 -p1
 %patch1 -p1
 %patch2 -p1
+%patch3 -p1
 
 # fix perl path
 %{__sed} -i 's|^#!/usr/local/bin/perl|#!%{__perl}|' contrib/relogger.pl
@@ -203,6 +206,7 @@ make DESTDIR=%{buildroot} install
 
 %{__install} -d -m 755 %{buildroot}%{_sysconfdir}/%{name}/conf.d
 %{__install} -p -m 644 %{SOURCE1} %{buildroot}%{_sysconfdir}/%{name}/syslog-ng.conf
+%{__install} -p -m 644 %{SOURCE3} %{buildroot}%{_sysconfdir}/%{name}/remotelogging.conf
 
 %{__install} -d -m 755 %{buildroot}%{_sysconfdir}/logrotate.d
 %{__install} -p -m 644 %{SOURCE2} %{buildroot}%{_sysconfdir}/logrotate.d/syslog
@@ -284,6 +288,7 @@ fi
 %dir %{_sysconfdir}/%{name}/conf.d
 %dir %{_sysconfdir}/%{name}/patterndb.d
 %config(noreplace) %{_sysconfdir}/%{name}/%{name}.conf
+%config(noreplace) %{_sysconfdir}/%{name}/remotelogging.conf
 %config(noreplace) %{_sysconfdir}/%{name}/scl.conf
 %config(noreplace) %{_sysconfdir}/logrotate.d/syslog
 %{_unitdir}/%{name}.service
-- 
1.8.3.1

