From 3c0c5fa0443c07f7c092b1d97989f881fc623172 Mon Sep 17 00:00:00 2001
From: Jerry Sun <jerry.sun@windriver.com>
Date: Tue, 30 Jan 2018 09:43:42 -0500
Subject: [PATCH 1/1] removed default values for depricated fields; they were
 causing warnings

---
 manifests/init.pp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/manifests/init.pp b/manifests/init.pp
index 54ddd50..09d3f13 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -347,8 +347,8 @@ class murano(
   $rabbit_os_port             = $::os_service_default,
   $rabbit_os_hosts            = $::os_service_default,
   $rabbit_os_virtual_host     = $::os_service_default,
-  $rabbit_os_user             = 'guest',
-  $rabbit_os_password         = 'guest',
+  $rabbit_os_user             = $::os_service_default,
+  $rabbit_os_password         = $::os_service_default,
 ) inherits murano::params {
 
   include ::murano::deps
-- 
1.8.3.1

