Index: git/cephclient/wrapper.py
===================================================================
--- git.orig/cephclient/wrapper.py	2015-10-26 16:00:32.768154102 +0200
+++ git/cephclient/wrapper.py	2015-11-04 14:46:09.491855340 +0200
@@ -412,6 +412,10 @@
         else:
             return self.get('osd/pool/stats', **kwargs)
 
+    def osd_get_pool_quota(self, pool, **kwargs):
+        return self.get('osd/pool/get-quota?pool={0}'
+                        .format(pool), **kwargs)
+
     def osd_stat(self, **kwargs):
         return self.get('osd/stat', **kwargs)
 
