---
 cephclient/client.py |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

--- a/cephclient/client.py
+++ b/cephclient/client.py
@@ -53,7 +53,12 @@ class CephClient(object):
 
         self.log.debug("Params: {0}".format(str(self.params)))
 
-        self.endpoint = self.params['endpoint']
+        if 'endpoint' in self.params:
+            self.endpoint = self.params['endpoint']
+        else:
+            # default endpoint
+            self.endpoint = 'http://localhost:5001/api/v0.1/'
+
         if 'timeout' not in self.params:
             self.timeout = None
 
