diff --git a/eventlet/wsgi.py b/eventlet/wsgi.py
index 7258277..057ef64 100644
--- a/eventlet/wsgi.py
+++ b/eventlet/wsgi.py
@@ -512,7 +512,12 @@ class HttpProtocol(BaseHTTPServer.BaseHTTPRequestHandler):
                 hook(self.environ, *args, **kwargs)
 
             if self.server.log_output:
-                self.server.log.info(self.server.log_format % {
+                if (self.requestline.startswith('GET') and
+                    (status_code[0] == '200')):
+                    logger = self.server.log.debug
+                else:
+                    logger = self.server.log.info
+                logger(self.server.log_format % {
                     'client_ip': self.get_client_ip(),
                     'client_port': self.client_address[1],
                     'date_time': self.log_date_time_string(),
