---
 SPECS/pam.spec | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/SPECS/pam.spec b/SPECS/pam.spec
index 0cbf153..a702aca 100644
--- a/SPECS/pam.spec
+++ b/SPECS/pam.spec
@@ -27,6 +27,11 @@ Source14: 20-nproc.conf
 Source15: pamtmp.conf
 Source16: postlogin.pamd
 Source17: postlogin.5
+Source18: common-account
+Source19: common-auth
+Source20: common-password
+Source21: common-session
+Source22: common-session-noninteractive
 Patch1:  pam-1.0.90-redhat-modules.patch
 Patch2:  pam-1.1.6-std-noclose.patch
 Patch4:  pam-1.1.0-console-nochmod.patch
@@ -207,6 +212,11 @@ install -m 644 %{SOURCE8} $RPM_BUILD_ROOT%{_pamconfdir}/fingerprint-auth
 install -m 644 %{SOURCE9} $RPM_BUILD_ROOT%{_pamconfdir}/smartcard-auth
 install -m 644 %{SOURCE10} $RPM_BUILD_ROOT%{_pamconfdir}/config-util
 install -m 644 %{SOURCE16} $RPM_BUILD_ROOT%{_pamconfdir}/postlogin
+install -m 644 %{SOURCE18} $RPM_BUILD_ROOT%{_pamconfdir}/common-account
+install -m 644 %{SOURCE19} $RPM_BUILD_ROOT%{_pamconfdir}/common-auth
+install -m 644 %{SOURCE20} $RPM_BUILD_ROOT%{_pamconfdir}/common-password
+install -m 644 %{SOURCE21} $RPM_BUILD_ROOT%{_pamconfdir}/common-session
+install -m 644 %{SOURCE22} $RPM_BUILD_ROOT%{_pamconfdir}/common-session-noninteractive
 install -m 644 %{SOURCE14} $RPM_BUILD_ROOT%{_secconfdir}/limits.d/20-nproc.conf
 install -m 600 /dev/null $RPM_BUILD_ROOT%{_secconfdir}/opasswd
 install -d -m 755 $RPM_BUILD_ROOT/var/log
@@ -297,6 +307,11 @@ fi
 %config(noreplace) %{_pamconfdir}/smartcard-auth
 %config(noreplace) %{_pamconfdir}/config-util
 %config(noreplace) %{_pamconfdir}/postlogin
+%config(noreplace) %{_pamconfdir}/common-account
+%config(noreplace) %{_pamconfdir}/common-auth
+%config(noreplace) %{_pamconfdir}/common-password
+%config(noreplace) %{_pamconfdir}/common-session
+%config(noreplace) %{_pamconfdir}/common-session-noninteractive
 %doc Copyright
 %doc doc/txts
 %doc doc/sag/*.txt doc/sag/html
-- 
1.8.3.1

