From 30a087a13a78b77537a969db2a30b531246b0bd7 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Mon, 26 Sep 2016 17:39:58 -0400
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/lshell.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/lshell.spec b/SPECS/lshell.spec
index 0fd4d17..e5f1317 100644
--- a/SPECS/lshell.spec
+++ b/SPECS/lshell.spec
@@ -2,7 +2,7 @@
 
 Name:           lshell
 Version:        0.9.16
-Release:        6%{?dist}
+Release:        5.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        A Python-based limited shell
 
 License:        GPLv3+
-- 
1.8.3.1

