From 70d970536ec4312be28c7c39b20fe90199c495e0 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:26 -0400
Subject: [PATCH 6/7] WRS: Patch106: dhclient-dhcp6-wrs-install-uuid.patch

---
 client/scripts/linux | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/client/scripts/linux b/client/scripts/linux
index ddf50e8..c4ab9a1 100755
--- a/client/scripts/linux
+++ b/client/scripts/linux
@@ -77,15 +77,22 @@ exit_with_hooks() {
   exit $exit_status
 }
 
+# Select wrs-install-uuid from ipv4 or ipv6
+if [ -n "$new_dhcp6_wrs_install_uuid" ]; then
+  wrs_install_uuid=$new_dhcp6_wrs_install_uuid
+else
+  wrs_install_uuid=$new_wrs_install_uuid
+fi
+
 # Enforce wrs-install-uuid on management and infrastrucure interfaces
 source /etc/platform/platform.conf
-if [ -n "$new_wrs_install_uuid" ]; then
+if [ -n "$wrs_install_uuid" ]; then
   if [ "$nodetype" == "controller" ]; then
     source /etc/build.info
     file="/www/pages/feed/rel-$SW_VERSION/install_uuid"
     INSTALL_UUID=$(cat "$file")
   fi
-  if [ "$INSTALL_UUID" != "$new_wrs_install_uuid" ]; then
+  if [ "$INSTALL_UUID" != "$wrs_install_uuid" ]; then
     exit 1
   fi
 elif [ "$interface" == "$management_interface" -o \
-- 
1.9.1

