From d6daacb050008d473b986f574434f9b8ae7139e4 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:23 -0400
Subject: [PATCH 5/7] WRS: Patch105: dhclient-handle-wrs-install-uuid.patch

---
 client/scripts/linux | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/client/scripts/linux b/client/scripts/linux
index 59e764f..ddf50e8 100755
--- a/client/scripts/linux
+++ b/client/scripts/linux
@@ -77,6 +77,25 @@ exit_with_hooks() {
   exit $exit_status
 }
 
+# Enforce wrs-install-uuid on management and infrastrucure interfaces
+source /etc/platform/platform.conf
+if [ -n "$new_wrs_install_uuid" ]; then
+  if [ "$nodetype" == "controller" ]; then
+    source /etc/build.info
+    file="/www/pages/feed/rel-$SW_VERSION/install_uuid"
+    INSTALL_UUID=$(cat "$file")
+  fi
+  if [ "$INSTALL_UUID" != "$new_wrs_install_uuid" ]; then
+    exit 1
+  fi
+elif [ "$interface" == "$management_interface" -o \
+       "$interface" == "$infrastructure_interface" ]; then
+  if [ "$nodetype" != "controller" -o \
+       -e "/etc/platform/.initial_config_complete" ]; then
+    exit 1
+  fi
+fi
+
 # Invoke the local dhcp client enter hooks, if they exist.
 if [ -f /etc/dhclient-enter-hooks ]; then
   exit_status=0
-- 
1.9.1

