From f321c8a8b800a7c2ca9394d3c76bec72b98c0d77 Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Fri, 11 May 2018 15:38:56 -0400
Subject: [PATCH] always restart memcached service

---
 SPECS/memcached.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/memcached.spec b/SPECS/memcached.spec
index c8575c8..f389035 100644
--- a/SPECS/memcached.spec
+++ b/SPECS/memcached.spec
@@ -16,6 +16,7 @@ Source1:        memcached.sysconfig
 
 # https://github.com/memcached/memcached/issues/218
 Patch1:         0001-systemd-fix-upstream-provided-service.patch
+Patch2:         0002-always-restart-memcached-service.patch 
 
 BuildRequires:  libevent-devel systemd-units
 BuildRequires:  perl-generators
@@ -44,6 +45,7 @@ access to the memcached binary include files.
 %prep
 %setup -q
 %patch1 -p1 -b .unit
+%patch2 -p1
 
 %build
 # compile with full RELRO
-- 
1.8.3.1

