From 0c36ecaef39328e85f41ebe8164dc7da5949542a Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 11 Apr 2017 11:14:25 -0400
Subject: [PATCH] Disable timeout for mkfs command

---
 manifests/resource/up.pp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/manifests/resource/up.pp b/manifests/resource/up.pp
index b626f55..f9de8ab 100644
--- a/manifests/resource/up.pp
+++ b/manifests/resource/up.pp
@@ -54,6 +54,7 @@ define drbd::resource::up (
       }
       exec { "drbd_format_volume_${name}":
         command     => "mkfs.${fs_type} ${mkfs_opts} ${device}",
+        timeout     => 0,
         refreshonly => true,
         require     => Exec["drbd_make_primary_${name}"],
         before      => $before,
-- 
1.8.3.1

