From bb7b75184f7037e6d8d844874ae248fce1d06736 Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Fri, 11 May 2018 15:24:28 -0400
Subject: [PATCH] Always restart memcached service

---
 scripts/memcached.service | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scripts/memcached.service b/scripts/memcached.service
index 1bb9d33..8e58485 100644
--- a/scripts/memcached.service
+++ b/scripts/memcached.service
@@ -71,5 +71,9 @@ RestrictAddressFamilies=AF_INET AF_INET6 AF_UNIX
 # Takes away the ability to create or manage any kind of namespace
 RestrictNamespaces=true
 
+# WRS
+Restart=always
+RestartSec=0
+
 [Install]
 WantedBy=multi-user.target
-- 
1.8.3.1

