From 7954a4416c5605803df8f570148f948195bac267 Mon Sep 17 00:00:00 2001
From: Jorge Saffe <jorge.saffe@windriver.com>
Date: Thu, 19 Sep 2024 22:18:43 +0200
Subject: [PATCH 6/6] Update Postgres Auth and Password Encryption

---
 manifests/db/postgresql.pp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/manifests/db/postgresql.pp b/manifests/db/postgresql.pp
index a7ddedf..adadfb5 100644
--- a/manifests/db/postgresql.pp
+++ b/manifests/db/postgresql.pp
@@ -45,7 +45,12 @@ define openstacklib::db::postgresql (
 in a future release. Use password instead')
     $password_hash_real = $password_hash
   } elsif $password != undef {
-    $password_hash_real = postgresql::postgresql_password($user, $password)
+    $password_hash_real = postgresql::postgresql_password(
+      $user,
+      $password,
+      $password =~ Sensitive[String],
+      $postgresql::server::password_encryption,
+    )
   } else {
     fail('password should be set')
   }
-- 
2.39.2

