From c194c17c2022a34ce1c1eddd930eb943aaa2e1ce Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Fri, 11 May 2018 21:39:23 -0400
Subject: [PATCH] Update package versioning for Titanium format

---
 SPECS/puppet-horizon.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-horizon.spec b/SPECS/puppet-horizon.spec
index 955f56c..511cf25 100644
--- a/SPECS/puppet-horizon.spec
+++ b/SPECS/puppet-horizon.spec
@@ -1,7 +1,7 @@
 %{!?upstream_version: %global upstream_version %{version}%{?milestone}}
 Name:           puppet-horizon
 Version:        11.5.0
-Release:        1%{?dist}
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Puppet module for OpenStack Horizon
 License:        ASL 2.0
 
-- 
2.7.4

