From a59312d5e966405d0053acae383f315d8ffa9dd2 Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Fri, 11 May 2018 21:45:14 -0400
Subject: [PATCH 2/3] update memcached dependency

---
 SPECS/puppet-horizon.spec | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/SPECS/puppet-horizon.spec b/SPECS/puppet-horizon.spec
index 511cf25..ada0538 100644
--- a/SPECS/puppet-horizon.spec
+++ b/SPECS/puppet-horizon.spec
@@ -9,6 +9,9 @@ URL:            https://launchpad.net/puppet-horizon
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 
+# WRS Patches
+Patch0001:      0001-update-memcached-dependency.patch
+
 BuildArch:      noarch
 
 Requires:       puppet-apache
@@ -21,6 +24,9 @@ Puppet module for OpenStack Horizon
 
 %prep
 %setup -q -n openstack-horizon-%{upstream_version}
+# WRS Patches
+%patch0001 -p1
+
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
2.7.4

