From 55b381fed1b1bae9bd0bdfabd07246f40805252e Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 27 Sep 2016 21:23:24 -0400
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/netpbm.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/netpbm.spec b/SPECS/netpbm.spec
index c10c448..74c13eb 100644
--- a/SPECS/netpbm.spec
+++ b/SPECS/netpbm.spec
@@ -1,7 +1,7 @@
 Summary: A library for handling different graphics file formats
 Name: netpbm
 Version: 10.61.02
-Release: 9%{?dist}
+Release: 9.el7%{?_tis_dist}.%{tis_patch_ver}
 # See copyright_summary for details
 License: BSD and GPLv2 and IJG and MIT and Public Domain
 Group: System Environment/Libraries
-- 
1.8.3.1

