commit bb91ab26622a5ec695c6564af5a9e5e54fdc903c
Author: Shoaib Nasir <shoaib.nasir@windriver.com>
Date:   Thu Feb 15 15:04:55 2018 -0500

    WRS: Patch3: 0003-remove-eventlet_bindhost-from-Keystoneconf.patch

diff --git a/manifests/init.pp b/manifests/init.pp
index 4d79d30..d64638c 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -871,8 +871,9 @@ admin_token will be removed in a later release")
     # WRS: the following options are deprecated for removal
     # however public_bind_host and admin_bind_host are still required as long as
     # keystone is running under eventlet
-    'DEFAULT/public_bind_host': value => $public_bind_host;
-    'DEFAULT/admin_bind_host':  value => $admin_bind_host;
+    # WRS: bind_host options removed from keystone.conf [DEFAULT]
+    #'DEFAULT/public_bind_host': value => $public_bind_host;
+    #'DEFAULT/admin_bind_host':  value => $admin_bind_host;
     #'DEFAULT/public_port':      value => $public_port;
     #'DEFAULT/admin_port':       value => $admin_port;
     'DEFAULT/member_role_id':   value => $member_role_id;
@@ -1110,10 +1111,12 @@ Fernet or UUID tokens are recommended.")
   # WRS: The following options are deprecated for removal
   # however they are still required as long as keystone
   # is running under eventlet
-  keystone_config {
-    'eventlet_server/admin_workers':  value => $admin_workers;
-    'eventlet_server/public_workers': value => $public_workers;
-  }
+  # WRS(snasir): Removing these options from keystone.conf 
+  # since they are now populated in keystone-api.conf
+  #keystone_config {
+  #  'eventlet_server/admin_workers':  value => $admin_workers;
+  #  'eventlet_server/public_workers': value => $public_workers;
+  #}
 
   if $manage_service {
     if $enabled {
