--- a/smart/backends/rpm/pm.py	2014-02-14 14:14:18.877737747 +0800
+++ b/smart/backends/rpm/pm.py	2014-02-14 14:17:17.394560608 +0800
@@ -232,6 +232,10 @@
         if sysconf.get("rpm-order"):
             ts.order()
         probfilter = rpm.RPMPROB_FILTER_OLDPACKAGE
+        if sysconf.get("rpm-ignoresize", False):
+            probfilter |= rpm.RPMPROB_FILTER_DISKNODES
+            probfilter |= rpm.RPMPROB_FILTER_DISKSPACE
+
         if force or reinstall:
             probfilter |= rpm.RPMPROB_FILTER_REPLACEPKG
             probfilter |= rpm.RPMPROB_FILTER_REPLACEOLDFILES
