From f19fe3f4d454d77661bae46a85c9f1774e9b8eff Mon Sep 17 00:00:00 2001
From: Sai Lakshmi Teja Vanka <SaiLakshmiTeja.Vanka@windriver.com>
Date: Wed, 2 Jul 2025 09:35:54 +0000
Subject: [PATCH] Keep SOURCES.txt

---
 dhpython/fs.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/dhpython/fs.py b/dhpython/fs.py
index ce3b90e..6e9488f 100644
--- a/dhpython/fs.py
+++ b/dhpython/fs.py
@@ -479,9 +479,6 @@ class Scan:
                 self.result['nsp.txt'].add(join(dpath, 'namespace_packages.txt'))
             if 'PKG-INFO' in file_names:
                 self.result['egg-info'].add(join(dpath, 'PKG-INFO'))
-            if 'SOURCES.txt' in file_names:
-                os.remove(join(dpath, 'SOURCES.txt'))
-                file_names.remove('SOURCES.txt')
 
     def is_egg_file(self, fpath):
         """Check if given file contains egg-info."""
-- 
2.47.2

