From 4a6da18465135333b470cc3f1817b78e6177886a Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:57:43 -0400
Subject: [PATCH 1/3] WRS: spec-disable-irqbalance-service.patch

---
 SPECS/irqbalance.spec | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/SPECS/irqbalance.spec b/SPECS/irqbalance.spec
index 9d4e948..3e70224 100644
--- a/SPECS/irqbalance.spec
+++ b/SPECS/irqbalance.spec
@@ -89,21 +89,6 @@ install -p -m 0644 ./irqbalance.1 %{buildroot}%{_mandir}/man1/
 %{_mandir}/man1/*
 %config(noreplace) %{_sysconfdir}/sysconfig/irqbalance
 
-%post
-%systemd_post irqbalance.service
-
-%preun
-%systemd_preun irqbalance.service
-
-%postun
-%systemd_postun_with_restart irqbalance.service
-
-%triggerun -- irqbalance < 2:0.56-3
-if /sbin/chkconfig --level 3 irqbalance ; then
-    /bin/systemctl enable irqbalance.service >/dev/null 2>&1 || :
-fi
-/sbin/chkconfig --del irqbalance >/dev/null 2>&1 || :
-
 %changelog
 * Tue May 16 2017 Petr Oros <poros@redhat.com> - 3:1.0.7-10
 - irqbalance node package patch
-- 
1.9.1

