From 1a60b6e068a6f6289a48bd8281d116ed6a51f03e Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 15:25:12 -0400
Subject: [PATCH 1/7] WRS: Patch101: support-disable-nsupdate.patch

---
 server/dhcpd.c    | 2 ++
 server/failover.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/server/dhcpd.c b/server/dhcpd.c
index 67fec83..7523093 100644
--- a/server/dhcpd.c
+++ b/server/dhcpd.c
@@ -504,8 +504,10 @@ main(int argc, char **argv) {
 	trace_srandom = trace_type_register ("random-seed", (void *)0,
 					     trace_seed_input,
 					     trace_seed_stop, MDL);
+#if defined (NSUPDATE)
 	trace_ddns_init();
 #endif
+#endif
 
 #if defined (PARANOIA)
 	/* get user and group info if those options were given */
diff --git a/server/failover.c b/server/failover.c
index 8944102..d26adfa 100644
--- a/server/failover.c
+++ b/server/failover.c
@@ -5232,7 +5232,9 @@ isc_result_t dhcp_failover_process_bind_update (dhcp_failover_state_t *state,
 	 */
 	if (msg->binding_status == FTS_ACTIVE &&
 	    (chaddr_changed || ident_changed)) {
+#if defined (NSUPDATE)
 		(void) ddns_removals(lease, NULL, NULL, ISC_FALSE);
+#endif
 
 		if (lease->scope != NULL)
 			binding_scope_dereference(&lease->scope, MDL);
-- 
1.9.1

