From e2dc69fb6aef8f45556d2884db02942d84543013 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 24 Jan 2017 11:12:02 -0500
Subject: [PATCH] Add TIS Patches

---
 SPECS/puppet-heat.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-heat.spec b/SPECS/puppet-heat.spec
index 4ca652e..5c1ec2f 100644
--- a/SPECS/puppet-heat.spec
+++ b/SPECS/puppet-heat.spec
@@ -8,6 +8,7 @@ License:        ASL 2.0
 URL:            https://launchpad.net/puppet-heat
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
+Patch0001:      0001-Roll-up-TIS-patches.patch
 
 BuildArch:      noarch
 
@@ -23,6 +24,7 @@ Puppet module for OpenStack Heat
 
 %prep
 %setup -q -n openstack-heat-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

