From d63b56b8511b808a4c23c4c15ed81e368f9b020c Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 27 Sep 2016 10:59:20 -0400
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/libevent.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/libevent.spec b/SPECS/libevent.spec
index 9c6cc3e..7d98b8f 100644
--- a/SPECS/libevent.spec
+++ b/SPECS/libevent.spec
@@ -1,6 +1,6 @@
 Name:           libevent
 Version:        2.0.21
-Release:        4%{?dist}
+Release:        4.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Abstract asynchronous event notification library
 
 Group:          System Environment/Libraries
-- 
1.8.3.1

