From a0c3f898cabd79b07cc6cc52245de712ee7b46bd Mon Sep 17 00:00:00 2001
From: John Kung <john.kung@windriver.com>
Date: Wed, 7 Aug 2019 15:01:11 -0400
Subject: [PATCH 1/1] Update-spec-include-stx-patches

Signed-off-by: Dongqi Chen <chen.dq@neusoft.com>
---
 SPECS/python-docker.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/python-docker.spec b/SPECS/python-docker.spec
index 92420e3..9a05819 100644
--- a/SPECS/python-docker.spec
+++ b/SPECS/python-docker.spec
@@ -22,6 +22,8 @@ License:        ASL 2.0
 URL:            https://pypi.org/project/%{srcname}
 Source0:        https://files.pythonhosted.org/packages/source/d/%{srcname}/%{srcname}-%{version}.tar.gz
 
+Patch1: 0001-Update-client-close-socket.patch
+
 BuildArch:      noarch
 
 %description
@@ -92,6 +94,8 @@ run containers, manage containers, manage Swarms, etc.
 
 %prep
 %setup -n %{srcname}-%{version}
+%patch1 -p1
+
 rm -fr docker.egg-info
 
 %build
-- 
2.7.4

