From 9c5837d4d7a60653e418157e3a9552ddcc36d29e Mon Sep 17 00:00:00 2001
From: Andy Ning <andy.ning@windriver.com>
Date: Wed, 28 Mar 2018 14:20:39 -0400
Subject: fix systemd tmpfiles ACL warnings

---
 SPECS/systemd.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/systemd.spec b/SPECS/systemd.spec
index 55e44a5..f1dea1e 100644
--- a/SPECS/systemd.spec
+++ b/SPECS/systemd.spec
@@ -547,6 +547,7 @@ Patch0505: 0505-remove-id-sas-path-symlink.patch
 Patch0506: 0506-fix-ACL-warnings-from-systemd-tmpfiles-set.patch
 Patch0507: 0507-move-vartmp-to-tmpfs.patch
 Patch0508: 0508-set-a-1GB-size-restriction-on-tpmfs.patch
+Patch0509: 0509-fix-systemd-tmpfiles-ACL-warnings.patch
 
 %global num_patches %{lua: c=0; for i,p in ipairs(patches) do c=c+1; end; print(c);}
 
-- 
1.8.3.1

