From 7d3bd33be8b2c9924a2987710f54f5c62853d86c Mon Sep 17 00:00:00 2001
From: jmckenna <jason.mckenna@windriver.com>
Date: Wed, 5 Apr 2017 09:35:06 -0400
Subject: [PATCH] Use generic OS title for Titanium

---
 new-kernel-pkg | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/new-kernel-pkg b/new-kernel-pkg
index 847e959..977ef2d 100755
--- a/new-kernel-pkg
+++ b/new-kernel-pkg
@@ -166,6 +166,8 @@ set_title() {
 	fi
     elif [ $ARCH = 's390' -o $ARCH = 's390x' ]; then
 	title=$(echo $version | sed 's/ /_/g')
+    elif [ -f /etc/build.info ]; then
+	title="Linux ($version)"
     elif [ -f /etc/os-release ]; then
 	. /etc/os-release
 	title="$NAME ($version) $VERSION"
-- 
1.9.1

