From 5da3820896a51fd018371bbc08900f22ca0ea4a3 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 3 May 2016 21:53:21 -0400
Subject: [PATCH 1/1] Create /var/run/resource-agents, if needed

---
 heartbeat/ocf-shellfuncs.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/heartbeat/ocf-shellfuncs.in b/heartbeat/ocf-shellfuncs.in
index 7b5f831..3565e20 100644
--- a/heartbeat/ocf-shellfuncs.in
+++ b/heartbeat/ocf-shellfuncs.in
@@ -166,6 +166,11 @@ __ocf_set_defaults() {
 		ha_log "ERROR: Need to tell us our resource instance name."
 		exit $OCF_ERR_ARGS
 	fi
+
+	# TODO: Find a better way to ensure this dir exists
+	if [ ! -d "$HA_RSCTMP" ]; then
+		mkdir -p $HA_RSCTMP
+	fi
 }
 
 hadate() {
-- 
2.7.4

