From f658722ec239a258cd3ec2e828ca3f37ba2fe536 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Fri, 17 Jun 2016 00:31:20 -0400
Subject: [PATCH 1/1] Notify rmon of shutdown before shutting down LVM and
 Filesystem

---
 heartbeat/Filesystem | 5 +++--
 heartbeat/LVM        | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/heartbeat/Filesystem b/heartbeat/Filesystem
index 05e4097..d5f3417 100755
--- a/heartbeat/Filesystem
+++ b/heartbeat/Filesystem
@@ -883,9 +883,10 @@ case $OP in
 	validate-all) Filesystem_validate_all
 		exit $?
 		;;
-	stop) Filesystem_stop
-		rc=$?
+	stop)
 		rmon_notify "disabled" 300
+		Filesystem_stop
+		rc=$?
 		exit $rc
 		;;
 esac
diff --git a/heartbeat/LVM b/heartbeat/LVM
index 38092f9..893ece8 100755
--- a/heartbeat/LVM
+++ b/heartbeat/LVM
@@ -599,9 +599,10 @@ case "$1" in
 		fi
 		exit $rc;;
 
-	stop)	LVM_stop $VOLUME
-		rc=$?
+	stop)
 		rmon_notify "disabled" 300
+		LVM_stop $VOLUME
+		rc=$?
 		exit $rc;;
 
 	status)	LVM_status $VOLUME $1
-- 
2.7.4

