From f658722ec239a258cd3ec2e828ca3f37ba2fe536 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Fri, 17 Jun 2016 00:31:20 -0400
Subject: [PATCH 1/1] Notify rmon of shutdown before shutting down LVM and
 Filesystem

---
 heartbeat/Filesystem | 9 +++++----
 heartbeat/LVM        | 9 +++++----
 2 files changed, 10 insertions(+), 8 deletions(-)

diff --git a/heartbeat/Filesystem b/heartbeat/Filesystem
index 05e4097..d5f3417 100755
--- a/heartbeat/Filesystem
+++ b/heartbeat/Filesystem
@@ -1200,10 +1200,11 @@ case $OP in
 	validate-all) Filesystem_validate_all
 		exit $?
 		;;
-	stop) Filesystem_stop
-        rc=$?
-        rmon_notify "disabled" 300
-        exit $rc
+	stop)
+		rmon_notify "disabled" 300
+		Filesystem_stop
+		rc=$?
+		exit $rc
 		;;
 esac
 
diff --git a/heartbeat/LVM b/heartbeat/LVM
index 38092f9..893ece8 100755
--- a/heartbeat/LVM
+++ b/heartbeat/LVM
@@ -879,10 +879,11 @@ case "$1" in
         fi
         exit $rc;;
 
-	stop)	LVM_stop $VOLUME
-        rc=$?
-        rmon_notify "disabled" 300
-        exit $rc;;
+	stop)
+		rmon_notify "disabled" 300
+		LVM_stop $VOLUME
+		rc=$?
+		exit $rc;;
 
 	status)	LVM_status $VOLUME $1
         rc=$?
-- 
2.7.4

