From 1b4403d3c5b9fc0f73e1af70e34deb2d54a9c686 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 14:55:36 -0400
Subject: WRS: spec-use-su-file-from-shadow.patch

---
 SPECS/util-linux.spec | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/SPECS/util-linux.spec b/SPECS/util-linux.spec
index 48100e4..4b045fd 100644
--- a/SPECS/util-linux.spec
+++ b/SPECS/util-linux.spec
@@ -41,6 +41,7 @@ Source12: util-linux-su.pamd
 Source13: util-linux-su-l.pamd
 Source14: util-linux-runuser.pamd
 Source15: util-linux-runuser-l.pamd
+Source16: su
 
 ### Obsoletes & Conflicts & Provides
 Conflicts: bash-completion < 1:2.1-1
@@ -612,7 +613,7 @@ chmod 755 ${RPM_BUILD_ROOT}%{_bindir}/sunhostid
 	install -m 644 %{SOURCE2} ./remote
 	install -m 644 %{SOURCE3} ./chsh
 	install -m 644 %{SOURCE3} ./chfn
-	install -m 644 %{SOURCE12} ./su
+	install -m 644 %{SOURCE16} ./su
 	install -m 644 %{SOURCE13} ./su-l
 	install -m 644 %{SOURCE14} ./runuser
 	install -m 644 %{SOURCE15} ./runuser-l
-- 
2.7.4

