From 4e88d0be6ebdd48f3d66840de5f142a660b8045a Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:11:21 -0400
Subject: [PATCH 2/5] WRS: openldap-service-file.patch

---
 SOURCES/slapd.service | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/SOURCES/slapd.service b/SOURCES/slapd.service
index 8a3a722..24b3938 100644
--- a/SOURCES/slapd.service
+++ b/SOURCES/slapd.service
@@ -1,6 +1,7 @@
 [Unit]
 Description=OpenLDAP Server Daemon
-After=syslog.target network-online.target
+Before=rsyncd.service
+After=network.target syslog-ng.target
 Documentation=man:slapd
 Documentation=man:slapd-config
 Documentation=man:slapd-hdb
@@ -9,11 +10,14 @@ Documentation=file:///usr/share/doc/openldap-servers/guide.html
 
 [Service]
 Type=forking
-PIDFile=/var/run/openldap/slapd.pid
+PIDFile=/var/run/slapd.pid
 Environment="SLAPD_URLS=ldap:/// ldapi:///" "SLAPD_OPTIONS="
 EnvironmentFile=/etc/sysconfig/slapd
 ExecStartPre=/usr/libexec/openldap/check-config.sh
-ExecStart=/usr/sbin/slapd -u ldap -h ${SLAPD_URLS} $SLAPD_OPTIONS
+ExecStart=/etc/init.d/openldap start
+ExecStop=/etc/init.d/openldap stop
+ExecReload=/etc/init.d/openldap restart
+RemainAfterExit=yes
 
 [Install]
 WantedBy=multi-user.target
-- 
1.9.1

