From 31cec68ea9c887b363dc31255fb96e88c67c4b63 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:11:21 -0400
Subject: [PATCH] WRS: openldap-spec-file.patch

---
 SPECS/openldap.spec | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/SPECS/openldap.spec b/SPECS/openldap.spec
index 85abac9..66a1377 100644
--- a/SPECS/openldap.spec
+++ b/SPECS/openldap.spec
@@ -296,7 +296,7 @@ mv check_password.so check_password.so.%{check_password_version}
 ln -s check_password.so.%{check_password_version} %{buildroot}%{_libdir}/openldap/check_password.so
 install -m 755 check_password.so.%{check_password_version} %{buildroot}%{_libdir}/openldap/
 # install -m 644 README %{buildroot}%{_libdir}/openldap
-install -d -m 755 %{buildroot}%{_sysconfdir}/openldap
+install -d -m 740 %{buildroot}%{_sysconfdir}/openldap
 cat > %{buildroot}%{_sysconfdir}/openldap/check_password.conf <<EOF
 # OpenLDAP pwdChecker library configuration
 
@@ -432,6 +432,7 @@ exit 0
 %post servers
 
 /sbin/ldconfig -n %{_libdir}/openldap
+mkdir -p /var/lib/openldap-data
 
 %systemd_post slapd.service
 
@@ -442,7 +443,6 @@ exit 0
 if [ ! -f %{_sysconfdir}/openldap/slapd.d/cn=config.ldif ]; then
 	if [ -f %{_sysconfdir}/openldap/slapd.conf ]; then
 		%{_libexecdir}/openldap/convert-config.sh &>/dev/null
-		mv %{_sysconfdir}/openldap/slapd.conf %{_sysconfdir}/openldap/slapd.conf.bak
 	else
 		%{_libexecdir}/openldap/convert-config.sh -f %{_datadir}/openldap-servers/slapd.ldif &>/dev/null
 	fi
@@ -641,8 +641,8 @@ exit 0
 %{_mandir}/man5/slapd*.5*
 %{_mandir}/man5/slapo-*.5*
 # obsolete configuration
-%ghost %config(noreplace,missingok) %attr(0640,ldap,ldap) %{_sysconfdir}/openldap/slapd.conf
-%ghost %config(noreplace,missingok) %attr(0640,ldap,ldap) %{_sysconfdir}/openldap/slapd.conf.bak
+# %ghost %config(noreplace,missingok) %attr(0640,ldap,ldap) %{_sysconfdir}/openldap/slapd.conf
+# %ghost %config(noreplace,missingok) %attr(0640,ldap,ldap) %{_sysconfdir}/openldap/slapd.conf.bak
 
 %files servers-sql
 %doc openldap-%{version}/servers/slapd/back-sql/docs/*
-- 
2.7.4

