From 388d312346bb408149f97b58287089a6754944c3 Mon Sep 17 00:00:00 2001
From: Saju Oommen <saju.oommen@windriver.com>
Date: Mon, 15 Jan 2018 13:59:26 -0500
Subject: [PATCH] remove-ldap-conf-cgcs-file

---
 SPECS/openldap.spec | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/SPECS/openldap.spec b/SPECS/openldap.spec
index 468ca0e..c38f1bd 100644
--- a/SPECS/openldap.spec
+++ b/SPECS/openldap.spec
@@ -26,7 +26,6 @@ Source55: libexec-generate-server-cert.sh
 Source56: libexec-update-ppolicy-schema.sh
 
 # WRS:
-Source100: ldap.conf.cgcs
 Source101: slapd.conf.cgcs
 Source102: initial_config.ldif
 Source103: initscript
@@ -417,7 +416,7 @@ rm -f %{buildroot}%{_localstatedir}/openldap-data/DB_CONFIG.example
 rmdir %{buildroot}%{_localstatedir}/openldap-data
 
 # WRS: slapd-config is backward compatible with slapd.conf
-install -m 600 %{SOURCE100} %{buildroot}%{_sysconfdir}/openldap/ldap.conf
+# WRS: SOURCE100 (ldap.conf.cgcs) is replaced by puppet template
 install -m 600 %{SOURCE101} %{buildroot}%{_sysconfdir}/openldap/slapd.conf
 install -m 600 %{SOURCE102} %{buildroot}%{_sysconfdir}/openldap/initial_config.ldif
 
-- 
2.7.4

