From c4d74c67ee001af849e7a30e824cc0f8e38ef948 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:12:36 -0400
Subject: [PATCH 5/7] WRS: haproxy-service-file.patch

---
 SOURCES/haproxy.service | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/SOURCES/haproxy.service b/SOURCES/haproxy.service
index 2d4c954..c2f1086 100644
--- a/SOURCES/haproxy.service
+++ b/SOURCES/haproxy.service
@@ -4,7 +4,8 @@ After=syslog.target network.target
 
 [Service]
 EnvironmentFile=/etc/sysconfig/haproxy
-ExecStart=/usr/sbin/haproxy-systemd-wrapper -f /etc/haproxy/haproxy.cfg -p /run/haproxy.pid $OPTIONS
+ExecStart=/etc/init.d/haproxy start
+ExecStop=/etc/init.d/haproxy stop
 ExecReload=/bin/kill -USR2 $MAINPID
 KillMode=mixed
 
-- 
1.9.1

