From a5329bf1468f55c8d6b983e5999c12139dc7479d Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:12:36 -0400
Subject: [PATCH 6/7] WRS: meta_add_support_for_tpm.patch

---
 SPECS/haproxy.spec | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/SPECS/haproxy.spec b/SPECS/haproxy.spec
index 3d112e0..c1547ef 100644
--- a/SPECS/haproxy.spec
+++ b/SPECS/haproxy.spec
@@ -30,6 +30,7 @@ Patch1:         iprange-return-type.patch
 Patch2:         haproxy-tcp-user-timeout.patch
 Patch3:         haproxy-systemd-wrapper-exit-code.patch
 Patch4:         haproxy-env-var.patch
+Patch5:         haproxy-tpm-support.patch
 
 BuildRequires:  pcre-devel
 BuildRequires:  zlib-devel
@@ -41,6 +42,9 @@ Requires(post):     systemd
 Requires(preun):    systemd
 Requires(postun):   systemd
 
+Requires: tpm2-openssl-engine
+
+
 %description
 HAProxy is a TCP/HTTP reverse proxy which is particularly suited for high
 availability environments. Indeed, it can:
@@ -62,6 +66,7 @@ availability environments. Indeed, it can:
 %patch2 -p1
 %patch3 -p1
 %patch4 -p1
+%patch5 -p1
 
 %build
 regparm_opts=
-- 
1.9.1

