From 3eac39ba534b92dbcb3a898442b09be7acc389bb Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:12:36 -0400
Subject: [PATCH] WRS: meta_remove_bad_logrotate.patch

Signed-off-by: zhipengl <zhipengs.liu@intel.com>
---
 SPECS/haproxy.spec | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/SPECS/haproxy.spec b/SPECS/haproxy.spec
index 53def97..39c0c86 100644
--- a/SPECS/haproxy.spec
+++ b/SPECS/haproxy.spec
@@ -18,7 +18,6 @@ URL:            http://www.haproxy.org/
 Source0:        http://www.haproxy.org/download/1.5/src/haproxy-%{version}.tar.gz
 Source1:        %{name}.service
 Source2:        %{name}.cfg
-Source3:        %{name}.logrotate
 Source4:        %{name}.sysconfig
 Source5:        halog.1
 
@@ -86,7 +85,6 @@ popd
 
 %{__install} -p -D -m 0644 %{SOURCE1} %{buildroot}%{_unitdir}/%{name}.service
 %{__install} -p -D -m 0640 %{SOURCE2} %{buildroot}%{haproxy_confdir}/%{name}.cfg
-%{__install} -p -D -m 0644 %{SOURCE3} %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
 %{__install} -p -D -m 0644 %{SOURCE4} %{buildroot}%{_sysconfdir}/sysconfig/%{name}
 %{__install} -p -D -m 0644 %{SOURCE5} %{buildroot}%{_mandir}/man1/halog.1
 %{__install} -d -m 0755 %{buildroot}%{haproxy_home}
@@ -139,7 +137,6 @@ fi
 %dir %{haproxy_datadir}
 %{haproxy_datadir}/*
 %config(noreplace) %{haproxy_confdir}/%{name}.cfg
-%config(noreplace) %{_sysconfdir}/logrotate.d/%{name}
 %config(noreplace) %{_sysconfdir}/sysconfig/%{name}
 %{_unitdir}/%{name}.service
 %{_sbindir}/%{name}
-- 
1.8.3.1

