From 0b326ac9fcee3fbac7e41e3bc269cc7251dcf9dd Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:12:36 -0400
Subject: WRS: spec-include-TiS-config.patch

---
 SPECS/haproxy.spec | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git a/SPECS/haproxy.spec b/SPECS/haproxy.spec
index 30dca25..988e7cf 100644
--- a/SPECS/haproxy.spec
+++ b/SPECS/haproxy.spec
@@ -22,6 +22,9 @@ Source3:        %{name}.logrotate
 Source4:        %{name}.sysconfig
 Source5:        halog.1
 
+# WRS
+Source6: 503.http
+
 Patch0:         halog-unused-variables.patch
 Patch1:         iprange-return-type.patch
 Patch2:         haproxy-tcp-user-timeout.patch
@@ -79,7 +82,7 @@ popd
 %{__make} install-man DESTDIR=%{buildroot} PREFIX=%{_prefix}
 
 %{__install} -p -D -m 0644 %{SOURCE1} %{buildroot}%{_unitdir}/%{name}.service
-%{__install} -p -D -m 0644 %{SOURCE2} %{buildroot}%{haproxy_confdir}/%{name}.cfg
+%{__install} -p -D -m 0640 %{SOURCE2} %{buildroot}%{haproxy_confdir}/%{name}.cfg
 %{__install} -p -D -m 0644 %{SOURCE3} %{buildroot}%{_sysconfdir}/logrotate.d/%{name}
 %{__install} -p -D -m 0644 %{SOURCE4} %{buildroot}%{_sysconfdir}/sysconfig/%{name}
 %{__install} -p -D -m 0644 %{SOURCE5} %{buildroot}%{_mandir}/man1/halog.1
@@ -106,6 +109,11 @@ do
     %{__rm} -f $textfile.old
 done
 
+# WRS
+%{__install} -d 755 %{buildroot}/etc/haproxy/errors/
+%{__install} -m 755 %{SOURCE6} %{buildroot}/etc/haproxy/errors/503.http
+
+
 %pre
 getent group %{haproxy_group} >/dev/null || groupadd -f -g 188 -r %{haproxy_group}
 if ! getent passwd %{haproxy_user} >/dev/null ; then
@@ -143,6 +151,10 @@ fi
 %{_mandir}/man1/*
 %attr(-,%{haproxy_user},%{haproxy_group}) %dir %{haproxy_home}
 
+# WRS
+%dir /etc/haproxy/errors/
+/etc/haproxy/errors/*
+
 %changelog
 * Thu Nov 30 2017 Ryan O'Hara <rohara@redhat.com> - 1.5.18-7
 - Rebuild with openssl-1.0.2k (#1509139)
-- 
2.7.4

