From 2e37207c026047e2ce1bc9a5278faddfea81c011 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:12:36 -0400
Subject: [PATCH 3/7] WRS: spec-add-haproxy-env-var-patch.patch

---
 SPECS/haproxy.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/haproxy.spec b/SPECS/haproxy.spec
index cbd9161..af94d46 100644
--- a/SPECS/haproxy.spec
+++ b/SPECS/haproxy.spec
@@ -30,6 +30,7 @@ Patch0:         halog-unused-variables.patch
 Patch1:         iprange-return-type.patch
 Patch2:         haproxy-tcp-user-timeout.patch
 Patch3:         haproxy-systemd-wrapper-exit-code.patch
+Patch4:         haproxy-env-var.patch
 
 BuildRequires:  pcre-devel
 BuildRequires:  zlib-devel
@@ -61,6 +62,7 @@ availability environments. Indeed, it can:
 %patch1 -p0
 %patch2 -p1
 %patch3 -p1
+%patch4 -p1
 
 %build
 regparm_opts=
-- 
1.9.1

