---
 support-files/wsrep.cnf.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/support-files/wsrep.cnf.sh b/support-files/wsrep.cnf.sh
index a539085..51ce3dc 100644
--- a/support-files/wsrep.cnf.sh
+++ b/support-files/wsrep.cnf.sh
@@ -30,6 +30,9 @@ bind-address=0.0.0.0
 ## WSREP options
 ##
 
+# Enable wsrep
+wsrep_on=1
+
 # Full path to wsrep provider library or 'none'
 wsrep_provider=none
 
-- 
1.9.1

