From a38da63c3677f78c33b3896699788bd5eb77116e Mon Sep 17 00:00:00 2001
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Tue, 6 Aug 2019 16:18:04 +0800
Subject: [PATCH 3/3] Compile issues

Signed-off-by: zhao.shuai <zhaos@neusoft.com>
---
 SPECS/kernel.spec | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index 418cdd0..a5de0aa 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -491,6 +491,12 @@ Patch40025: dpt_i2o-fix-build-warning.patch
 Patch40026: turn-off-write-same-in-smartqpi-driver.patch
 # Fix use-after-free in eventpoll_release_file
 Patch40027: epoll-fix-use-after-free-in-eventpoll_release_file.patch
+# Fix assorted compilation issues
+Patch40028: fix-compilation-issues.patch
+# Fix CentOS 7.6 upgrade compile error
+Patch40029: fix-CentOS-7.6-upgrade-compile-error.patch
+# Compile fix for disabling CONFIG_MEMCG_KMEM
+Patch40030: compile-fix-for-disabling-CONFIG_MEMCG_KMEM.patch
 
 BuildRoot: %{_tmppath}/kernel-%{KVRA}-root
 
@@ -862,6 +868,9 @@ ApplyOptionalPatch aic94xx-Skip-reading-user-settings-if-flash-is-not-f.patch
 ApplyOptionalPatch dpt_i2o-fix-build-warning.patch
 ApplyOptionalPatch turn-off-write-same-in-smartqpi-driver.patch
 ApplyOptionalPatch epoll-fix-use-after-free-in-eventpoll_release_file.patch
+ApplyOptionalPatch fix-compilation-issues.patch
+ApplyOptionalPatch fix-CentOS-7.6-upgrade-compile-error.patch
+ApplyOptionalPatch compile-fix-for-disabling-CONFIG_MEMCG_KMEM.patch
 
 # Any further pre-build tree manipulations happen here.
 
-- 
1.8.3.1

