From 6b9579fcfb774f20f114ebc621a925d35d3aa034 Mon Sep 17 00:00:00 2001
From: Bart Wensley <barton.wensley@windriver.com>
Date: Tue, 9 Jul 2019 06:36:33 -0500
Subject: [PATCH 1/1] Compile issues

Signed-off-by: Bart Wensley <barton.wensley@windriver.com>
---
 SPECS/kernel.spec | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index 3f774c2..b69967d 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -489,6 +489,12 @@ Patch40024: aic94xx-Skip-reading-user-settings-if-flash-is-not-f.patch
 Patch40025: dpt_i2o-fix-build-warning.patch
 # DRBD was choking on write same
 Patch40026: turn-off-write-same-in-smartqpi-driver.patch
+# Fix assorted compilation issues
+Patch40027: fix-compilation-issues.patch
+# Fix CentOS 7.6 upgrade compile error
+Patch40028: fix-CentOS-7.6-upgrade-compile-error.patch
+# Compile fix for disabling CONFIG_MEMCG_KMEM
+Patch40029: compile-fix-for-disabling-CONFIG_MEMCG_KMEM.patch
 
 BuildRoot: %{_tmppath}/kernel-%{KVRA}-root
 
@@ -859,6 +865,9 @@ ApplyOptionalPatch US103091-IMA-System-Configuration.patch
 ApplyOptionalPatch aic94xx-Skip-reading-user-settings-if-flash-is-not-f.patch
 ApplyOptionalPatch dpt_i2o-fix-build-warning.patch
 ApplyOptionalPatch turn-off-write-same-in-smartqpi-driver.patch
+ApplyOptionalPatch fix-compilation-issues.patch
+ApplyOptionalPatch fix-CentOS-7.6-upgrade-compile-error.patch
+ApplyOptionalPatch compile-fix-for-disabling-CONFIG_MEMCG_KMEM.patch
 
 # Any further pre-build tree manipulations happen here.
 
-- 
1.8.3.1

