From e49a8758922e1f23c4e77dd19cf4eb1f80263763 Mon Sep 17 00:00:00 2001
From: Bin Yang <bin.yang@intel.com>
Date: Wed, 31 Jul 2019 10:50:03 +0800
Subject: [PATCH 3/3] Compile issues

Signed-off-by: Bin Yang <bin.yang@intel.com>
---
 SPECS/kernel.spec | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index 9149019..b8fb9f9 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -491,6 +491,12 @@ Patch40025: dpt_i2o-fix-build-warning.patch
 Patch40026: turn-off-write-same-in-smartqpi-driver.patch
 # Fix use-after-free in eventpoll_release_file
 Patch40027: epoll-fix-use-after-free-in-eventpoll_release_file.patch
+# Fix assorted compilation issues
+Patch40028: fix-compilation-issues.patch
+# Fix CentOS 7.6 upgrade compile error
+Patch40029: fix-CentOS-7.6-upgrade-compile-error.patch
+# Compile fix for disabling CONFIG_MEMCG_KMEM
+Patch40030: compile-fix-for-disabling-CONFIG_MEMCG_KMEM.patch
 
 BuildRoot: %{_tmppath}/kernel-%{KVRA}-root
 
@@ -862,6 +868,9 @@ ApplyOptionalPatch aic94xx-Skip-reading-user-settings-if-flash-is-not-f.patch
 ApplyOptionalPatch dpt_i2o-fix-build-warning.patch
 ApplyOptionalPatch turn-off-write-same-in-smartqpi-driver.patch
 ApplyOptionalPatch epoll-fix-use-after-free-in-eventpoll_release_file.patch
+ApplyOptionalPatch fix-compilation-issues.patch
+ApplyOptionalPatch fix-CentOS-7.6-upgrade-compile-error.patch
+ApplyOptionalPatch compile-fix-for-disabling-CONFIG_MEMCG_KMEM.patch
 
 # Any further pre-build tree manipulations happen here.
 
-- 
2.7.4

