From ac67546ccbecd3ee18145be31d87d7253048adda Mon Sep 17 00:00:00 2001
Message-Id: <ac67546ccbecd3ee18145be31d87d7253048adda.1566587341.git.Jim.Somerville@windriver.com>
From: "zhao.shuai" <zhaos@neusoft.com>
Date: Mon, 5 Aug 2019 17:55:01 +0800
Subject: [PATCH 1/2] Kernel-source-patches-for-TiC

Signed-off-by: zhao.shuai <zhaos@neusoft.com>
Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
Signed-off-by: Robin Lu <bin1.lu@intel.com>
Signed-off-by: Zhang Zhiguo <zhangzhg@neusoft.com>

---
 SPECS/kernel.spec | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index 1927d5a..a851036 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -433,6 +433,17 @@ Patch1002: debrand-rh-i686-cpu.patch
 # empty final patch to facilitate testing of kernel patches
 Patch999999: linux-kernel-test.patch
 
+# StarlingX Cloud patches here.
+Patch40002: Notification-of-death-of-arbitrary-processes.patch
+Patch40004: PCI-Add-ACS-quirk-for-Intel-Fortville-NICs.patch
+Patch40005: affine-compute-kernel-threads.patch
+Patch40006: Affine-irqs-and-workqueues-with-kthread_cpus.patch
+Patch40007: Make-kernel-start-eth-devices-at-offset.patch
+Patch40008: intel-iommu-allow-ignoring-Ethernet-device-RMRR-with.patch
+Patch40022: US101216-IMA-support-in-Titanium-kernel.patch
+Patch40023: US103091-IMA-System-Configuration.patch
+Patch40026: turn-off-write-same-in-smartqpi-driver.patch
+
 # END OF PATCH DEFINITIONS
 
 BuildRoot: %{_tmppath}/kernel-%{KVERREL}-root
@@ -900,6 +911,17 @@ ApplyOptionalPatch debrand-single-cpu.patch
 ApplyOptionalPatch debrand-rh_taint.patch
 ApplyOptionalPatch debrand-rh-i686-cpu.patch
 
+# StarlingX Cloud patches here.
+ApplyOptionalPatch Notification-of-death-of-arbitrary-processes.patch
+ApplyOptionalPatch PCI-Add-ACS-quirk-for-Intel-Fortville-NICs.patch
+ApplyOptionalPatch affine-compute-kernel-threads.patch
+ApplyOptionalPatch Affine-irqs-and-workqueues-with-kthread_cpus.patch
+ApplyOptionalPatch Make-kernel-start-eth-devices-at-offset.patch
+ApplyOptionalPatch intel-iommu-allow-ignoring-Ethernet-device-RMRR-with.patch
+ApplyOptionalPatch US101216-IMA-support-in-Titanium-kernel.patch
+ApplyOptionalPatch US103091-IMA-System-Configuration.patch
+ApplyOptionalPatch turn-off-write-same-in-smartqpi-driver.patch
+
 # END OF PATCH APPLICATIONS
 
 # Any further pre-build tree manipulations happen here.
-- 
2.18.1

