From c1099550e41a13e84ed2822a0826bae2e9bcb9d9 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 20 Mar 2017 11:44:56 -0400
Subject: [PATCH 2/2] WRS: 0002-Add-WRS-patch.patch

---
 SPECS/puppet-oslo.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/puppet-oslo.spec b/SPECS/puppet-oslo.spec
index 2b89be7..d2d6331 100644
--- a/SPECS/puppet-oslo.spec
+++ b/SPECS/puppet-oslo.spec
@@ -9,6 +9,8 @@ URL:            https://launchpad.net/puppet-oslo
 
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 
+Patch0001:      0001-Remove-log_dir-from-conf-files.patch
+
 BuildArch:      noarch
 
 Requires:       puppet-inifile
@@ -22,6 +24,8 @@ Puppet module for managing Oslo.* library
 %prep
 %setup -q -n openstack-oslo-%{upstream_version}
 
+%patch0001 -p1
+
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
 find . \( -name "*.pl" -o -name "*.sh"  \) -exec chmod +x {} +
-- 
1.8.3.1

