From 4291fe6a18127e1169f4364b1f2e934ab06dc86b Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Thu, 14 Dec 2017 11:10:05 -0600
Subject: [PATCH] psycopg2 drivername for postgres

---
 SPECS/puppet-oslo.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-oslo.spec b/SPECS/puppet-oslo.spec
index 65bd995..9802ecc 100644
--- a/SPECS/puppet-oslo.spec
+++ b/SPECS/puppet-oslo.spec
@@ -10,6 +10,7 @@ URL:            https://launchpad.net/puppet-oslo
 Source0:        https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 
 Patch0001:      0001-Remove-log_dir-from-conf-files.patch
+Patch0002:      0002-add-psycopg2-drivername-to-postgresql-settings.patch
 
 BuildArch:      noarch
 
@@ -25,6 +26,7 @@ Puppet module for managing Oslo.* library
 %setup -q -n openstack-oslo-%{upstream_version}
 
 %patch0001 -p1
+%patch0002 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

