Index: keyring-5.3/keyring/backends/file.py
===================================================================
--- keyring-5.3.orig/keyring/backends/file.py
+++ keyring-5.3/keyring/backends/file.py
@@ -151,6 +151,7 @@ class BaseKeyring(FileBacked, KeyringBac
                  pass
              # must have the lock file with the correct group permissisions g+rw
              os.chmod(lockdir + "/" + lockfile, stat.S_IRWXG | stat.S_IRWXU)
+             os.chown(lockdir + "/" + lockfile,-1,345)
 
 
     def delete_password(self, service, username):
