---
 keyring/backends/file.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/keyring/backends/file.py
+++ b/keyring/backends/file.py
@@ -149,7 +149,7 @@ class BaseKeyring(FileBacked, KeyringBac
             # create the file without group/world permissions
             with open(self.file_path, 'w'):
                 pass
-            user_read_write = 0o644
+            user_read_write = 0o640
             os.chmod(self.file_path, user_read_write)
         if not os.path.isfile(lockdir + "/" + lockfile):
             with open(lockdir + "/" + lockfile, 'w'):
