From daeb87a5c097ad17ccbb90c6f15b9042bb991b7a Mon Sep 17 00:00:00 2001
From: Jerry Sun <jerry.sun@windriver.com>
Date: Mon, 8 Jan 2018 12:25:06 -0500
Subject: [PATCH 1/1] Added murano uid/gid to the group and passwd files

---
 group  | 1 +
 passwd | 1 +
 2 files changed, 2 insertions(+)

diff --git a/group b/group
index 9979b99..22fa91d 100644
--- a/group
+++ b/group
@@ -28,3 +28,4 @@ fm:x:195:fm
 libvirt:x:991:nova
 magnum:x:1870:magnum
 ironic:x:1874:ironic
+murano:x:105:murano
diff --git a/passwd b/passwd
index fb49ea3..dfb96b7 100644
--- a/passwd
+++ b/passwd
@@ -18,3 +18,4 @@ snmpd:x:169:169:net-snmp:/usr/share/snmp:/sbin/nologin
 fm:x:195:195:fm-mgr:/var/lib/fm:/sbin/nologin
 magnum:x:1870:1870:OpenStack Magnum Daemons:/var/lib/magnum:/sbin/nologin
 ironic:x:1874:1874:OpenStack Ironic Daemons:/var/lib/ironic:/sbin/nologin
+murano:x:105:105:OpenStack Murano Daemons:/home/murano:/sbin/nologin
-- 
1.8.3.1

