From b7fa16379bf880fb0fe5d36cd2dac9182176d433 Mon Sep 17 00:00:00 2001
From: Jerry Sun <jerry.sun@windriver.com>
Date: Thu, 3 Aug 2017 16:11:37 -0400
Subject: [PATCH 1/1] Added magnum uid/gid to the group and passwd files

---
 group  | 1 +
 passwd | 1 +
 2 files changed, 2 insertions(+)

diff --git a/group b/group
index 8924954..7d0244f 100644
--- a/group
+++ b/group
@@ -26,3 +26,4 @@ heat:x:187:heat
 nfv:x:172:nfv
 fm:x:195:fm
 libvirt:x:991:nova
+magnum:x:1870:magnum
diff --git a/passwd b/passwd
index 2fb16ee..fce82e7 100644
--- a/passwd
+++ b/passwd
@@ -16,3 +16,4 @@ nfv:x:172:172:nfvi:/var/lib/nfv:/sbin/nologin
 postgres:x:120:120:PostgreSQL Server:/var/lib/pgsql:/bin/sh
 snmpd:x:169:169:net-snmp:/usr/share/snmp:/sbin/nologin
 fm:x:195:195:fm-mgr:/var/lib/fm:/sbin/nologin
+magnum:x:1870:1870:OpenStack Magnum Daemons:/var/lib/magnum:/sbin/nologin
-- 
1.8.3.1

