From 15d0ef24f88290887f1e154352b53f373a04e783 Mon Sep 17 00:00:00 2001
From: slin14 <shuicheng.lin@intel.com>
Date: Thu, 16 Aug 2018 00:15:04 +0800
Subject: [PATCH] add-ironic-uid-gid

Signed-off-by: slin14 <shuicheng.lin@intel.com>
---
 group  | 1 +
 passwd | 1 +
 2 files changed, 2 insertions(+)

diff --git a/group b/group
index f6a75e5..d9050fa 100644
--- a/group
+++ b/group
@@ -27,3 +27,4 @@ nfv:x:172:nfv
 fm:x:195:fm
 libvirt:x:991:nova
 magnum:x:1870:magnum
+ironic:x:1874:ironic
diff --git a/passwd b/passwd
index fce82e7..fb49ea3 100644
--- a/passwd
+++ b/passwd
@@ -17,3 +17,4 @@ postgres:x:120:120:PostgreSQL Server:/var/lib/pgsql:/bin/sh
 snmpd:x:169:169:net-snmp:/usr/share/snmp:/sbin/nologin
 fm:x:195:195:fm-mgr:/var/lib/fm:/sbin/nologin
 magnum:x:1870:1870:OpenStack Magnum Daemons:/var/lib/magnum:/sbin/nologin
+ironic:x:1874:1874:OpenStack Ironic Daemons:/var/lib/ironic:/sbin/nologin
-- 
2.7.4

