From c254a8d2bfdcd4509f84b34ca0d92068a2b28618 Mon Sep 17 00:00:00 2001
From: Bart Wensley <barton.wensley@windriver.com>
Date: Wed, 29 Jun 2016 08:15:32 -0400
Subject: [PATCH 1/1] updating gids and uids to support upgrade from wrl

---
 group  | 1 +
 passwd | 9 +++++----
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/group b/group
index 3a0560d..c21e2de 100644
--- a/group
+++ b/group
@@ -22,6 +22,7 @@ lock::54:
 audio::63:
 nobody::99:
 users::100:
+postgres:x:120:
 rabbitmq:x:121:
 glance:x:161:glance
 nova:x:162:nova
diff --git a/passwd b/passwd
index 3f28c4e..548435f 100644
--- a/passwd
+++ b/passwd
@@ -12,12 +12,13 @@ games:*:12:100:games:/usr/games:/sbin/nologin
 ftp:*:14:50:FTP User:/var/ftp:/sbin/nologin
 nobody:*:99:99:Nobody:/:/sbin/nologin
 rabbitmq:x:121:121::/var/lib/rabbitmq:/bin/sh
-nova:x:994:992::/var/lib/nova:/bin/false
-keystone:x:993:991::/home/keystone:/bin/sh
+nova:x:994:162::/var/lib/nova:/bin/false
+keystone:x:993:163::/home/keystone:/bin/sh
 glance:x:161:161:OpenStack Glance Daemons:/opt/cgcs/glance:/sbin/nologin
 cinder:x:165:165:OpenStack Cinder Daemons:/var/lib/cinder:/sbin/nologin
 sysinv:x:168:168:sysinv Daemons:/var/lib/sysinv:/sbin/nologin
 neutron:x:164:164:OpenStack Neutron Daemons:/var/lib/neutron:/sbin/nologin
-heat:x:992:990::/home/heat:/bin/sh
-ceilometer:x:991:989::/home/ceilometer:/bin/sh
+heat:x:992:187::/home/heat:/bin/sh
+ceilometer:x:991:166::/home/ceilometer:/bin/sh
 nfv:x:172:172:nfvi:/var/lib/nfv:/sbin/nologin
+postgres:x:120:120:PostgreSQL Server:/var/lib/pgsql:/bin/sh
-- 
1.8.3.1

