From 858e259bf7125695c068301d0ef56cc4750d6544 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Thu, 15 Sep 2016 13:32:03 -0400
Subject: [PATCH 1/1] Close tftp sockets immediately

---
 src/tftp.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/tftp.c b/src/tftp.c
index d7d050f..ecc6ce8 100644
--- a/src/tftp.c
+++ b/src/tftp.c
@@ -768,9 +768,11 @@ int do_tftp_script_run(void)
   if ((transfer = daemon->tftp_done_trans))
     {
       daemon->tftp_done_trans = transfer->next;
+#if 0 /* Disable delayed closing of TFTP UDP socket */
 #ifdef HAVE_SCRIPT
       queue_tftp(transfer->file->size, transfer->file->filename, &transfer->peer);
 #endif
+#endif
       free_transfer(transfer);
       return 1;
     }
-- 
1.9.1

