From a19b16baa019609714fb741db4e3c73d67f2adf1 Mon Sep 17 00:00:00 2001
From: jmckenna <jason.mckenna@windriver.com>
Date: Tue, 16 Jan 2018 08:14:08 -0500
Subject: [PATCH 1/2] Titanium release info

---
 SPECS/shim-signed.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/shim-signed.spec b/SPECS/shim-signed.spec
index d2a13b1..9cfcb2f 100644
--- a/SPECS/shim-signed.spec
+++ b/SPECS/shim-signed.spec
@@ -1,6 +1,6 @@
 Name:           shim-signed
 Version:        15
-Release:        1%{?dist}%{?buildid}
+Release:        1%{?_tis_dist}.%{tis_patch_ver}
 Summary:        First-stage UEFI bootloader
 %define unsigned_release 1%{?dist}
 
-- 
1.8.3.1

