From a19b16baa019609714fb741db4e3c73d67f2adf1 Mon Sep 17 00:00:00 2001
From: jmckenna <jason.mckenna@windriver.com>
Date: Tue, 16 Jan 2018 08:14:08 -0500
Subject: [PATCH 1/2] Titanium release info

---
 SPECS/shim.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/shim.spec b/SPECS/shim.spec
index 4296515..f004748 100644
--- a/SPECS/shim.spec
+++ b/SPECS/shim.spec
@@ -6,7 +6,7 @@
 
 Name:		shim
 Version:	15
-Release:	8%{?dist}
+Release:	8%{?_tis_dist}.%{tis_patch_ver}
 Summary:	First-stage UEFI bootloader
 License:	BSD
 URL:		https://github.com/rhboot/shim/
-- 
2.7.4

