diff --git a/SPECS/shim.spec b/SPECS/shim.spec
index f004748..1fd493c 100644
--- a/SPECS/shim.spec
+++ b/SPECS/shim.spec
@@ -19,6 +19,9 @@ ExcludeArch:	%{ix86}
 # and we don't have shim-unsigned-arm builds *yet*
 ExcludeArch:	%{arm}
 
+%global srcbasename shimx64
+%global srcbasenameia32 shimia32
+
 Source0:	shim.rpmmacros
 Source1:	centossecureboot001.crt
 Source2:	centos-ca-secureboot.der
@@ -28,9 +31,9 @@ Source2:	centos-ca-secureboot.der
 Source10:	BOOTAA64.CSV
 Source20:	shimaa64.efi
 Source11:	BOOTIA32.CSV
-Source21:	shimia32.efi
+Source21:	%{srcbasenameia32}.efi
 Source12:	BOOTX64.CSV
-Source22:	shimx64.efi
+Source22:	%{srcbasename}.efi
 #Source13:	BOOTARM.CSV
 #Source23:	shimarm.efi
 
@@ -43,11 +46,11 @@ BuildRequires:	pesign >= 0.112-20.fc27
 # (shim 16+) by making the unsigned packages all provide "shim-unsigned", so
 # we can just BuildRequires that.
 %ifarch x86_64
-BuildRequires:	%{unsignedx64} = %{shimverx64}
-BuildRequires:	%{unsignedia32} = %{shimveria32}
+BuildRequires:	%{unsignedx64}
+BuildRequires:	%{unsignedia32}
 %endif
 %ifarch aarch64
-BuildRequires:	%{unsignedaa64} = %{shimveraa64}
+BuildRequires:	%{unsignedaa64}
 %endif
 #%%ifarch arm
 #BuildRequires:	%%{unsignedarm} = %%{shimverarm}
-- 
2.7.4

