From a1d4efe436a005962d677a146a261ac175b6b55f Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Wed, 1 Nov 2017 14:35:36 -0500
Subject: [PATCH] WRS: 0001-Update-package-versioning-for-TIS-format.patch

---
 SPECS/puppet-rabbitmq.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-rabbitmq.spec b/SPECS/puppet-rabbitmq.spec
index 987ba3b..dc2c81b 100644
--- a/SPECS/puppet-rabbitmq.spec
+++ b/SPECS/puppet-rabbitmq.spec
@@ -14,7 +14,7 @@
 
 Name:           puppet-rabbitmq
 Version:        5.6.0
-Release:        4%{?alphatag}%{?dist}
+Release:        4.5ac45degit.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Installs, configures, and manages RabbitMQ.
 License:        ASL 2.0
 
-- 
1.8.3.1

