From fbb9b6275c4c199643d51bf6d192ae7623dcef99 Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Wed, 24 Jan 2018 16:04:04 -0600
Subject: [PATCH 4/4] metapatch for revert patch

---
 SPECS/puppet-rabbitmq.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/puppet-rabbitmq.spec b/SPECS/puppet-rabbitmq.spec
index e2a29cb..17a12bc 100644
--- a/SPECS/puppet-rabbitmq.spec
+++ b/SPECS/puppet-rabbitmq.spec
@@ -24,6 +24,7 @@ Source0:        https://github.com/puppetlabs/%{upstream_name}/archive/%{commit}
 Patch0001:      0001-Roll-up-TIS-patches.patch
 Patch0002:      0002-Changed-cipher-specification-to-openssl-format.patch
 Patch0003:      0003-Eliminate-Puppet-4-deprecation-warnings.patch
+Patch0004:      0004-Partially-revert-upstream-commit-f7c3a4a637d59f3065d.patch
 
 BuildArch:      noarch
 
@@ -39,6 +40,7 @@ Installs, configures, and manages RabbitMQ.
 %patch0001 -p1
 %patch0002 -p1
 %patch0003 -p1
+%patch0004 -p1
 
 
 find . -type f -name ".*" -exec rm {} +
-- 
1.8.3.1

