From d8ab20a896750f9fcd257d3e64cb0ef34e35140a Mon Sep 17 00:00:00 2001
From: David Balme <david.balme@windriver.com>
Date: Thu, 13 Oct 2016 08:40:27 -0400
Subject: [PATCH 10/15] add TMOUT variable

---
 SPECS/setup.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 8f5fc46..a6465dc 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -8,6 +8,7 @@ URL: https://pagure.io/setup/
 Source0: http://releases.pagure.org/%{name}/%{name}-%{version}.tar.bz2
 Source1: motd
 Source2: prompt.sh
+Source3: custom.sh
 BuildArch: noarch
 BuildRequires: bash tcsh perl
 #require system release for saner dependency order
@@ -73,8 +74,9 @@ touch %{buildroot}/etc/subuid
 touch %{buildroot}/etc/subgid
 chmod 0600 %{buildroot}/etc/{exports,fstab}
 install -m 644 %{SOURCE1} %{buildroot}/etc/
-install -m 644 %{SOURCE2} %{buildroot}/etc/profile.d/prompt.sh
 mkdir -p %{buildroot}/etc/profile.d
+install -m 644 %{SOURCE2} %{buildroot}/etc/profile.d/prompt.sh
+install -m 644 %{SOURCE3} %{buildroot}/etc/profile.d/custom.sh
 echo "#Add any required envvar overrides to this file, it is sourced from /etc/profile" >%{buildroot}/etc/profile.d/sh.local
 echo "#Add any required envvar overrides to this file, is sourced from /etc/csh.login" >%{buildroot}/etc/profile.d/csh.local
 
@@ -133,6 +135,7 @@ end
 %config(noreplace) /etc/motd
 %dir /etc/profile.d
 /etc/profile.d/prompt.sh
+/etc/profile.d/custom.sh
 %config(noreplace) /etc/profile.d/sh.local
 %config(noreplace) /etc/profile.d/csh.local
 %config(noreplace) %verify(not md5 size mtime) /etc/shells
-- 
2.7.4

