From 13bee9ed7d91fae3d66f91d4e4aa139ca3d05f66 Mon Sep 17 00:00:00 2001
From: David Balme <david.balme@windriver.com>
Date: Thu, 13 Oct 2016 08:40:27 -0400
Subject: [PATCH 1/1] add TMOUT variable

---
 SPECS/setup.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 89a4d2f..1f5c96a 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -8,6 +8,7 @@ URL: https://fedorahosted.org/setup/
 Source0: https://fedorahosted.org/releases/s/e/%{name}/%{name}-%{version}.tar.bz2
 Source1: motd
 Source2: prompt.sh
+Source3: custom.sh
 BuildArch: noarch
 BuildRequires: bash tcsh perl
 #require system release for saner dependency order
@@ -70,6 +71,7 @@ touch %{buildroot}/etc/subgid
 chmod 0600 %{buildroot}/etc/{exports,fstab}
 install -m 644 %{SOURCE1} %{buildroot}/etc/
 install -m 644 %{SOURCE2} %{buildroot}/etc/profile.d/prompt.sh
+install -m 644 %{SOURCE3} %{buildroot}/etc/profile.d/custom.sh
 
 # remove unpackaged files from the buildroot
 rm -f %{buildroot}/etc/Makefile
@@ -125,6 +127,7 @@ end
 %config(noreplace) /etc/motd
 %dir /etc/profile.d
 /etc/profile.d/prompt.sh
+/etc/profile.d/custom.sh
 %config(noreplace) %verify(not md5 size mtime) /etc/shells
 %ghost %attr(0644,root,root) %verify(not md5 size mtime) /var/log/lastlog
 %ghost %verify(not md5 size mtime) %config(noreplace,missingok) /etc/fstab
-- 
1.8.3.1

