From 22d06a6c5c7b44db10060bf95b623dc3c1943a9a Mon Sep 17 00:00:00 2001
From: Michel Thebeau <michel.thebeau@windriver.com>
Date: Fri, 19 Aug 2016 09:28:43 -0400
Subject: [PATCH 08/15] spec: add patch to remove unused groups

Signed-off-by: Michel Thebeau <michel.thebeau@windriver.com>
---
 SPECS/setup.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index a612d24..3ad2458 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -26,6 +26,7 @@ Patch8: tis-uid-gid.patch
 Patch9: updating-gids-and-uids-to-support-upgrade-from-wrl.patch
 Patch10: passwd-remove-unused-default-users-and-groups.patch
 Patch11: snmpd-fm-user-group.patch
+Patch12: remove-unused-default-groups.patch
 
 %description
 The setup package contains a set of important system configuration and
@@ -45,6 +46,7 @@ setup files, such as passwd, group, and profile.
 %patch9 -p1
 %patch10 -p1
 %patch11 -p1
+%patch12 -p1
 
 ./shadowconvert.sh
 
-- 
2.7.4

