From f0882fad769c71cb70b44251c79f39e7e13dbc48 Mon Sep 17 00:00:00 2001
From: Kam Nasim <kam.nasim@windriver.com>
Date: Fri, 12 Aug 2016 17:35:28 -0400
Subject: [PATCH 06/15] meta patch for snmpd-user-group.patch

---
 SPECS/setup.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 10d151b..db3ed17 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -25,6 +25,7 @@ Patch7: setup-2.8.71-shlocal.patch
 Patch8: tis-uid-gid.patch
 Patch9: updating-gids-and-uids-to-support-upgrade-from-wrl.patch
 Patch10: passwd-remove-unused-default-users-and-groups.patch
+Patch11: snmpd-fm-user-group.patch
 
 %description
 The setup package contains a set of important system configuration and
@@ -43,6 +44,7 @@ setup files, such as passwd, group, and profile.
 %patch8 -p1
 %patch9 -p1
 %patch10 -p1
+%patch11 -p1
 
 ./shadowconvert.sh
 
-- 
2.7.4

