From 1285d1381237a94df55df913aa268cd5bb9c6b89 Mon Sep 17 00:00:00 2001
From: Michel Thebeau <michel.thebeau@windriver.com>
Date: Thu, 21 Jul 2016 11:47:55 -0400
Subject: [PATCH 05/15] spec: add patch to remove unused users and groups

Signed-off-by: Michel Thebeau <michel.thebeau@windriver.com>
---
 SPECS/setup.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 18283cd..10d151b 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -24,6 +24,7 @@ Patch6: setup-2.8.71-tapeid.patch
 Patch7: setup-2.8.71-shlocal.patch
 Patch8: tis-uid-gid.patch
 Patch9: updating-gids-and-uids-to-support-upgrade-from-wrl.patch
+Patch10: passwd-remove-unused-default-users-and-groups.patch
 
 %description
 The setup package contains a set of important system configuration and
@@ -41,6 +42,7 @@ setup files, such as passwd, group, and profile.
 %patch7 -p1
 %patch8 -p1
 %patch9 -p1
+%patch10 -p1
 
 ./shadowconvert.sh
 
-- 
2.7.4

