commit f944ef677dc090e91b790ac54064d61d071edb5c
Author: Shoaib Nasir <shoaib.nasir@windriver.com>
Date:   Mon Sep 25 12:20:43 2017 -0400

    Add ironic-uid-gid.patch to SPECS

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 2ec3541..55dd30b 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -28,6 +28,7 @@ Patch9: snmpd-fm-user-group.patch
 Patch10: remove-unused-default-groups.patch
 Patch11: add-fm-user-to-snmpd-group.patch
 Patch12: add-magnum-uid-gid.patch
+Patch13: add-ironic-uid-gid.patch
 
 %description
 The setup package contains a set of important system configuration and
@@ -48,6 +49,7 @@ setup files, such as passwd, group, and profile.
 %patch10 -p1
 %patch11 -p1
 %patch12 -p1
+%patch13 -p1
 
 ./shadowconvert.sh
 
