From 11086bd4422e8f24a0b070eb16e53b08f4561c61 Mon Sep 17 00:00:00 2001
From: Jerry Sun <jerry.sun@windriver.com>
Date: Thu, 3 Aug 2017 16:18:34 -0400
Subject: [PATCH 1/1] meta add magnum uid and gid

---
 SPECS/setup.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 9ee24ca..2ec3541 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -27,6 +27,7 @@ Patch8: passwd-remove-unused-default-users-and-groups.patch
 Patch9: snmpd-fm-user-group.patch
 Patch10: remove-unused-default-groups.patch
 Patch11: add-fm-user-to-snmpd-group.patch
+Patch12: add-magnum-uid-gid.patch
 
 %description
 The setup package contains a set of important system configuration and
@@ -46,6 +47,7 @@ setup files, such as passwd, group, and profile.
 %patch9 -p1
 %patch10 -p1
 %patch11 -p1
+%patch12 -p1
 
 ./shadowconvert.sh
 
-- 
1.8.3.1

