From f882ce44d7e8574e9affc5e6471265029f9724ca Mon Sep 17 00:00:00 2001
From: Michel Thebeau <michel.thebeau@windriver.com>
Date: Thu, 21 Jul 2016 11:47:55 -0400
Subject: [PATCH] spec: add patch to remove unused users and groups

Signed-off-by: Michel Thebeau <michel.thebeau@windriver.com>
---
 SPECS/setup.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 3debacf..89ff683 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -22,6 +22,7 @@ Patch4: setup-2.8.71-filesystems.patch
 Patch5: setup-2.8.71-fullpath.patch
 Patch6: tis-uid-gid.patch
 Patch7: updating-gids-and-uids-to-support-upgrade-from-wrl.patch
+Patch8: passwd-remove-unused-default-users-and-groups.patch
 
 %description
 The setup package contains a set of important system configuration and
@@ -37,6 +38,7 @@ setup files, such as passwd, group, and profile.
 %patch5 -p1
 %patch6 -p1
 %patch7 -p1
+%patch8 -p1
 
 ./shadowconvert.sh
 
-- 
1.8.3.1

