From 236314295289b829e6216022a247017f7c0851c7 Mon Sep 17 00:00:00 2001
From: slin14 <shuicheng.lin@intel.com>
Date: Tue, 7 Aug 2018 22:41:01 +0800
Subject: [PATCH 01/15] setup.spec: to include Titanium Cloud specific changes

To include files under cgcs/recipes-base/setup/files/*

Signed-off-by: slin14 <shuicheng.lin@intel.com>
---
 SPECS/setup.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 317132b..4ac1019 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -6,6 +6,7 @@ License: Public Domain
 Group: System Environment/Base
 URL: https://pagure.io/setup/
 Source0: http://releases.pagure.org/%{name}/%{name}-%{version}.tar.bz2
+Source1: motd
 BuildArch: noarch
 BuildRequires: bash tcsh perl
 #require system release for saner dependency order
@@ -59,6 +60,7 @@ chmod 0644 %{buildroot}/var/log/lastlog
 touch %{buildroot}/etc/fstab
 touch %{buildroot}/etc/subuid
 touch %{buildroot}/etc/subgid
+install -m 644 %{SOURCE1} %{buildroot}/etc/
 mkdir -p %{buildroot}/etc/profile.d
 echo "#Add any required envvar overrides to this file, it is sourced from /etc/profile" >%{buildroot}/etc/profile.d/sh.local
 echo "#Add any required envvar overrides to this file, is sourced from /etc/csh.login" >%{buildroot}/etc/profile.d/csh.local
@@ -115,6 +117,7 @@ end
 %attr(0600,root,root) %config(noreplace,missingok) /etc/securetty
 %config(noreplace) /etc/csh.login
 %config(noreplace) /etc/csh.cshrc
+%config(noreplace) /etc/motd
 %dir /etc/profile.d
 %config(noreplace) /etc/profile.d/sh.local
 %config(noreplace) /etc/profile.d/csh.local
-- 
2.7.4

