From 3eb03183dc24b865dd3e84495a82899f39665690 Mon Sep 17 00:00:00 2001
From: slin14 <shuicheng.lin@intel.com>
Date: Tue, 7 Aug 2018 23:17:05 +0800
Subject: [PATCH 11/15] spec-include-add-fm-user-to-snmpd-group

Signed-off-by: slin14 <shuicheng.lin@intel.com>
---
 SPECS/setup.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index a6465dc..6a871f3 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -28,6 +28,7 @@ Patch9: updating-gids-and-uids-to-support-upgrade-from-wrl.patch
 Patch10: passwd-remove-unused-default-users-and-groups.patch
 Patch11: snmpd-fm-user-group.patch
 Patch12: remove-unused-default-groups.patch
+Patch13: add-fm-user-to-snmpd-group.patch
 
 %description
 The setup package contains a set of important system configuration and
@@ -48,6 +49,7 @@ setup files, such as passwd, group, and profile.
 %patch10 -p1
 %patch11 -p1
 %patch12 -p1
+%patch13 -p1
 
 ./shadowconvert.sh
 
-- 
2.7.4

