From 23dda8869b96f7df3db3fc885ca960b4889c82c0 Mon Sep 17 00:00:00 2001
From: Jerry Sun <jerry.sun@windriver.com>
Date: Thu, 3 Aug 2017 16:18:34 -0400
Subject: [PATCH 12/15] meta add magnum uid and gid

---
 SPECS/setup.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 6a871f3..367a13b 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -29,6 +29,7 @@ Patch10: passwd-remove-unused-default-users-and-groups.patch
 Patch11: snmpd-fm-user-group.patch
 Patch12: remove-unused-default-groups.patch
 Patch13: add-fm-user-to-snmpd-group.patch
+Patch14: add-magnum-uid-gid.patch
 
 %description
 The setup package contains a set of important system configuration and
@@ -50,6 +51,7 @@ setup files, such as passwd, group, and profile.
 %patch11 -p1
 %patch12 -p1
 %patch13 -p1
+%patch14 -p1
 
 ./shadowconvert.sh
 
-- 
2.7.4

