From e6d1a2794975f708a682a2094df65e14ddaa1eb0 Mon Sep 17 00:00:00 2001
From: Don Penney <dpenney@yow-dpenney-lx.corp.ad.wrs.com>
Date: Mon, 25 Apr 2016 13:15:13 -0400
Subject: [PATCH 02/15] WRS: spec-include-tis-uid-gid.patch

---
 SPECS/setup.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 4ac1019..72cbcba 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -21,6 +21,7 @@ Patch4: setup-2.8.71-filesystems.patch
 Patch5: setup-2.8.71-fullpath.patch
 Patch6: setup-2.8.71-tapeid.patch
 Patch7: setup-2.8.71-shlocal.patch
+Patch8: tis-uid-gid.patch
 
 %description
 The setup package contains a set of important system configuration and
@@ -36,6 +37,7 @@ setup files, such as passwd, group, and profile.
 %patch5 -p1
 %patch6 -p1
 %patch7 -p1
+%patch8 -p1
 
 ./shadowconvert.sh
 
-- 
2.7.4

