From 35ebbf2ca7e5e412f55cdaa875845728d203b34d Mon Sep 17 00:00:00 2001
From: Kam Nasim <kam.nasim@windriver.com>
Date: Fri, 12 Aug 2016 17:35:28 -0400
Subject: [PATCH] meta patch for snmpd-user-group.patch

---
 SPECS/setup.spec | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/SPECS/setup.spec b/SPECS/setup.spec
index 89ff683..d40113f 100644
--- a/SPECS/setup.spec
+++ b/SPECS/setup.spec
@@ -1,7 +1,7 @@
 Summary: A set of system configuration and setup files
 Name: setup
 Version: 2.8.71
-Release: 7%{?dist}
+Release: 8%{?dist}
 License: Public Domain
 Group: System Environment/Base
 URL: https://fedorahosted.org/setup/
@@ -23,6 +23,7 @@ Patch5: setup-2.8.71-fullpath.patch
 Patch6: tis-uid-gid.patch
 Patch7: updating-gids-and-uids-to-support-upgrade-from-wrl.patch
 Patch8: passwd-remove-unused-default-users-and-groups.patch
+Patch9: snmpd-fm-user-group.patch
 
 %description
 The setup package contains a set of important system configuration and
@@ -39,6 +40,7 @@ setup files, such as passwd, group, and profile.
 %patch6 -p1
 %patch7 -p1
 %patch8 -p1
+%patch9 -p1
 
 ./shadowconvert.sh
 
-- 
1.8.3.1

