From 772b57dd30d231773bb6e63d06a8d6ba3d37e7d6 Mon Sep 17 00:00:00 2001
From: Jerry Sun <jerry.sun@windriver.com>
Date: Tue, 30 Jan 2018 09:51:47 -0500
Subject: [PATCH 1/1] fixed puppet deprecated warnings

---
 manifests/init.pp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/manifests/init.pp b/manifests/init.pp
index 4f10c2f..a741a54 100644
--- a/manifests/init.pp
+++ b/manifests/init.pp
@@ -122,7 +122,7 @@ class magnum(
   $rabbit_userid              = $::os_service_default,
   $rabbit_virtual_host        = $::os_service_default,
   $rabbit_password            = $::os_service_default,
-  $rpc_backend                = 'rabbit',
+  $rpc_backend                = undef,
   # WRS
   $password_symbols    = '23456789,ABCDEFGHJKLMNPQRSTUVWXYZ,abcdefghijkmnopqrstuvwxyz',
 ) {
-- 
1.8.3.1

