From 2628193e8aef471caab27ada848fa8d7de6d93ec Mon Sep 17 00:00:00 2001
From: Daniel Badea <daniel.badea@windriver.com>
Date: Wed, 13 Jun 2018 14:16:53 +0000
Subject: [PATCH] drbd slow before swact

---
 manifests/resource/up.pp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/manifests/resource/up.pp b/manifests/resource/up.pp
index c3557e1..11cf7ee 100644
--- a/manifests/resource/up.pp
+++ b/manifests/resource/up.pp
@@ -23,7 +23,7 @@ define drbd::resource::up (
   # drbd is very tenacious about asking for aproval if there is data on the
   # volume already.
   exec { "initialize DRBD metadata for ${name}":
-    command => "yes yes | drbdadm create-md ${name}",
+    command => "yes yes | drbdadm create-md ${name} -W--peer-max-bio-size=128k",
     onlyif  => "test -e ${disk}",
     unless  => "drbdadm dump-md ${name} || (drbdadm cstate ${name} | egrep -q '^(PausedSync|Sync|Connected|WFConnection|StandAlone|Verify)') || (drbdadm show-gi ${name} | grep 'meta-data: need apply-al')",
     before  => Service['drbd'],
-- 
1.8.3.1

