From d43499ba2cdd43110e3f689243aecc7acce8a6b8 Mon Sep 17 00:00:00 2001
From: Ryan Tryhorn <ryan.tryhorn@windriver.com>
Date: Wed, 26 Apr 2017 17:43:43 -0400
Subject: [PATCH 1/1] Further parallelize shadow-utils build

---
 SPECS/shadow-utils.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/shadow-utils.spec b/SPECS/shadow-utils.spec
index dde526c..607247d 100644
--- a/SPECS/shadow-utils.spec
+++ b/SPECS/shadow-utils.spec
@@ -133,7 +133,7 @@ export LC_ALL=C
         --with-group-name-max-length=32
 # update the japanese translation
 (cd po; make ja.gmo)
-make
+make -j"%(nproc)"
 
 %install
 rm -rf $RPM_BUILD_ROOT
-- 
1.8.3.1

