From 6cb96c430ee1b3bb97450d16f3dee57331be8242 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Tue, 27 Sep 2016 10:52:27 -0400
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/shadow-utils.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/shadow-utils.spec b/SPECS/shadow-utils.spec
index d7e2ba9..1aacd11 100644
--- a/SPECS/shadow-utils.spec
+++ b/SPECS/shadow-utils.spec
@@ -1,7 +1,7 @@
 Summary: Utilities for managing accounts and shadow password files
 Name: shadow-utils
 Version: 4.1.5.1
-Release: 24%{?dist}
+Release: 24.el7%{?_tis_dist}.%{tis_patch_ver}
 Epoch: 2
 URL: http://pkg-shadow.alioth.debian.org/
 Source0: http://pkg-shadow.alioth.debian.org/releases/shadow-%{version}.tar.bz2
-- 
1.8.3.1

