From aa9e9fcaa006d87e706ed8fcd445108ab48bd479 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 21 Nov 2016 13:35:26 -0500
Subject: [PATCH 1/1] shadow-utils: add BuildRequires systemd to provide macro

---
 SPECS/shadow-utils.spec | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/SPECS/shadow-utils.spec b/SPECS/shadow-utils.spec
index 1aacd11..dde526c 100644
--- a/SPECS/shadow-utils.spec
+++ b/SPECS/shadow-utils.spec
@@ -46,6 +46,10 @@ BuildRequires: libsemanage-devel
 BuildRequires: libacl-devel libattr-devel
 BuildRequires: gnome-doc-utils docbook-style-xsl gettext
 #BuildRequires: autoconf, automake, libtool, gettext-devel
+
+# systemd provides %{_unitdir}
+BuildRequires: systemd
+
 Requires: libselinux >= 1.25.2-1
 Requires: audit-libs >= 1.6.5
 Requires: setup
-- 
1.8.3.1

