diff --git a/SPECS/python-keyring.spec b/SPECS/python-keyring.spec
index 09cda6c..c0e5e9c 100644
--- a/SPECS/python-keyring.spec
+++ b/SPECS/python-keyring.spec
@@ -43,6 +43,7 @@ Python keyring lib also provides following build-in keyrings.\
 Summary:        Python 2 library to access the system keyring service
 BuildRequires:  python2-devel
 BuildRequires:  python2-setuptools
+BuildRequires:  python2-setuptools_scm
 BuildRequires:  python2-keyczar
 BuildRequires:  python2-mock
 
@@ -71,19 +72,16 @@ Obsoletes:      %{name}-gnome < 0.6
 Summary:        Python 3 library to access the system keyring service
 BuildRequires:  python%{python3_pkgversion}-devel
 BuildRequires:  python%{python3_pkgversion}-setuptools
+BuildRequires:  python%{python3_pkgversion}-setuptools_scm
 %if 0%{?with_tests}
-%if 0%{?fedora} < 32
 BuildRequires:  python%{python3_pkgversion}-importlib-metadata
-%endif
 BuildRequires:  python%{python3_pkgversion}-pytest
 BuildRequires:  python%{python3_pkgversion}-pytest-cov
 BuildRequires:  python%{python3_pkgversion}-pytest-flakes
 %endif
 
 Requires:       python%{python3_pkgversion}-SecretStorage
-%if 0%{?fedora} < 32
 Requires:       python%{python3_pkgversion}-importlib-metadata
-%endif
 
 %{?python_provide:%python_provide python%{python3_pkgversion}-keyring}
 
-- 
2.7.4

