From 94704d9260d7057549b9f66461f7c2328f6080d3 Mon Sep 17 00:00:00 2001
Message-Id: <94704d9260d7057549b9f66461f7c2328f6080d3.1578602281.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Thu, 9 Jan 2020 15:36:55 -0500
Subject: [PATCH 1/1] Remove regmap mmio as it is built into the kernel

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 Makefile | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/Makefile b/Makefile
index 0dfd0a8..3386289 100644
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,6 @@ ccflags-y += -I$(PWD)/scripts/dtc/libfdt-4-12
 
 ifeq ($(kerval),3.10)
 	obj-m += spi-bitbang-mod.o
-	obj-m += regmap-mmio-mod.o
 endif
 
 obj-m += i2c-altera.o
@@ -45,8 +44,6 @@ intel-max10-y += lib-4-12/fdt_sw.o
 intel-max10-y += lib-4-12/fdt_strerror.o
 intel-max10-y += lib-4-12/fdt_empty_tree.o
 
-regmap-mmio-mod-y := drivers/base/regmap/regmap-mmio-3-10.o
-
 i2c-altera-y := drivers/i2c/busses/i2c-altera.o
 
 intel-generic-qspi-y := drivers/mtd/spi-nor/intel-generic-qspi.o
@@ -99,9 +96,6 @@ load-drv:
 	if ! egrep -q '^intel_generic_qspi ' /proc/modules; then insmod intel-generic-qspi.ko; fi
 	if ! egrep -q '^intel_on_chip_flash ' /proc/modules; then insmod intel-on-chip-flash.ko; fi
 	if ! egrep -q '^altera_asmip2 ' /proc/modules; then insmod altera-asmip2.ko; fi
-ifeq ($(kerval),3.10)
-	if ! egrep -q '^regmap_mmio_mod ' /proc/modules; then insmod regmap-mmio-mod.ko; fi
-endif
 	if ! egrep -q '^i2c_altera ' /proc/modules; then insmod i2c-altera.ko; fi
 ifeq ($(kerval),3.10)
 	if ! egrep -q '^spi_bitbang_mod ' /proc/modules; then insmod spi-bitbang-mod.ko; fi
@@ -137,9 +131,6 @@ ifeq ($(kerval),3.10)
 	if egrep -q '^spi_bitbang_mod ' /proc/modules; then rmmod spi_bitbang_mod; fi
 endif
 	if egrep -q '^i2c_altera ' /proc/modules; then rmmod i2c_altera; fi
-ifeq ($(kerval),3.10)
-	if egrep -q '^regmap_mmio_mod ' /proc/modules; then rmmod regmap_mmio_mod; fi
-endif
 	if egrep -q '^altera_asmip2 ' /proc/modules; then rmmod altera_asmip2; fi
 	if egrep -q '^intel_on_chip_flash ' /proc/modules; then rmmod intel_on_chip_flash; fi
 	if egrep -q '^intel_generic_qspi ' /proc/modules; then rmmod intel_generic_qspi; fi
-- 
1.8.3.1

