From 56181573a8d619e4685bd0023214724416eac91e Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Mon, 2 Oct 2017 09:24:33 -0500
Subject: [PATCH 1/1] Add WRS Patches

---
 SPECS/puppet.spec | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/SPECS/puppet.spec b/SPECS/puppet.spec
index 023f74f..284accf 100644
--- a/SPECS/puppet.spec
+++ b/SPECS/puppet.spec
@@ -32,6 +32,15 @@ Source4:        start-puppet-wrapper
 # Puppetlabs messed up with default paths
 Patch01:        0001-Fix-puppet-paths.patch
 Patch02:        0002-Revert-maint-Remove-puppetmaster.service.patch
+
+# WRS Patches
+Patch1001:      1001-Add-timestamps-to-logs.patch
+Patch1002:      1002-Set-hasstatus-to-false-by-default.patch
+Patch1003:      1003-Update-getpid-function.patch
+Patch1004:      1004-Block-enabling-of-services.patch
+Patch1005:      1005-Set-strict_variables-and-basemodulepath-in-puppet.co.patch
+
+
 Group:          System Environment/Base
 
 BuildRoot:      %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
@@ -119,6 +128,14 @@ The server can also function as a certificate authority and file server.
 %setup -q
 %patch01 -p1 -b .paths
 %patch02 -p1 -b .server
+
+%patch1001 -p1
+%patch1002 -p1
+%patch1003 -p1
+%patch1004 -p1
+%patch1005 -p1
+
+
 # Unbundle
 #rm -r lib/puppet/vendor/*{pathspec,rgen}*
 
-- 
1.8.3.1

