From ce598a7645d2b75b9d9485a2810917f8e3e188aa Mon Sep 17 00:00:00 2001
From: eric zhang <eric.zhang@windriver.com>
Date: Fri, 5 Jan 2018 22:24:21 -0500
Subject: [PATCH 1/1] driver update

Signed-off-by: eric zhang <eric.zhang@windriver.com>
---
 SPECS/libmlx5.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/libmlx5.spec b/SPECS/libmlx5.spec
index 3d27272..03a727d 100644
--- a/SPECS/libmlx5.spec
+++ b/SPECS/libmlx5.spec
@@ -9,7 +9,7 @@
 
 Name: libmlx5
 Version: 41mlnx1
-Release: OFED.4.2.1.2.0.42120
+Release: OFED.4.2.1.2.0.42120%{?_tis_dist}.%{tis_patch_ver}
 Summary: Mellanox ConnectX-IB InfiniBand HCA Userspace Driver
 
 Group: System Environment/Libraries
-- 
1.8.3.1

