From b16235e006dff73c7d90a5c4362ec45d976c8411 Mon Sep 17 00:00:00 2001
From: Angie Wang <Angie.Wang@windriver.com>
Date: Mon, 29 Jan 2018 13:32:42 -0500
Subject: [PATCH 1/1] meta fix panko puppet warning

---
 SPECS/puppet-panko.spec | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/SPECS/puppet-panko.spec b/SPECS/puppet-panko.spec
index f19cd32..a235ee2 100644
--- a/SPECS/puppet-panko.spec
+++ b/SPECS/puppet-panko.spec
@@ -4,7 +4,7 @@
 
 Name:                   puppet-panko
 Version:                11.3.0
-Release:                1%{?dist}
+Release:                1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:                Puppet module for OpenStack Panko Service
 License:                ASL 2.0
 
@@ -12,6 +12,8 @@ URL:                    https://launchpad.net/puppet-panko
 
 Source0:                https://tarballs.openstack.org/%{name}/%{name}-%{upstream_version}.tar.gz
 
+Patch0001:              0001-Fix-panko-puppet-warning.patch
+
 BuildArch:              noarch
 
 Requires:               puppet-inifile
@@ -26,6 +28,7 @@ Installs and configures OpenStack Panko Events Service.
 
 %prep
 %setup -q -n %{upstream_name}-%{upstream_version}
+%patch0001 -p1
 
 find . -type f -name ".*" -exec rm {} +
 find . -size 0 -exec rm {} +
-- 
1.8.3.1

