From 89fe761fdee16c6a129427b9717a7e99e0f1d55a Mon Sep 17 00:00:00 2001
From: ec <ec@starlingx.com>
Date: Thu, 19 Dec 2019 05:12:09 +0000
Subject: [PATCH] Update package versioning for TIS format

---
 SPECS/sudo.spec | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/SPECS/sudo.spec b/SPECS/sudo.spec
index c8d2f64..b6402bb 100644
--- a/SPECS/sudo.spec
+++ b/SPECS/sudo.spec
@@ -1,7 +1,7 @@
 Summary: Allows restricted root access for specified users
 Name: sudo
 Version: 1.8.23
-Release: 4%{?dist}.1
+Release: 4.el7_7.1%{?_tis_dist}.%{tis_patch_ver}
 License: ISC
 Group: Applications/System
 URL: http://www.courtesan.com/sudo/
-- 
2.7.4

