From 6013de8f9571574f6aba9431f7f51ee9ef8cf199 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 16:50:44 -0400
Subject: WRS: 0003-Further-parallelize-sudo-build.patch

---
 SPECS/sudo.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/sudo.spec b/SPECS/sudo.spec
index acbcb26..8c3f395 100644
--- a/SPECS/sudo.spec
+++ b/SPECS/sudo.spec
@@ -164,7 +164,7 @@ export CFLAGS="$RPM_OPT_FLAGS $F_PIE" LDFLAGS="-pie -Wl,-z,relro -Wl,-z,now" SHL
         --with-sssd
 #       --without-kerb5 \
 #       --without-kerb4
-make
+make -j"%(nproc)"
 
 make check
 
-- 
2.7.4

