From b0ad59c3a7992b5f171e8252cd02d1d207601f6a Mon Sep 17 00:00:00 2001
From: Giao Le <giao.le@windriver.com>
Date: Tue, 22 Nov 2016 12:53:18 -0500
Subject: [PATCH 1/1] add-worker-work-patch

---
 SPECS/python-gunicorn.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/python-gunicorn.spec b/SPECS/python-gunicorn.spec
index 86b6667..acd6b90 100644
--- a/SPECS/python-gunicorn.spec
+++ b/SPECS/python-gunicorn.spec
@@ -16,6 +16,7 @@ URL:            http://gunicorn.org/
 Source0:        http://pypi.python.org/packages/source/g/%{upstream_name}/%{upstream_name}-%{version}.tar.gz
 # distro-specific, not upstreamable
 Patch100:       %{name}-dev-log.patch
+Patch200:	0001-add-worker-abort-hook.patch
 
 BuildArch:      noarch
 BuildRequires:  python2-devel
@@ -48,6 +49,7 @@ Django, and Paster applications.
 %prep
 %setup -q -n %{upstream_name}-%{version}
 %patch100 -p1
+%patch200 -p1
 
 %if 0%{?with_python3}
 rm -rf %{py3dir}
-- 
1.8.3.1

