--- a/src/test/meson.build
+++ b/src/test/meson.build
@@ -224,9 +224,9 @@ tests += [
          [],
          []],
 
-        [['src/test/test-util.c'],
-         [],
-         []],
+#        [['src/test/test-util.c'],
+#         [],
+#         []],
 
         [['src/test/test-json.c'],
          [],
@@ -1103,10 +1103,10 @@ tests += [
           libsystemd_network],
          []],
 
-        [['src/libsystemd-network/test-dhcp-server.c'],
-         [libshared,
-          libsystemd_network],
-         []],
+#        [['src/libsystemd-network/test-dhcp-server.c'],
+#         [libshared,
+#          libsystemd_network],
+#         []],
 
         [['src/libsystemd-network/test-ipv4ll.c',
           'src/libsystemd-network/arp-util.h',
