From 43036215d0f88d6c8789a596578ea7cc0002ff61 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:05:58 -0400
Subject: [PATCH 4/9] WRS: fix-snmpd-service.patch

---
 SOURCES/snmpd.service | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/SOURCES/snmpd.service b/SOURCES/snmpd.service
index 8ec215b..84b6ca3 100644
--- a/SOURCES/snmpd.service
+++ b/SOURCES/snmpd.service
@@ -6,8 +6,9 @@ After=syslog.target network.target
 Type=notify
 Environment=OPTIONS="oamcontroller -Lsd -Lf /dev/null -p /var/run/snmpd.pid"
 EnvironmentFile=-/etc/sysconfig/snmpd
-ExecStart=/usr/sbin/snmpd $OPTIONS -f
-ExecReload=/bin/kill -HUP $MAINPID
+ExecStart=/etc/init.d/snmpd start
+ExecStop=/etc/init.d/snmpd stop
+ExecReload=/etc/init.d/snmpd restart
 
 [Install]
 WantedBy=multi-user.target
-- 
1.9.1

