From 294f4a58442180dc2787b3c67aaea8e83910ae6c Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:05:58 -0400
Subject: [PATCH 5/9] WRS: fix-snmpd-initscript.patch

---
 SPECS/net-snmp.spec | 1 -
 1 file changed, 1 deletion(-)

diff --git a/SPECS/net-snmp.spec b/SPECS/net-snmp.spec
index df248ac..a433f40 100644
--- a/SPECS/net-snmp.spec
+++ b/SPECS/net-snmp.spec
@@ -419,7 +419,6 @@ mkdir -p ${RPM_BUILD_ROOT}%{_sysconfdir}/rc.d/init.d
 install -m 755 %SOURCE13 ${RPM_BUILD_ROOT}%{_sysconfdir}/rc.d/init.d/snmpd
 
 install -d ${RPM_BUILD_ROOT}%{_initrddir}
-install -m 755 %SOURCE2 ${RPM_BUILD_ROOT}%{_initrddir}/snmpd
 install -m 755 %SOURCE3 ${RPM_BUILD_ROOT}%{_initrddir}/snmptrapd
 
 install -d ${RPM_BUILD_ROOT}%{_sysconfdir}/sysconfig
-- 
1.9.1

