From d316d562a3adc3eca5952823d18687b208b1a065 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 2 Oct 2017 17:05:58 -0400
Subject: [PATCH 2/9] WRS: snmpd-service-OPTIONS.patch

---
 SOURCES/snmpd.service | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SOURCES/snmpd.service b/SOURCES/snmpd.service
index adb394d..8ec215b 100644
--- a/SOURCES/snmpd.service
+++ b/SOURCES/snmpd.service
@@ -4,7 +4,7 @@ After=syslog.target network.target
 
 [Service]
 Type=notify
-Environment=OPTIONS="-LS0-6d"
+Environment=OPTIONS="oamcontroller -Lsd -Lf /dev/null -p /var/run/snmpd.pid"
 EnvironmentFile=-/etc/sysconfig/snmpd
 ExecStart=/usr/sbin/snmpd $OPTIONS -f
 ExecReload=/bin/kill -HUP $MAINPID
-- 
1.9.1

