diff --git a/net-snmp.spec b/net-snmp.spec
index f0c1d48..182265b 100644
--- a/SPECS/net-snmp.spec
+++ b/SPECS/net-snmp.spec
@@ -47,6 +47,7 @@ Source14: snmp.conf.cgcs
 Patch1: net-snmp-5.7.2-pie.patch
 Patch2: net-snmp-5.5-dir-fix.patch
 Patch3: net-snmp-5.6-multilib.patch
+Patch4: 0001-rpm-headerGetEntry.patch
 Patch5: net-snmp-5.6-test-debug.patch
 Patch6: net-snmp-5.7.2-systemd.patch
 Patch7: net-snmp-5.7.2-fips.patch
@@ -262,6 +263,7 @@ The net-snmp-sysvinit package provides SysV init scripts for Net-SNMP daemons.
 
 %patch2 -p1 -b .dir-fix
 %patch3 -p1 -b .multilib
+%patch4 -p1
 %patch5 -p1
 %patch6 -p1 -b .systemd
 %patch7 -p1 -b .fips
