commit de50acda387041d6810ffbd91cea3bf52b7dbc8d
Author: Shoaib Nasir <shoaib.nasir@windriver.com>
Date:   Thu Dec 7 17:36:26 2017 -0500

     WRS Patch1: 0001-Ironic-dbsync-NeutronGlance-dep-resolution.patchng

diff --git a/manifests/db/sync.pp b/manifests/db/sync.pp
index 9e4e36a..f4e3fa1 100644
--- a/manifests/db/sync.pp
+++ b/manifests/db/sync.pp
@@ -15,6 +15,17 @@ class ironic::db::sync(
   include ::ironic::deps
   include ::ironic::params
 
+  # NOTE(dtantsur): previous ironic-dbsync was run as root. it will fail to run
+  # as "ironic" user, if there is an old log file owned by root. Let's fix it.
+  # To be removed in Rocky.
+  file { '/var/log/ironic/ironic-dbsync.log':
+    ensure  => 'present',
+    owner   => 'ironic',
+    group   => 'ironic',
+    # /var/log/ironic comes from ironic-common
+    require => Anchor['ironic::install::end']
+  }
+
   exec { 'ironic-dbsync':
     command     => "${::ironic::params::dbsync_command} ${extra_params}",
     path        => '/usr/bin',
@@ -29,5 +40,6 @@ class ironic::db::sync(
       Anchor['ironic::dbsync::begin']
     ],
     notify      => Anchor['ironic::dbsync::end'],
+    require     => File['/var/log/ironic/ironic-dbsync.log'],
   }
 }
diff --git a/manifests/glance.pp b/manifests/glance.pp
index 09a65ce..bad84cd 100644
--- a/manifests/glance.pp
+++ b/manifests/glance.pp
@@ -93,19 +93,19 @@ class ironic::glance (
   $swift_account_project_name = undef,
 ) {
 
-  $api_servers_real = pick($::ironic::glance_api_servers, $api_servers)
+  $api_servers_real = $api_servers
   if is_array($api_servers_real) {
     $api_servers_converted = join($api_servers_real, ',')
   } else {
     $api_servers_converted = $api_servers_real
   }
 
-  $num_retries_real = pick($::ironic::glance_num_retries, $num_retries)
-  $api_insecure_real = pick($::ironic::glance_api_insecure, $api_insecure)
+  $num_retries_real = $num_retries
+  $api_insecure_real = $api_insecure
 
-  $swift_account_real = pick($::ironic::conductor::swift_account, $swift_account)
-  $swift_temp_url_key_real = pick($::ironic::conductor::swift_temp_url_key, $swift_temp_url_key)
-  $swift_temp_url_duration_real = pick($::ironic::conductor::swift_temp_url_duration, $swift_temp_url_duration)
+  $swift_account_real = $swift_account
+  $swift_temp_url_key_real = $swift_temp_url_key
+  $swift_temp_url_duration_real = $swift_temp_url_duration
 
 
   if ($swift_account_project_name and !is_service_default($swift_account_real)) {
diff --git a/manifests/neutron.pp b/manifests/neutron.pp
index f8cbd5d..35f5e07 100644
--- a/manifests/neutron.pp
+++ b/manifests/neutron.pp
@@ -55,7 +55,7 @@ class ironic::neutron (
   $project_domain_name = $::os_service_default,
 ) {
 
-  $api_endpoint_real = pick($::ironic::api::neutron_url, $api_endpoint)
+  $api_endpoint_real = $api_endpoint
 
   ironic_config {
     'neutron/url':                 value => $api_endpoint_real;
