From bb6fd3da3ace960eb587e7ff01d5816ea2baaa54 Mon Sep 17 00:00:00 2001
From: Jack Ding <jack.ding@windriver.com>
Date: Sun, 13 May 2018 18:22:15 -0400
Subject: [PATCH] Add dependencies and comment out incompatible service
 parameters

---
 SPECS/memcached.spec | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/SPECS/memcached.spec b/SPECS/memcached.spec
index f389035..86653a1 100644
--- a/SPECS/memcached.spec
+++ b/SPECS/memcached.spec
@@ -17,6 +17,7 @@ Source1:        memcached.sysconfig
 # https://github.com/memcached/memcached/issues/218
 Patch1:         0001-systemd-fix-upstream-provided-service.patch
 Patch2:         0002-always-restart-memcached-service.patch 
+Patch3:         0003-Add-dependencies-and-comment-out-incompatible-servic.patch
 
 BuildRequires:  libevent-devel systemd-units
 BuildRequires:  perl-generators
@@ -46,6 +47,7 @@ access to the memcached binary include files.
 %setup -q
 %patch1 -p1 -b .unit
 %patch2 -p1
+%patch3 -p1
 
 %build
 # compile with full RELRO
-- 
1.8.3.1

