From a211d6907fdef7106b32acf70e36b6795a155c5b Mon Sep 17 00:00:00 2001
From: Al Bailey <Al.Bailey@windriver.com>
Date: Thu, 4 Jan 2018 12:53:29 -0600
Subject: [PATCH 1/2] Update package versioning for TIS format

---
 SPECS/puppet-staging.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/puppet-staging.spec b/SPECS/puppet-staging.spec
index 0848033..0dd5d59 100644
--- a/SPECS/puppet-staging.spec
+++ b/SPECS/puppet-staging.spec
@@ -8,7 +8,7 @@
 
 Name:           puppet-staging
 Version:        1.0.4
-Release:        1%{?alphatag}%{?dist}
+Release:        1.b466d93git.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        Compressed file staging and deployment
 License:        Apache-2.0
 
-- 
1.8.3.1

