From 0d7ba07f1c39d47b701caf2da0699f4f217d5746 Mon Sep 17 00:00:00 2001
From: Douglas Henrique Koerich <douglashenrique.koerich@windriver.com>
Date: Mon, 1 Aug 2022 13:48:40 -0400
Subject: [PATCH 3/3] STX patches to fix phc2sys exit with 255

Signed-off-by: Douglas Henrique Koerich <douglashenrique.koerich@windriver.com>
---
 SPECS/linuxptp.spec | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/SPECS/linuxptp.spec b/SPECS/linuxptp.spec
index e4c01a8..1218533 100644
--- a/SPECS/linuxptp.spec
+++ b/SPECS/linuxptp.spec
@@ -49,6 +49,11 @@ Patch13: 0004-port-Don-t-renew-raw-transport.patch
 Patch14: 0005-clockcheck-Increase-minimum-interval.patch
 # StarlingX
 Patch15: 0006-Add-option-to-disable-default-port-selection-in-phc2.patch
+# StarlingX: Upstream commits to fix phc2sys exit with 255
+Patch16: 0007-sysoff-Change-sysoff_measure-to-return-errno.patch
+Patch17: 0008-sysoff-Change-log-level-of-ioctl-error-messages.patch
+Patch18: 0009-sysoff-Retry-on-EBUSY-when-probing-supported-ioctls.patch
+Patch19: 0010-phc2sys-Don-t-exit-when-reading-of-PHC-fails-with-EB.patch
 
 BuildRequires:	kernel-headers > 4.18.0-87
 BuildRequires:	systemd
@@ -78,6 +83,10 @@ Supporting legacy APIs and other platforms is not a goal.
 %patch13 -p1
 %patch14 -p1
 %patch15 -p1
+%patch16 -p1
+%patch17 -p1
+%patch18 -p1
+%patch19 -p1
 mv linuxptp-testsuite-%{testsuite_ver}* testsuite
 mv clknetsim-%{clknetsim_ver}* testsuite/clknetsim
 
-- 
2.29.2

