From 97cd48cae1ac9b431747211dbaf59c304bf93293 Mon Sep 17 00:00:00 2001
From: Matt Peters <matt.peters@windriver.com>
Date: Fri, 1 Jun 2018 20:17:47 -0500
Subject: [PATCH] log rotation config

---
 rhel/etc_logrotate.d_openvswitch | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/rhel/etc_logrotate.d_openvswitch b/rhel/etc_logrotate.d_openvswitch
index ed7d733..f9ae766 100644
--- a/rhel/etc_logrotate.d_openvswitch
+++ b/rhel/etc_logrotate.d_openvswitch
@@ -6,7 +6,11 @@
 # without warranty of any kind.
 
 /var/log/openvswitch/*.log {
-    daily
+    nodateext
+    size 10M
+    start 1
+    rotate 20
+    notifempty
     compress
     sharedscripts
     missingok
-- 
1.8.3.1

