From 8f08d53f0db99289b106451a8c520355a93ca2f7 Mon Sep 17 00:00:00 2001
From: Gerry Kopec <Gerry.Kopec@windriver.com>
Date: Wed, 9 Mar 2022 21:35:36 -0500
Subject: [PATCH] stx: Add status_of_proc wrapper for centos compat

---
 init-functions | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/init-functions b/init-functions
index 207b270..88662e6 100644
--- a/init-functions
+++ b/init-functions
@@ -209,6 +209,9 @@ status_of_proc () {
     fi
 }
 
+# Wrapper for centos compatibility
+status() { status_of_proc $@; };
+
 log_use_fancy_output () {
     TPUT=/usr/bin/tput
     EXPR=/usr/bin/expr
-- 
2.29.2

