From 536a805bdb2be64b91df6fe18589340ad8aec066 Mon Sep 17 00:00:00 2001
From: Don Penney <don.penney@windriver.com>
Date: Fri, 19 Jan 2018 23:36:54 -0500
Subject: [PATCH] Add su root to logrotate.conf

---
 examples/logrotate-default | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/examples/logrotate-default b/examples/logrotate-default
index 7da6bb7..c7a6f08 100644
--- a/examples/logrotate-default
+++ b/examples/logrotate-default
@@ -11,6 +11,9 @@ create
 # use date as a suffix of the rotated file
 dateext
 
+# Use su to rotate as root to ensure we don't fail due to dir permissions
+su root root
+
 # uncomment this if you want your log files compressed
 #compress
 
-- 
1.8.3.1

