From f1173c242506fb5d1b311efe2f7d657b04ae6b24 Mon Sep 17 00:00:00 2001
From: Matt Peters <matt.peters@windriver.com>
Date: Fri, 15 Jun 2018 13:37:14 -0500
Subject: [PATCH] dpdk usertools

---
 SPECS/openvswitch.spec | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/SPECS/openvswitch.spec b/SPECS/openvswitch.spec
index 71b4634..22bdec9 100644
--- a/SPECS/openvswitch.spec
+++ b/SPECS/openvswitch.spec
@@ -150,6 +150,7 @@ BuildRequires: libcap-ng libcap-ng-devel
 %ifarch %{dpdkarches}
 # DPDK driver dependencies
 BuildRequires: zlib-devel libpcap-devel numactl-devel
+Requires: python-pyelftools
 
 # Virtual provide for depending on DPDK-enabled OVS
 Provides: openvswitch-dpdk = %{version}-%{release}
@@ -421,6 +422,13 @@ install -p -D -m 0755 \
 touch $RPM_BUILD_ROOT%{_sysconfdir}/openvswitch/conf.db
 touch $RPM_BUILD_ROOT%{_sysconfdir}/openvswitch/system-id.conf
 
+%if %{with dpdk}
+%ifarch %{dpdkarches}
+  install -m 0755 %{dpdkdir}-%{dpdkver}/usertools/dpdk-pmdinfo.py $RPM_BUILD_ROOT%{_datadir}/openvswitch/scripts/dpdk-pmdinfo.py
+  install -m 0755 %{dpdkdir}-%{dpdkver}/usertools/dpdk-devbind.py $RPM_BUILD_ROOT%{_datadir}/openvswitch/scripts/dpdk-devbind.py
+%endif
+%endif
+
 # remove unpackaged files
 rm -f $RPM_BUILD_ROOT/%{_bindir}/ovs-benchmark \
         $RPM_BUILD_ROOT/%{_bindir}/ovs-docker \
@@ -656,6 +664,9 @@ fi
 %{_datadir}/openvswitch/scripts/ovs-vtep
 %{_datadir}/openvswitch/scripts/ovs-ctl
 %{_datadir}/openvswitch/scripts/ovs-systemd-reload
+%{_datadir}/openvswitch/scripts/dpdk-pmdinfo.py
+%{_datadir}/openvswitch/scripts/dpdk-devbind.py
+%exclude %{_datadir}/openvswitch/scripts/*.py[oc]
 %config %{_datadir}/openvswitch/vswitch.ovsschema
 %config %{_datadir}/openvswitch/vtep.ovsschema
 %{_bindir}/ovs-appctl
-- 
1.8.3.1

