From babffcf0f064379438fabcd00329bd431413e28a Mon Sep 17 00:00:00 2001
From: Matt Peters <matt.peters@windriver.com>
Date: Fri, 1 Jun 2018 20:13:57 -0500
Subject: [PATCH] log rotation config

---
 SPECS/openvswitch.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/openvswitch.spec b/SPECS/openvswitch.spec
index ff5694a..c51d53f 100644
--- a/SPECS/openvswitch.spec
+++ b/SPECS/openvswitch.spec
@@ -107,6 +107,7 @@ ExclusiveArch: x86_64 aarch64 ppc64le s390x
 
 # ovs-patches
 Patch01: run-services-as-root-user.patch
+Patch02: log-rotation-config.patch
 
 # OVS (including OVN) backports (0 - 300)
 
-- 
1.8.3.1

