diff --git a/SOURCES/x86_64-native-linuxapp-gcc-config b/SOURCES/x86_64-native-linuxapp-gcc-config
index f81d420..eab161c 100644
--- a/SOURCES/x86_64-native-linuxapp-gcc-config
+++ b/SOURCES/x86_64-native-linuxapp-gcc-config
@@ -197,12 +197,12 @@ CONFIG_RTE_LIBRTE_FM10K_DEBUG_DRIVER=n
 CONFIG_RTE_LIBRTE_FM10K_RX_OLFLAGS_ENABLE=y
 CONFIG_RTE_LIBRTE_FM10K_INC_VECTOR=y
 # Compile burst-oriented Mellanox ConnectX-3 (MLX4) PMD
-CONFIG_RTE_LIBRTE_MLX4_PMD=n
+CONFIG_RTE_LIBRTE_MLX4_PMD=y
 CONFIG_RTE_LIBRTE_MLX4_DEBUG=n
 CONFIG_RTE_LIBRTE_MLX4_DEBUG_BROKEN_VERBS=n
 CONFIG_RTE_LIBRTE_MLX4_TX_MP_CACHE=8
 # Compile burst-oriented Mellanox ConnectX-4 & ConnectX-5 (MLX5) PMD
-CONFIG_RTE_LIBRTE_MLX5_PMD=n
+CONFIG_RTE_LIBRTE_MLX5_PMD=y
 CONFIG_RTE_LIBRTE_MLX5_DEBUG=n
 CONFIG_RTE_LIBRTE_MLX5_TX_MP_CACHE=8
 # Compile burst-oriented Broadcom PMD driver
diff --git a/SPECS/openvswitch.spec b/SPECS/openvswitch.spec
index 29255d5..f392e95 100644
--- a/SPECS/openvswitch.spec
+++ b/SPECS/openvswitch.spec
@@ -151,6 +151,7 @@ BuildRequires: libcap-ng libcap-ng-devel
 %ifarch %{dpdkarches}
 # DPDK driver dependencies
 BuildRequires: zlib-devel libpcap-devel numactl-devel
+BuildRequires: rdma-core-devel
 Requires: python-pyelftools
 
 # Virtual provide for depending on DPDK-enabled OVS
@@ -352,7 +353,12 @@ cd -
         --dpdk \
         < rhel/usr_lib_systemd_system_ovs-vswitchd.service.in \
         > rhel/usr_lib_systemd_system_ovs-vswitchd.service
-make %{?_smp_mflags}
+make %{?_smp_mflags} \
+%if %{with dpdk}
+%ifarch %{dpdkarches}
+  LDFLAGS="-libverbs -lmlx4 -lmlx5"
+%endif
+%endif
 
 %install
 rm -rf $RPM_BUILD_ROOT
