From 5cabd4876e772f59a18a71a15f8fdfac0b0cf0ea Mon Sep 17 00:00:00 2001
From: Steven Webster <steven.webster@windriver.com>
Date: Wed, 17 Oct 2018 09:54:53 -0400
Subject: [PATCH] net-mlx5-fix-memory-region-cache-lookup

---
 SPECS/openvswitch.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/openvswitch.spec b/SPECS/openvswitch.spec
index f286a13..6d95f35 100644
--- a/SPECS/openvswitch.spec
+++ b/SPECS/openvswitch.spec
@@ -123,7 +123,7 @@ Patch422: 0003-vhost-extract-virtqueue-cleaning-and-freeing-functio.patch
 Patch423: 0004-vhost-destroy-unused-virtqueues-when-multiqueue-not-.patch
 Patch424: 0005-vhost-add-flag-for-built-in-virtio-driver.patch
 Patch425: 0006-vhost-drop-virtqueues-only-with-built-in-virtio-driv.patch
-
+Patch426: net-mlx5-fix-memory-region-cache-lookup.patch
 
 BuildRequires: gcc
 BuildRequires: python2-sphinx
-- 
1.8.3.1

