From 78489fa5c3e98ac9db6244a6707622b4603bf0cb Mon Sep 17 00:00:00 2001
From: Hayde Martinez <hayde.martinez.landa@intel.com>
Date: Thu, 11 Oct 2018 14:44:00 -0500
Subject: [PATCH] iommu-width-fix

Signed-off-by: Hayde Martinez <hayde.martinez.landa@intel.com>
---
 SPECS/openvswitch.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SPECS/openvswitch.spec b/SPECS/openvswitch.spec
index 29255d5..a3a1c87 100644
--- a/SPECS/openvswitch.spec
+++ b/SPECS/openvswitch.spec
@@ -108,6 +108,7 @@ ExclusiveArch: x86_64 aarch64 ppc64le s390x
 # ovs-patches
 Patch01: run-services-as-root-user.patch
 Patch02: log-rotation-config.patch
+Patch03: iommu-width-fix.patch
 
 # OVS (including OVN) backports (0 - 300)
 
-- 
2.7.4

