From c7352436ea37647d140f76e818f211e764170d3d Mon Sep 17 00:00:00 2001
Message-Id: <c7352436ea37647d140f76e818f211e764170d3d.1527700333.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Wed, 30 May 2018 13:12:03 -0400
Subject: [PATCH 1/1] Compile issues

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 SPECS/kernel.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index ca03e10..4dd7ae4 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -473,6 +473,8 @@ Patch40024: aic94xx-Skip-reading-user-settings-if-flash-is-not-f.patch
 Patch40025: dpt_i2o-fix-build-warning.patch
 # DRBD was choking on write same
 Patch40026: turn-off-write-same-in-smartqpi-driver.patch
+# Fix assorted compilation issues
+Patch40027: fix-compilation-issues.patch
 
 BuildRoot: %{_tmppath}/kernel-%{KVRA}-root
 
@@ -832,6 +834,7 @@ ApplyOptionalPatch US103091-IMA-System-Configuration.patch
 ApplyOptionalPatch aic94xx-Skip-reading-user-settings-if-flash-is-not-f.patch
 ApplyOptionalPatch dpt_i2o-fix-build-warning.patch
 ApplyOptionalPatch turn-off-write-same-in-smartqpi-driver.patch
+ApplyOptionalPatch fix-compilation-issues.patch
 
 # Any further pre-build tree manipulations happen here.
 
-- 
1.8.3.1

