From ec433bddc585744616d73daefc1b32326002a8ab Mon Sep 17 00:00:00 2001
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Fri, 1 Jun 2018 16:17:43 -0400
Subject: [PATCH 5/5] Lower the linux firmware version requirement

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 SPECS/kernel.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index d45c419..9fdd37a 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -301,7 +301,7 @@ Provides: kernel-modeset = 1\
 Provides: kernel-uname-r = %{KVRA}%{?1:.%{1}}\
 Requires(pre): %{kernel_prereq}\
 Requires(pre): %{initrd_prereq}\
-Requires(pre): linux-firmware >= 20180113-61\
+Requires(pre): linux-firmware >= 20170606-56\
 Requires(post): %{_sbindir}/new-kernel-pkg\
 Requires(post): system-release\
 Requires(preun): %{_sbindir}/new-kernel-pkg\
-- 
2.7.4

