From adbf49eafdcbf817f11a9decead1df92cff75543 Mon Sep 17 00:00:00 2001
Message-Id: <adbf49eafdcbf817f11a9decead1df92cff75543.1527884268.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Fri, 1 Jun 2018 16:17:43 -0400
Subject: [PATCH 1/1] Lower the linux firmware version requirement

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 SPECS/kernel.spec | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SPECS/kernel.spec b/SPECS/kernel.spec
index 4dd7ae4..89c5a74 100644
--- a/SPECS/kernel.spec
+++ b/SPECS/kernel.spec
@@ -300,7 +300,7 @@ Provides: kernel-modeset = 1\
 Provides: kernel-uname-r = %{KVRA}%{?1:.%{1}}\
 Requires(pre): %{kernel_prereq}\
 Requires(pre): %{initrd_prereq}\
-Requires(pre): linux-firmware >= 20180113-61\
+Requires(pre): linux-firmware >= 20170606-56\
 Requires(post): %{_sbindir}/new-kernel-pkg\
 Requires(post): system-release\
 Requires(preun): %{_sbindir}/new-kernel-pkg\
-- 
1.8.3.1

