From feb40bdf71992f9374e82aeb081431d2b91f1224 Mon Sep 17 00:00:00 2001
From: slin14 <shuicheng.lin@intel.com>
Date: Mon, 15 Oct 2018 23:21:15 +0800
Subject: [PATCH] add-rpm-4.14-compatibility

Signed-off-by: slin14 <shuicheng.lin@intel.com>
---
 SPECS/nfs-utils.spec | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/SPECS/nfs-utils.spec b/SPECS/nfs-utils.spec
index b866788..96fce13 100644
--- a/SPECS/nfs-utils.spec
+++ b/SPECS/nfs-utils.spec
@@ -621,14 +621,7 @@ if [ -h /etc/systemd/system/multi-user.target.wants/nfs.target ]; then
 	fi
 fi
 
-%triggerun -- nfs-utils < 1:1.2.9-0.5
-/bin/systemctl stop nfs-secure.service >/dev/null 2>&1 || :
-/bin/systemctl disable nfs-secure.service >/dev/null 2>&1 || :
-
-%triggerun -- nfs-utils < 1:1.2.4-2
-/bin/systemctl enable nfs-lock.service >/dev/null 2>&1 || :
-
-%triggerin -- nfs-utils > 1:1.3.0-0.39
+%triggerin -- nfs-utils
 # reset configuration files and running daemons
 if [ $1 -eq 2 ] ; then
 	/bin/systemctl enable nfs-client.target >/dev/null 2>&1 || :
-- 
2.7.4

