From 81d1abb89facf77ad3b1fade4384ccf3b0482eb3 Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 15 Oct 2018 23:14:20 +0800
Subject: WRS: nfs-utils-spec-file-disable-statd-service.patch

---
 SPECS/nfs-utils.spec | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/SPECS/nfs-utils.spec b/SPECS/nfs-utils.spec
index 0a5e9fb..d10c87c 100644
--- a/SPECS/nfs-utils.spec
+++ b/SPECS/nfs-utils.spec
@@ -472,6 +472,10 @@ rm $RPM_BUILD_ROOT%{_sbindir}/rpc.svcgssd
 rm $RPM_BUILD_ROOT%{_mandir}/man8/rpc.svcgssd.8
 rm $RPM_BUILD_ROOT%{_mandir}/man8/svcgssd.8
 
+# WRS - remove these service files as rpc-statd is started by nfscommon
+rm -f $RPM_BUILD_ROOT%{_unitdir}/rpc-statd.service
+rm -f $RPM_BUILD_ROOT%{_unitdir}/rpc-statd-notify.service
+
 mkdir -p $RPM_BUILD_ROOT/run/sysconfig
 mkdir -p $RPM_BUILD_ROOT/usr/lib/systemd/scripts
 mkdir -p $RPM_BUILD_ROOT%{_sysconfdir}/gssproxy
@@ -488,9 +492,6 @@ ln -s rpc-gssd.service nfs-secure.service
 ln -s rpc-gssd.service rpcgssd.service
 ln -s nfs-idmapd.service  nfs-idmap.service
 ln -s nfs-idmapd.service  rpcidmapd.service
-ln -s rpc-statd.service nfs-lock.service
-ln -s rpc-statd.service nfslock.service
-
 
 mkdir -p $RPM_BUILD_ROOT%{_sharedstatedir}/nfs/rpc_pipefs
 
@@ -555,6 +556,12 @@ fi
 %post
 if [ $1 -eq 1 ] ; then
 	# Initial installation
+        if [ -f %{_unitdir}/rpc-statd.service ] ; then
+            rm -f %{_unitdir}/rpc-statd.service
+        fi
+        if [ -f %{_unitdir}/rpc-statd-notify.service ] ; then
+            rm -f %{_unitdir}/rpc-statd-notify.service
+        fi
 	/bin/systemctl enable nfs-client.target >/dev/null 2>&1 || :
 	/bin/systemctl restart nfs-config  >/dev/null 2>&1 || :
 fi
-- 
2.7.4

