From 68ed40412a6a5caa0474715363a06914819b50bf Mon Sep 17 00:00:00 2001
From: Scott Little <scott.little@windriver.com>
Date: Mon, 15 Oct 2018 22:50:46 +0800
Subject: [PATCH] WRS: spec-include-TiS-changes.patch

---
 SPECS/nfs-utils.spec | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/SPECS/nfs-utils.spec b/SPECS/nfs-utils.spec
index 6dfddc0..0a5e9fb 100644
--- a/SPECS/nfs-utils.spec
+++ b/SPECS/nfs-utils.spec
@@ -569,6 +569,9 @@ if [ $1 -eq 0 ]; then
 	%systemd_preun nfs-client.target
 	%systemd_preun nfs-server.server
 
+       /bin/systemctl disable nfscommon.service >/dev/null 2>&1 || :
+       /bin/systemctl disable nfsserver.service >/dev/null 2>&1 || :
+
     /usr/sbin/userdel rpcuser 2>/dev/null || :
     /usr/sbin/groupdel rpcuser 2>/dev/null || :
     /usr/sbin/userdel nfsnobody 2>/dev/null || :
-- 
2.7.4

