diff --git a/SPECS/shim.spec b/SPECS/shim.spec
index 6aa8346..a8945ab 100644
--- a/SPECS/shim.spec
+++ b/SPECS/shim.spec
@@ -1,6 +1,6 @@
 Name:           shim
 Version:        0.9
-Release:        1.el7.centos
+Release:        1.el7%{?_tis_dist}.%{tis_patch_ver}
 Summary:        First-stage UEFI bootloader
 
 License:        BSD
@@ -10,8 +10,10 @@ Source1:	centos.crt
 # currently here's what's in our dbx: # nothing.
 #Source2:	dbx.esl
 Source3:	shim-find-debuginfo.sh
+Source4:	tis-shim.crt
 
 Patch0001:	0001-Typo-on-aarch64.patch
+Patch0002:	0001-Use-TiS-cert.patch
 
 BuildRequires: git openssl-devel openssl
 BuildRequires: pesign >= 0.106-1
@@ -81,6 +83,7 @@ git commit -a -q -m "%{version} baseline."
 git am --ignore-whitespace %{patches} </dev/null
 git config --unset user.email
 git config --unset user.name
+cp %{SOURCE4} .
 
 %build
 MAKEFLAGS="VENDOR_CERT_FILE=%{SOURCE1}"
